<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
    <meta charset="utf-8">
    <meta name="generator" content="quarto-1.6.40">

    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

    <meta name="author" content="Shreya Dimri">
    <meta name="dcterms.date" content="2025-04-30">

    <title>Green Nest Material Meta-Analysis</title>
    <style>
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      div.columns{display: flex; gap: min(4vw, 1.5em);}
      div.column{flex: auto; overflow-x: auto;}
      div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
      ul.task-list{list-style: none;}
      ul.task-list li input[type="checkbox"] {
        width: 0.8em;
        margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
        vertical-align: middle;
      }
      /* CSS for syntax highlighting */
      pre > code.sourceCode { white-space: pre; position: relative; }
      pre > code.sourceCode > span { line-height: 1.25; }
      pre > code.sourceCode > span:empty { height: 1.2em; }
      .sourceCode { overflow: visible; }
      code.sourceCode > span { color: inherit; text-decoration: inherit; }
      div.sourceCode { margin: 1em 0; }
      pre.sourceCode { margin: 0; }
      @media screen {
      div.sourceCode { overflow: auto; }
      }
      @media print {
      pre > code.sourceCode { white-space: pre-wrap; }
      pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
      }
      pre.numberSource code
        { counter-reset: source-line 0; }
      pre.numberSource code > span
        { position: relative; left: -4em; counter-increment: source-line; }
      pre.numberSource code > span > a:first-child::before
        { content: counter(source-line);
          position: relative; left: -1em; text-align: right; vertical-align: baseline;
          border: none; display: inline-block;
          -webkit-touch-callout: none; -webkit-user-select: none;
          -khtml-user-select: none; -moz-user-select: none;
          -ms-user-select: none; user-select: none;
          padding: 0 4px; width: 4em;
        }
      pre.numberSource { margin-left: 3em;  padding-left: 4px; }
      div.sourceCode
        {   }
      @media screen {
      pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
      }
    </style>

    <style>
      body.hypothesis-enabled #quarto-embed-header {
        padding-right: 36px;
      }

      #quarto-embed-header {
        height: 3em;
        width: 100%;
        display: flex;
        justify-content: space-between;
        align-items: center;
        border-bottom: solid 1px;
      }

      #quarto-embed-header h6 {
        font-size: 1.1em;
        padding-top: 0.6em;
        margin-left: 1em;
        margin-right: 1em;
        font-weight: 400;
      }

      #quarto-embed-header a.quarto-back-link,
      #quarto-embed-header a.quarto-download-embed {
        font-size: 0.8em;
        margin-top: 1em;
        margin-bottom: 1em;
        margin-left: 1em;
        margin-right: 1em;
      }

      .quarto-back-container {
        padding-left: 0.5em;
        display: flex;
      }

      .headroom {
          will-change: transform;
          transition: transform 200ms linear;
      }

      .headroom--pinned {
          transform: translateY(0%);
      }

      .headroom--unpinned {
          transform: translateY(-100%);
      }      
    </style>

    <script>
    window.document.addEventListener("DOMContentLoaded", function () {

      var header = window.document.querySelector("#quarto-embed-header");
      const titleBannerEl = window.document.querySelector("body > #title-block-header");
      if (titleBannerEl) {
        titleBannerEl.style.paddingTop = header.clientHeight + "px";
      }
      const contentEl = window.document.getElementById('quarto-content');
      for (const child of contentEl.children) {
        child.style.paddingTop = header.clientHeight + "px";
        child.style.marginTop = "1em";
      }

      // Use the article root if the `back` call doesn't work. This isn't perfect
      // but should typically work
      window.quartoBackToArticle = () => {
        var currentUrl = window.location.href;
        window.history.back();
        setTimeout(() => {
            // if location was not changed in 100 ms, then there is no history back
            if(currentUrl === window.location.href){              
                // redirect to site root
                window.location.href = "../index.html";
            }
        }, 100);
      }

      const headroom = new window.Headroom(header, {
        tolerance: 5,
        onPin: function () {
        },
        onUnpin: function () {
        },
      });
      headroom.init();
    });
    </script>

    
<script src="../site_libs/manuscript-notebook/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-86daaaaad7353f9cc0c554efc1dd6d94.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-55d5ce056ca1cdc2899b79b01b4a9390.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
     <script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>  <link href="../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../site_libs/pagedtable-1.1/js/pagedtable.js"></script>   <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script> 
      </head>

  <body class="quarto-notebook">
    <div id="quarto-embed-header" class="headroom fixed-top bg-primary">
      
      <a onclick="window.quartoBackToArticle(); return false;" class="btn btn-primary quarto-back-link" href=""><i class="bi bi-caret-left"></i> Back to Article</a>
      <h6><i class="bi bi-journal-code"></i> Green Nest Material Meta-Analysis</h6>

            <a href="../notebook/data_analysis.qmd" class="btn btn-primary quarto-download-embed" download="data_analysis.qmd">Download Source</a>
          </div>

     <header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Green Nest Material Meta-Analysis</h1>
            <p class="subtitle lead">The adaptive value of green nest material across birds: A systematic review and meta-analysis</p>
          </div>

    
    <div class="quarto-title-meta-container">
      <div class="quarto-title-meta-column-start">
        
        <div class="quarto-title-meta">

                <div>
            <div class="quarto-title-meta-heading">Author</div>
            <div class="quarto-title-meta-contents">
                        <p>Shreya Dimri </p>
                      </div>
          </div>
                
                <div>
            <div class="quarto-title-meta-heading">Published</div>
            <div class="quarto-title-meta-contents">
              <p class="date">April 30, 2025</p>
            </div>
          </div>
          
                
              </div>
      </div>
      <div class="quarto-title-meta-column-end quarto-other-formats-target">
      </div>
    </div>



    <div class="quarto-other-links-text-target">
    </div>  </div>
</header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active" data-toc-expanded="3">
    <h2 id="toc-title">Table of Contents</h2>
   
  <ul class="collapse">
  <li><a href="#preparing-datasets" id="toc-preparing-datasets" class="nav-link active" data-scroll-target="#preparing-datasets">Preparing datasets</a></li>
  <li><a href="#overall-effect-intercept-only-meta-analytic-model" id="toc-overall-effect-intercept-only-meta-analytic-model" class="nav-link" data-scroll-target="#overall-effect-intercept-only-meta-analytic-model">Overall effect : Intercept-only meta-analytic model</a>
  <ul class="collapse">
  <li><a href="#using-lnrr" id="toc-using-lnrr" class="nav-link" data-scroll-target="#using-lnrr">using lnRR</a></li>
  <li><a href="#lnrr-heterogeneity" id="toc-lnrr-heterogeneity" class="nav-link" data-scroll-target="#lnrr-heterogeneity">lnRR-heterogeneity</a></li>
  <li><a href="#using-smdh" id="toc-using-smdh" class="nav-link" data-scroll-target="#using-smdh">using SMDH</a></li>
  <li><a href="#smdh-heterogeneity" id="toc-smdh-heterogeneity" class="nav-link" data-scroll-target="#smdh-heterogeneity">SMDH-heterogeneity</a></li>
  <li><a href="#bivariate-meta-analytical-model" id="toc-bivariate-meta-analytical-model" class="nav-link" data-scroll-target="#bivariate-meta-analytical-model">Bivariate Meta-analytical model</a></li>
  </ul></li>
  <li><a href="#sensitivity-analysis" id="toc-sensitivity-analysis" class="nav-link" data-scroll-target="#sensitivity-analysis">Sensitivity Analysis</a></li>
  <li><a href="#mechanistic-hypothesis" id="toc-mechanistic-hypothesis" class="nav-link" data-scroll-target="#mechanistic-hypothesis">Mechanistic Hypothesis</a></li>
  <li><a href="#exploratory-analysis" id="toc-exploratory-analysis" class="nav-link" data-scroll-target="#exploratory-analysis">Exploratory Analysis</a></li>
  <li><a href="#sensitivity-analyses-robustness-checks" id="toc-sensitivity-analyses-robustness-checks" class="nav-link" data-scroll-target="#sensitivity-analyses-robustness-checks"><strong>Sensitivity analyses / robustness checks</strong></a>
  <ul class="collapse">
  <li><a href="#other-analysis" id="toc-other-analysis" class="nav-link" data-scroll-target="#other-analysis">Other Analysis</a></li>
  <li><a href="#heterogeneity" id="toc-heterogeneity" class="nav-link" data-scroll-target="#heterogeneity">Heterogeneity</a></li>
  <li><a href="#publication-bias" id="toc-publication-bias" class="nav-link" data-scroll-target="#publication-bias">Publication Bias</a></li>
  </ul></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content quarto-banner-title-block" id="quarto-document-content">      

       <div class="cell-container code-fold"><div class="cell-decorator"><pre>In [1]:</pre></div><div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Setting up workspace</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co"># For plotting trees, and although not strictly necessary, we may need the `ggtree` R package, which you should be able to install by running:</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co"># if (!require("BiocManager", quietly = TRUE))</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages("BiocManager")</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install("ggtree")</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="co">#devtools::install_github("daniel1noble/orchaRd", ref = "main", force = TRUE)</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(here, tidyverse, metafor,ggpubr,orchaRd,ggpubfigs) <span class="co">#Packages needed</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'ggpubfigs' is not available for this version of R

A version of this package for your version of R might be available elsewhere,
see the ideas at
https://cran.r-project.org/doc/manuals/r-patched/R-admin.html#Installing-packages</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in p_install(package, character.only = TRUE, ...):</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in library(package, lib.loc = lib.loc, character.only = TRUE,
logical.return = TRUE, : there is no package called 'ggpubfigs'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in pacman::p_load(here, tidyverse, metafor, ggpubr, orchaRd, ggpubfigs): Failed to install/load:
ggpubfigs</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list=</span><span class="fu">ls</span>()) <span class="do">## cleaning up</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">430</span>) <span class="co">#set seed for reproducibility (for random number generation)</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Loading cleaned dataset</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>dataset_analysis<span class="ot">&lt;-</span> <span class="fu">read_csv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/03_data_cleaning/dataset_analysis.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 279 Columns: 71
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (35): paper_ID, fulltext_screening, variable_note, authors, population_l...
dbl (35): year_publication, Observation_ID, experiment_ID, group_ID, repeate...
lgl  (1): fulltext_notes

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Functions needed</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="co"># function to estimate typical sampling error variance obtained from </span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="co"># https://github.com/Yefeng0920/heterogeneity_ecoevo/tree/main/function</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>sigma2_v <span class="ot">&lt;-</span> <span class="cf">function</span>(mod){</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  sigma2_v <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="dv">1</span> <span class="sc">/</span> mod<span class="sc">$</span>vi) <span class="sc">*</span> (mod<span class="sc">$</span>k <span class="sc">-</span> <span class="dv">1</span>) <span class="sc">/</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>    (<span class="fu">sum</span>(<span class="dv">1</span> <span class="sc">/</span> mod<span class="sc">$</span>vi)<span class="sc">^</span><span class="dv">2</span> <span class="sc">-</span> <span class="fu">sum</span>((<span class="dv">1</span> <span class="sc">/</span> mod<span class="sc">$</span>vi)<span class="sc">^</span><span class="dv">2</span>))</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(sigma2_v)</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<section id="creating-phylogeny-for-the-plot" class="level3">
<h3 class="anchored" data-anchor-id="creating-phylogeny-for-the-plot">Creating Phylogeny for the plot</h3>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [2]:</pre></div><div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages("pacman")</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="co"># load packages</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(rotl,</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>               ape,</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>               ggtree)</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Doesn't matter if I take dataset_lnRR or dataset_SMDH</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>bird_species <span class="ot">&lt;-</span> <span class="fu">unique</span>(dataset_analysis<span class="sc">$</span>bird_species)</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a><span class="co"># extracting taxonomic information</span></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>taxa <span class="ot">&lt;-</span> rotl<span class="sc">::</span><span class="fu">tnrs_match_names</span>(<span class="at">names =</span> bird_species)</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>taxa</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["search_string"],"name":[1],"type":["chr"],"align":["left"]},{"label":["unique_name"],"name":[2],"type":["chr"],"align":["left"]},{"label":["approximate_match"],"name":[3],"type":["lgl"],"align":["right"]},{"label":["score"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["ott_id"],"name":[5],"type":["int"],"align":["right"]},{"label":["is_synonym"],"name":[6],"type":["lgl"],"align":["right"]},{"label":["flags"],"name":[7],"type":["chr"],"align":["left"]},{"label":["number_matches"],"name":[8],"type":["int"],"align":["right"]}],"data":[{"1":"cyanistes caeruleus","2":"Cyanistes caeruleus","3":"FALSE","4":"1","5":"746120","6":"FALSE","7":"","8":"1"},{"1":"sturnus unicolor","2":"Sturnus unicolor","3":"FALSE","4":"1","5":"366470","6":"FALSE","7":"sibling_higher","8":"1"},{"1":"tachycineta bicolor","2":"Tachycineta bicolor","3":"FALSE","4":"1","5":"136028","6":"FALSE","7":"","8":"1"},{"1":"sturnus vulgaris","2":"Sturnus vulgaris","3":"FALSE","4":"1","5":"565813","6":"FALSE","7":"sibling_higher","8":"1"},{"1":"parus major","2":"Parus major","3":"FALSE","4":"1","5":"515143","6":"FALSE","7":"","8":"1"},{"1":"passer cinnamomeus","2":"Passer cinnamomeus","3":"FALSE","4":"1","5":"7661508","6":"FALSE","7":"","8":"1"},{"1":"buteo buteo","2":"Buteo buteo","3":"FALSE","4":"1","5":"119214","6":"FALSE","7":"sibling_higher","8":"1"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># check approximate matches</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>taxa[taxa<span class="sc">$</span>approximate_match<span class="sc">==</span><span class="cn">TRUE</span> <span class="sc">&amp;</span> <span class="sc">!</span>(<span class="fu">is.na</span>(taxa<span class="sc">$</span>approximate_match)),]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["search_string"],"name":[1],"type":["chr"],"align":["left"]},{"label":["unique_name"],"name":[2],"type":["chr"],"align":["left"]},{"label":["approximate_match"],"name":[3],"type":["lgl"],"align":["right"]},{"label":["score"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["ott_id"],"name":[5],"type":["int"],"align":["right"]},{"label":["is_synonym"],"name":[6],"type":["lgl"],"align":["right"]},{"label":["flags"],"name":[7],"type":["chr"],"align":["left"]},{"label":["number_matches"],"name":[8],"type":["int"],"align":["right"]}],"data":[],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># phylogenetic information for our species from the Open Tree of Life</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>tree <span class="ot">&lt;-</span> rotl<span class="sc">::</span><span class="fu">tol_induced_subtree</span>(<span class="at">ott_ids =</span> taxa[[<span class="st">"ott_id"</span>]], <span class="at">label_format =</span> <span class="st">"name"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Progress [----------------------------------] 0/99 (  0%) ?s
Progress [================================] 99/99 (100%)  0s
                                                            </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in collapse_singles(tr, show_progress): Dropping singleton nodes with
labels: mrcaott246ott2907, mrcaott246ott3600042, mrcaott246ott47588,
mrcaott246ott7113, Passeriformes, mrcaott246ott3212, mrcaott246ott428578,
mrcaott246ott44866, mrcaott246ott5929, mrcaott246ott32658, mrcaott246ott4820,
mrcaott246ott310390, mrcaott246ott176461, mrcaott246ott22325,
mrcaott246ott10351, mrcaott246ott193904, mrcaott246ott5935, mrcaott246ott1566,
mrcaott1566ott3598440, mrcaott1566ott496009, mrcaott1566ott2175,
mrcaott2175ott600902, mrcaott2175ott259082, mrcaott2175ott59905,
mrcaott2175ott2224, mrcaott2224ott366470, mrcaott3364ott310375,
mrcaott3364ott3370, mrcaott4083ott35042, mrcaott4083ott370807,
mrcaott4083ott469177, mrcaott4083ott18592, mrcaott4083ott4088,
mrcaott4088ott9416, mrcaott9416ott840030, mrcaott9416ott96148,
mrcaott9416ott7068473, Passer, mrcaott9416ott7661508, mrcaott1488ott72472,
mrcaott1488ott63797, mrcaott1488ott284404, mrcaott1488ott107463,
mrcaott1488ott17016, mrcaott1488ott107476, mrcaott1488ott104673,
mrcaott1488ott16185, mrcaott16185ott274931, Hirundinidae, mrcaott16185ott67916,
mrcaott16185ott16194, mrcaott16185ott136028, mrcaott136028ott1050413,
Tachycineta, mrcaott136028ott621794, mrcaott2375ott73144, mrcaott2375ott124085,
mrcaott2375ott71358, mrcaott2375ott814750, mrcaott2375ott61147,
mrcaott84656ott325811, mrcaott84656ott325806, mrcaott84656ott875992, Parus,
mrcaott84656ott3598691, mrcaott123763ott995871, mrcaott123763ott258794,
Accipitriformes, mrcaott1858ott1036186, mrcaott1858ott806938, Accipitridae,
Accipitrinae, mrcaott1858ott47576, mrcaott1858ott103122, mrcaott1858ott8285,
mrcaott8285ott34284, mrcaott8285ott11582, mrcaott8285ott919198,
mrcaott8285ott68001, mrcaott8285ott243416, mrcaott8285ott327100,
mrcaott8285ott376409, mrcaott8285ott33798, mrcaott8285ott33791,
mrcaott8285ott1008547, mrcaott8285ott1034654, mrcaott8285ott33792,
mrcaott33792ott458092, mrcaott33792ott120833, mrcaott33792ott34339,
mrcaott33792ott458099, mrcaott33792ott33796, mrcaott33796ott34349,
mrcaott33796ott33797, mrcaott33796ott554772, mrcaott33796ott215782,
mrcaott215782ott236917, mrcaott215782ott433865, mrcaott215782ott805051</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>ape<span class="sc">::</span><span class="fu">is.binary</span>(tree)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># here are the species included in the tree</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sort</span>(tree<span class="sc">$</span>tip.label)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Buteo_buteo"         "Cyanistes_caeruleus" "Parus_major"        
[4] "Passer_cinnamomeus"  "Sturnus_unicolor"    "Sturnus_vulgaris"   
[7] "Tachycineta_bicolor"</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># removing the underscore "_" from the tree tip.label</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>tree<span class="sc">$</span>tip.label <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_"</span>,<span class="st">" "</span>, tree<span class="sc">$</span>tip.label)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="fu">sort</span>(tree<span class="sc">$</span>tip.label)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Buteo buteo"         "Cyanistes caeruleus" "Parus major"        
[4] "Passer cinnamomeus"  "Sturnus unicolor"    "Sturnus vulgaris"   
[7] "Tachycineta bicolor"</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># compute branch lengths of tree</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>phylo_branch <span class="ot">&lt;-</span> <span class="fu">compute.brlen</span>(tree, <span class="at">method =</span> <span class="st">"Grafen"</span>, <span class="at">power =</span> <span class="dv">1</span>)</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="co"># check tree is ultrametric</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a><span class="fu">is.ultrametric</span>(phylo_branch) <span class="co"># TRUE</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># matrix to be included in the models</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>bird_phylo <span class="ot">&lt;-</span> <span class="fu">vcv</span>(phylo_branch, <span class="at">cor =</span> T)</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>bird_phylo</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>                    Sturnus unicolor Sturnus vulgaris Passer cinnamomeus
Sturnus unicolor           1.0000000        0.8333333          0.6666667
Sturnus vulgaris           0.8333333        1.0000000          0.6666667
Passer cinnamomeus         0.6666667        0.6666667          1.0000000
Tachycineta bicolor        0.1666667        0.1666667          0.1666667
Parus major                0.1666667        0.1666667          0.1666667
Cyanistes caeruleus        0.1666667        0.1666667          0.1666667
Buteo buteo                0.0000000        0.0000000          0.0000000
                    Tachycineta bicolor Parus major Cyanistes caeruleus
Sturnus unicolor              0.1666667   0.1666667           0.1666667
Sturnus vulgaris              0.1666667   0.1666667           0.1666667
Passer cinnamomeus            0.1666667   0.1666667           0.1666667
Tachycineta bicolor           1.0000000   0.6666667           0.6666667
Parus major                   0.6666667   1.0000000           0.8333333
Cyanistes caeruleus           0.6666667   0.8333333           1.0000000
Buteo buteo                   0.0000000   0.0000000           0.0000000
                    Buteo buteo
Sturnus unicolor              0
Sturnus vulgaris              0
Passer cinnamomeus            0
Tachycineta bicolor           0
Parus major                   0
Cyanistes caeruleus           0
Buteo buteo                   1</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(tree) <span class="co">#check: https://www.rdocumentation.org/packages/ape/versions/5.3/topics/plot.phylo</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="data_analysis_files/figure-html/unnamed-chunk-1-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1"><img src="data_analysis_files/figure-html/unnamed-chunk-1-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
</div></div>
</section>
<section id="visualizing-random-effects-structure" class="level3">
<h3 class="anchored" data-anchor-id="visualizing-random-effects-structure">Visualizing random effects structure</h3>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [3]:</pre></div><div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages("devtools")</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="co">#devtools::install_github("davidsjoberg/ggsankey")</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggsankey)</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>sankey_data <span class="ot">&lt;-</span> dataset_analysis <span class="sc">%&gt;%</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Observation_ID, repeated_trait_ID_coded, paper_ID, experiment_ID_coded, group_ID_coded)<span class="sc">%&gt;%</span></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(Observation_ID, repeated_trait_ID_coded, paper_ID, experiment_ID_coded, group_ID_coded), as.character)) <span class="sc">%&gt;%</span></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">make_long</span>(Observation_ID, repeated_trait_ID_coded, paper_ID, experiment_ID_coded, group_ID_coded)</span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>sankey_data <span class="ot">&lt;-</span> dataset_analysis <span class="sc">%&gt;%</span></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(paper_ID, bird_species, population_ID)<span class="sc">%&gt;%</span></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(paper_ID, bird_species, population_ID), as.character)) <span class="sc">%&gt;%</span></span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">make_long</span>(paper_ID, bird_species, population_ID)</span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(sankey_data, <span class="fu">aes</span>(<span class="at">x =</span> x, </span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a>               <span class="at">next_x =</span> next_x, </span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>               <span class="at">node =</span> node, </span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a>               <span class="at">next_node =</span> next_node,</span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a>               <span class="at">fill =</span> <span class="fu">factor</span>(node))) <span class="sc">+</span></span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_sankey</span>() <span class="sc">+</span></span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_discrete</span>(<span class="at">drop=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="data_analysis_files/figure-html/unnamed-chunk-2-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-2"><img src="data_analysis_files/figure-html/unnamed-chunk-2-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(sankey_data, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">next_x =</span> next_x, <span class="at">node =</span> node, <span class="at">next_node =</span> next_node, <span class="at">fill =</span> <span class="fu">factor</span>(node))) <span class="sc">+</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_alluvial</span>(<span class="at">flow.alpha =</span> .<span class="dv">6</span>) <span class="sc">+</span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis_d</span>(<span class="at">drop =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_alluvial</span>(<span class="at">base_size =</span> <span class="dv">18</span>) <span class="sc">+</span></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> .<span class="dv">5</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="data_analysis_files/figure-html/unnamed-chunk-2-2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-3"><img src="data_analysis_files/figure-html/unnamed-chunk-2-2.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
</div></div>
</section>
<section id="preparing-datasets" class="level1">
<h1>Preparing datasets</h1>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [4]:</pre></div><div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="do">## For lnRR</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="co"># selecting dataset which are not flagged</span></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>dataset_lnRR<span class="ot">&lt;-</span>dataset_analysis<span class="sc">%&gt;%</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>   <span class="fu">filter</span>(lnRR_sign <span class="sc">!=</span> <span class="st">"NA"</span> <span class="sc">|</span> lnRR_variance <span class="sc">!=</span> <span class="st">"NA"</span>)<span class="sc">%&gt;%</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(proxy_decision <span class="sc">==</span> <span class="st">"include"</span>)</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a><span class="co"># selecting dataset which are flagged</span></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>lnRR_con_flagged<span class="ot">&lt;-</span>dataset_analysis<span class="sc">%&gt;%</span></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>   <span class="fu">filter</span>(lnRR_sign <span class="sc">!=</span> <span class="st">"NA"</span> <span class="sc">|</span> lnRR_variance <span class="sc">!=</span> <span class="st">"NA"</span>)<span class="sc">%&gt;%</span></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(proxy_decision <span class="sc">!=</span> <span class="st">"SMDH only"</span> <span class="sc">|</span> proxy_decision <span class="sc">!=</span> <span class="st">"contingency table"</span> )</span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a><span class="co"># selecting dataset without those using 0 as ES because authors reported no effect</span></span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>lnRR_sin_missingES <span class="ot">&lt;-</span> dataset_lnRR <span class="sc">%&gt;%</span> </span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a>   <span class="fu">filter</span>(<span class="sc">!</span>proxy_comment <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"use 0 as ES"</span>))</span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a>lnRR_flagged_no_missingES<span class="ot">&lt;-</span> lnRR_con_flagged <span class="sc">%&gt;%</span> </span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a>   <span class="fu">filter</span>(<span class="sc">!</span>proxy_comment <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"use 0 as ES"</span>))</span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a><span class="do">## For SMDH</span></span>
<span id="cb27-20"><a href="#cb27-20" aria-hidden="true" tabindex="-1"></a><span class="co"># selecting dataset which are not flagged</span></span>
<span id="cb27-21"><a href="#cb27-21" aria-hidden="true" tabindex="-1"></a>dataset_SMDH<span class="ot">&lt;-</span>dataset_analysis<span class="sc">%&gt;%</span></span>
<span id="cb27-22"><a href="#cb27-22" aria-hidden="true" tabindex="-1"></a>   <span class="fu">filter</span>(proxy_decision <span class="sc">==</span> <span class="st">"include"</span><span class="sc">|</span> proxy_decision <span class="sc">==</span> <span class="st">"SMDH only"</span><span class="sc">|</span>proxy_decision <span class="sc">==</span><span class="st">"contingency table"</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(SMDH_sign <span class="sc">!=</span> <span class="st">"NA"</span>)</span>
<span id="cb27-23"><a href="#cb27-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-24"><a href="#cb27-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-25"><a href="#cb27-25" aria-hidden="true" tabindex="-1"></a><span class="co"># selecting dataset which are flagged</span></span>
<span id="cb27-26"><a href="#cb27-26" aria-hidden="true" tabindex="-1"></a>SMDH_con_flagged<span class="ot">&lt;-</span>dataset_analysis<span class="sc">%&gt;%</span></span>
<span id="cb27-27"><a href="#cb27-27" aria-hidden="true" tabindex="-1"></a>   <span class="fu">filter</span>(SMDH_sign <span class="sc">!=</span> <span class="st">"NA"</span>)</span>
<span id="cb27-28"><a href="#cb27-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-29"><a href="#cb27-29" aria-hidden="true" tabindex="-1"></a><span class="co"># selecting dataset without those using 0 as ES because authors reported no effect</span></span>
<span id="cb27-30"><a href="#cb27-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-31"><a href="#cb27-31" aria-hidden="true" tabindex="-1"></a>SMDH_sin_missingES <span class="ot">&lt;-</span> dataset_SMDH <span class="sc">%&gt;%</span> </span>
<span id="cb27-32"><a href="#cb27-32" aria-hidden="true" tabindex="-1"></a>   <span class="fu">filter</span>(<span class="sc">!</span>proxy_comment <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"use 0 as ES"</span>))</span>
<span id="cb27-33"><a href="#cb27-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-34"><a href="#cb27-34" aria-hidden="true" tabindex="-1"></a>SMDH_flagged_no_missingES<span class="ot">&lt;-</span> SMDH_con_flagged <span class="sc">%&gt;%</span> </span>
<span id="cb27-35"><a href="#cb27-35" aria-hidden="true" tabindex="-1"></a>   <span class="fu">filter</span>(<span class="sc">!</span>proxy_comment <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"use 0 as ES"</span>))</span>
<span id="cb27-36"><a href="#cb27-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-37"><a href="#cb27-37" aria-hidden="true" tabindex="-1"></a><span class="co"># selecting dataset which are not flagged but without the SMDH calculated from inferential statistics </span></span>
<span id="cb27-38"><a href="#cb27-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-39"><a href="#cb27-39" aria-hidden="true" tabindex="-1"></a>SMDH_no_inferential<span class="ot">&lt;-</span>dataset_analysis<span class="sc">%&gt;%</span></span>
<span id="cb27-40"><a href="#cb27-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(proxy_decision <span class="sc">==</span> <span class="st">"include"</span>)<span class="sc">%&gt;%</span> <span class="fu">filter</span>(SMDH_sign <span class="sc">!=</span> <span class="st">"NA"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
</section>
<section id="overall-effect-intercept-only-meta-analytic-model" class="level1">
<h1>Overall effect : Intercept-only meta-analytic model</h1>
<p><em>“Does adding green material to the nest have an adaptive function across the bird species that perform this behaviour?”</em></p>
<section id="using-lnrr" class="level2">
<h2 class="anchored" data-anchor-id="using-lnrr">using lnRR</h2>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [5]:</pre></div><div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>VCV_lnRR<span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> lnRR_variance,</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_lnRR,</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>intercept_lnRR<span class="ot">&lt;-</span> <span class="fu">rma.mv</span>(<span class="at">yi =</span> lnRR_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_lnRR, <span class="co"># specify lnRR's sampling variance-covariance matrix;</span></span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> dataset_lnRR)</span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the model</span></span>
<span id="cb28-21"><a href="#cb28-21" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(intercept_lnRR, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/intercept_lnRR.rds"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<p>Exploring heterogeneity in intercept only model</p>
<p>This is the function from Yefeng’s Github for interpretting heterogeneity. I have to download the data used for this function to be able to use it. For now I am using the values from the table directly so I should just delete it later before finishing the code.</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [6]:</pre></div><div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ## Function is from the Pluralistic approach to heterogeneity from Yefeng's Pre-print repository from GitHub</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="co"># #-----------------------heterogeneity interpretation-----------------------#</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a><span class="co"># het_interpret &lt;- function(observed_value, het_type, es_type, data) {</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a><span class="co">#   if (!het_type %in% c("I2", "CVH", "M", "sigma2", "V_bar")) {</span></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a><span class="co">#     stop("Invalid heterogeneity type. Choose from 'I2', 'CVH', 'M', 'sigma2', or 'V_bar'.")</span></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a><span class="co">#   }</span></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a><span class="co">#   if (!es_type %in% unique(data$es.type)) {</span></span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a><span class="co">#     stop("Invalid effect size type. Check the levels of 'es.type'.")</span></span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a><span class="co">#   }</span></span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a><span class="co">#   filtered_data &lt;- data %&gt;% filter(es.type == es_type)</span></span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a><span class="co">#   if (nrow(filtered_data) == 0) {</span></span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a><span class="co">#     stop("No data available for the selected effect size type.")</span></span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a><span class="co">#   }</span></span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb29-20"><a href="#cb29-20" aria-hidden="true" tabindex="-1"></a><span class="co">#   het_values &lt;- filtered_data[[het_type]]</span></span>
<span id="cb29-21"><a href="#cb29-21" aria-hidden="true" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb29-22"><a href="#cb29-22" aria-hidden="true" tabindex="-1"></a><span class="co">#   percentiles &lt;- quantile(het_values, probs = seq(0, 1, by = 0.05), na.rm = TRUE)</span></span>
<span id="cb29-23"><a href="#cb29-23" aria-hidden="true" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb29-24"><a href="#cb29-24" aria-hidden="true" tabindex="-1"></a><span class="co">#   lower_bound &lt;- max(percentiles[percentiles &lt;= observed_value], na.rm = TRUE)</span></span>
<span id="cb29-25"><a href="#cb29-25" aria-hidden="true" tabindex="-1"></a><span class="co">#   upper_bound &lt;- min(percentiles[percentiles &gt;= observed_value], na.rm = TRUE)</span></span>
<span id="cb29-26"><a href="#cb29-26" aria-hidden="true" tabindex="-1"></a><span class="co">#   lower_percentile &lt;- names(percentiles)[percentiles == lower_bound]</span></span>
<span id="cb29-27"><a href="#cb29-27" aria-hidden="true" tabindex="-1"></a><span class="co">#   upper_percentile &lt;- names(percentiles)[percentiles == upper_bound]</span></span>
<span id="cb29-28"><a href="#cb29-28" aria-hidden="true" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb29-29"><a href="#cb29-29" aria-hidden="true" tabindex="-1"></a><span class="co">#   percentile_range &lt;- paste0(lower_percentile, "-", upper_percentile, "th percentile")</span></span>
<span id="cb29-30"><a href="#cb29-30" aria-hidden="true" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb29-31"><a href="#cb29-31" aria-hidden="true" tabindex="-1"></a><span class="co">#   # return the results</span></span>
<span id="cb29-32"><a href="#cb29-32" aria-hidden="true" tabindex="-1"></a><span class="co">#   return(tibble::tibble(</span></span>
<span id="cb29-33"><a href="#cb29-33" aria-hidden="true" tabindex="-1"></a><span class="co">#     observed_value = observed_value,</span></span>
<span id="cb29-34"><a href="#cb29-34" aria-hidden="true" tabindex="-1"></a><span class="co">#     het_type = het_type,</span></span>
<span id="cb29-35"><a href="#cb29-35" aria-hidden="true" tabindex="-1"></a><span class="co">#     es_type = es_type,</span></span>
<span id="cb29-36"><a href="#cb29-36" aria-hidden="true" tabindex="-1"></a><span class="co">#     percentile_range = percentile_range</span></span>
<span id="cb29-37"><a href="#cb29-37" aria-hidden="true" tabindex="-1"></a><span class="co">#   ))</span></span>
<span id="cb29-38"><a href="#cb29-38" aria-hidden="true" tabindex="-1"></a><span class="co"># }</span></span>
<span id="cb29-39"><a href="#cb29-39" aria-hidden="true" tabindex="-1"></a><span class="co"># x&lt;-het_interpret(observed_value= I2_main_model_vector[1], </span></span>
<span id="cb29-40"><a href="#cb29-40" aria-hidden="true" tabindex="-1"></a><span class="co">#                  het_type = "I2", </span></span>
<span id="cb29-41"><a href="#cb29-41" aria-hidden="true" tabindex="-1"></a><span class="co">#                  es_type = "lnRR_sign", </span></span>
<span id="cb29-42"><a href="#cb29-42" aria-hidden="true" tabindex="-1"></a><span class="co">#                  data= intercept_lnRR)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
</section>
<section id="lnrr-heterogeneity" class="level2">
<h2 class="anchored" data-anchor-id="lnrr-heterogeneity">lnRR-heterogeneity</h2>
<p>Let us look at the heterogeneity in lnRR intercept-only model</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [7]:</pre></div><div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Then also:</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="co"># # Typical sampling variance (which captures the statistical noise of the data, </span></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="co"># # but is rarely reported in the current meta-analytic practice):</span></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a><span class="fu">sigma2_v</span>(intercept_lnRR)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.0003310316</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># To calculate heterogeneity, these are the functions from orchaRd:</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="co"># # I2, CV and M</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>sigma2_main_model_vector <span class="ot">&lt;-</span> <span class="fu">sum</span>(intercept_lnRR<span class="sc">$</span>sigma2)</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a><span class="co"># 0.11, 0.27, and 0.57 for lnRR</span></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>I2_main_model_vector <span class="ot">&lt;-</span> orchaRd<span class="sc">::</span><span class="fu">i2_ml</span>(intercept_lnRR)</span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a><span class="co"># 0.88, 0.95, and 0.99 for lnRR</span></span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>CV_main_model_vector <span class="ot">&lt;-</span> orchaRd<span class="sc">::</span><span class="fu">cvh1_ml</span>(intercept_lnRR)</span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a><span class="co"># CVH2 (CVH1) - 1.36 (1.16), 3.76 (1.94), and 12.1 (3.48) for lnRR</span></span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a>M_main_model_vector <span class="ot">&lt;-</span> orchaRd<span class="sc">::</span><span class="fu">m1_ml</span>(intercept_lnRR)</span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a><span class="co"># M2(M1) - 0.58 (0.54), 0.79 (0.66), and 0.78 for lnRR;</span></span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a><span class="co"># # # Total unstandardized raw variance (i.e. total heterogeneity)</span></span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(<span class="fu">sum</span>(intercept_lnRR<span class="sc">$</span>sigma2), <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.022</code></pre>
</div>
</div></div>
</section>
<section id="using-smdh" class="level2">
<h2 class="anchored" data-anchor-id="using-smdh">using SMDH</h2>
<p>We are only running these analysis for the fitness proxy labelled “included”. Some proxies have been flagged and we will run them in a sensitivity analysis only</p>
<p>their sampling variances will be fitted as variance–covariance matrices assuming a 0.5 correlation (ρs) between sampling variances from the same paper_ID (Noble et al.&nbsp;2017).</p>
<p>The MLMA using SMDH as the effect size measure incorporate the imputed sampling variance</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [8]:</pre></div><div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>VCV_SMDH <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> SMDH_variance,</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_SMDH,</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a>intercept_SMDH<span class="ot">&lt;-</span> <span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify SMDH as the effect size measure;</span></span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_SMDH, <span class="co"># specify SMDH's sampling variance-covariance matrix;</span></span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb34-14"><a href="#cb34-14" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb34-15"><a href="#cb34-15" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb34-16"><a href="#cb34-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb34-17"><a href="#cb34-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb34-18"><a href="#cb34-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb34-19"><a href="#cb34-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> dataset_SMDH)</span>
<span id="cb34-20"><a href="#cb34-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-21"><a href="#cb34-21" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the model</span></span>
<span id="cb34-22"><a href="#cb34-22" aria-hidden="true" tabindex="-1"></a><span class="co"># saveRDS(intercept_SMDH, file = here::here("model/intercept_SMDH.rds"))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
</section>
<section id="smdh-heterogeneity" class="level2">
<h2 class="anchored" data-anchor-id="smdh-heterogeneity">SMDH-heterogeneity</h2>
<p>Now looking at the intercept only model</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [9]:</pre></div><div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(intercept_SMDH)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Multivariate Meta-Analysis Model (k = 258; method: REML)

   logLik   Deviance        AIC        BIC       AICc   
-328.9026   657.8052   669.8052   691.0997   670.1412   

Variance Components:

            estim    sqrt  nlvls  fixed          factor 
sigma^2.1  0.0000  0.0010     26     no        paper_ID 
sigma^2.2  0.2543  0.5043    258     no  Observation_ID 
sigma^2.3  0.0000  0.0021     47     no  group_ID_coded 
sigma^2.4  0.0000  0.0042     19     no   population_ID 
sigma^2.5  0.0000  0.0002      7     no    bird_species 

Test for Heterogeneity:
Q(df = 257) = 1124.7869, p-val &lt; .0001

Model Results:

estimate      se    tval   df    pval   ci.lb   ci.ub     
  0.1720  0.0662  2.5979  257  0.0099  0.0416  0.3024  ** 

---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">predict</span>(intercept_SMDH)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
   pred     se  ci.lb  ci.ub   pi.lb  pi.ub 
 0.1720 0.0662 0.0416 0.3024 -0.8296 1.1736 </code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Then also:</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="co"># # Typical sampling variance (which captures the statistical noise of the data, </span></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a><span class="co"># # but is rarely reported in the current meta-analytic practice):</span></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a><span class="fu">sigma2_v</span>(intercept_SMDH)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.1321033</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co"># To calculate heterogeneity, these are the functions from orchaRd:</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a><span class="co"># # I2, CV and M</span></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>sigma2_main_model_vector <span class="ot">&lt;-</span> <span class="fu">sum</span>(intercept_SMDH<span class="sc">$</span>sigma2)</span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a><span class="co"># 0.54, 1.25, and 3.03 for SMD;</span></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>I2_main_model_vector <span class="ot">&lt;-</span> orchaRd<span class="sc">::</span><span class="fu">i2_ml</span>(intercept_SMDH)</span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a><span class="co"># 0.78, 0.89, and 0.96 for SMD</span></span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a><span class="co"># 0.88, 0.95, and 0.99 for lnRR</span></span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a>CV_main_model_vector <span class="ot">&lt;-</span> orchaRd<span class="sc">::</span><span class="fu">cvh1_ml</span>(intercept_SMDH)</span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a><span class="co"># CVH2 (CVH1) - 1.36 (1.16), 3.76 (1.94), and 12.1 (3.48) for lnRR</span></span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a>M_main_model_vector <span class="ot">&lt;-</span> orchaRd<span class="sc">::</span><span class="fu">m1_ml</span>(intercept_SMDH)</span>
<span id="cb41-11"><a href="#cb41-11" aria-hidden="true" tabindex="-1"></a><span class="co"># M2 (M1) - 0.58 (0.54), 0.79 (0.66), and 0.78 for lnRR</span></span>
<span id="cb41-12"><a href="#cb41-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-13"><a href="#cb41-13" aria-hidden="true" tabindex="-1"></a><span class="co"># # # Total unstandardized raw variance (i.e. total heterogeneity)</span></span>
<span id="cb41-14"><a href="#cb41-14" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(<span class="fu">sum</span>(intercept_SMDH<span class="sc">$</span>sigma2), <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.254</code></pre>
</div>
</div></div>
</section>
<section id="bivariate-meta-analytical-model" class="level2">
<h2 class="anchored" data-anchor-id="bivariate-meta-analytical-model">Bivariate Meta-analytical model</h2>
<p>For the overall effect size (i.e., the meta-analytical mean), we will run the bivariate multilevel meta-analytic model suggested by Yang et al.&nbsp;(2024), which models both lnRR and SMD(H) simultaneously (i.e., including them both as response variables and estimating their correlation; random).</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [10]:</pre></div><div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>dataset_bivariate <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">paper_ID =</span> <span class="fu">c</span>(dataset_lnRR<span class="sc">$</span>paper_ID, dataset_SMDH<span class="sc">$</span>paper_ID),</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>                      <span class="at">Observation_ID =</span> <span class="fu">c</span>(dataset_lnRR<span class="sc">$</span>Observation_ID, dataset_SMDH<span class="sc">$</span>Observation_ID),</span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">group_ID_coded =</span> <span class="fu">c</span>(dataset_lnRR<span class="sc">$</span>group_ID_coded, dataset_SMDH<span class="sc">$</span>group_ID_coded),</span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">population_ID =</span> <span class="fu">c</span>(dataset_lnRR<span class="sc">$</span>population_ID, dataset_SMDH<span class="sc">$</span>population_ID),</span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">bird_species =</span> <span class="fu">c</span>(dataset_lnRR<span class="sc">$</span>bird_species, dataset_SMDH<span class="sc">$</span>bird_species),</span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">ES =</span> <span class="fu">c</span>(dataset_lnRR<span class="sc">$</span>lnRR_sign, dataset_SMDH<span class="sc">$</span>SMDH_sign),</span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a>                      <span class="at">V_ES =</span> <span class="fu">c</span>(dataset_lnRR<span class="sc">$</span>lnRR_variance, dataset_SMDH<span class="sc">$</span>SMDH_variance),</span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a>                      <span class="at">ES_measure =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"lnRR"</span>, <span class="fu">nrow</span>(dataset_lnRR)), <span class="fu">rep</span>(<span class="st">"SMDH"</span>, <span class="fu">nrow</span>(dataset_SMDH)))</span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a>                      )</span>
<span id="cb43-10"><a href="#cb43-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-11"><a href="#cb43-11" aria-hidden="true" tabindex="-1"></a>VCV_bivariate <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> V_ES, <span class="co"># sampling variances of lnRR and SMD that are correlated within the same study;</span></span>
<span id="cb43-12"><a href="#cb43-12" aria-hidden="true" tabindex="-1"></a>             <span class="at">cluster =</span> paper_ID, <span class="co"># study identity;</span></span>
<span id="cb43-13"><a href="#cb43-13" aria-hidden="true" tabindex="-1"></a>             <span class="at">type =</span> ES_measure, <span class="co"># different types of effect size measures underlying the observed effect sizes;</span></span>
<span id="cb43-14"><a href="#cb43-14" aria-hidden="true" tabindex="-1"></a>             <span class="at">data =</span> dataset_bivariate, <span class="co"># the long format data frame;</span></span>
<span id="cb43-15"><a href="#cb43-15" aria-hidden="true" tabindex="-1"></a>             <span class="at">rho =</span> <span class="fl">0.2</span>, <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5.</span></span>
<span id="cb43-16"><a href="#cb43-16" aria-hidden="true" tabindex="-1"></a>             <span class="at">nearpd =</span> <span class="cn">TRUE</span> <span class="co"># specify whether impose the non positive definite VCV matrix (not invertible) to the nearest positive semi-definite matrix </span></span>
<span id="cb43-17"><a href="#cb43-17" aria-hidden="true" tabindex="-1"></a>             )</span>
<span id="cb43-18"><a href="#cb43-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-19"><a href="#cb43-19" aria-hidden="true" tabindex="-1"></a>Bivariate_model <span class="ot">&lt;-</span> <span class="fu">rma.mv</span>(<span class="at">yi =</span> ES, <span class="co"># specify the effect size estimate (the variable ES in our case);</span></span>
<span id="cb43-20"><a href="#cb43-20" aria-hidden="true" tabindex="-1"></a>                <span class="at">V =</span> VCV_bivariate, <span class="co"># specify the imputed variance-covariance matrix;</span></span>
<span id="cb43-21"><a href="#cb43-21" aria-hidden="true" tabindex="-1"></a>                <span class="at">mods =</span> <span class="sc">~</span> ES_measure <span class="sc">-</span> <span class="dv">1</span>, <span class="co"># specify the variable "ES_measure" indicating the types of effect size measures;</span></span>
<span id="cb43-22"><a href="#cb43-22" aria-hidden="true" tabindex="-1"></a>                <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> ES_measure <span class="sc">|</span> paper_ID, </span>
<span id="cb43-23"><a href="#cb43-23" aria-hidden="true" tabindex="-1"></a>                              <span class="sc">~</span> ES_measure <span class="sc">|</span> Observation_ID,</span>
<span id="cb43-24"><a href="#cb43-24" aria-hidden="true" tabindex="-1"></a>                              <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb43-25"><a href="#cb43-25" aria-hidden="true" tabindex="-1"></a>                              <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb43-26"><a href="#cb43-26" aria-hidden="true" tabindex="-1"></a>                              <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species</span>
<span id="cb43-27"><a href="#cb43-27" aria-hidden="true" tabindex="-1"></a>                              ), <span class="co"># add correlated random effects corresponding to the lnRR and SMD parameters in the same study;</span></span>
<span id="cb43-28"><a href="#cb43-28" aria-hidden="true" tabindex="-1"></a>                <span class="at">struct =</span> <span class="st">"UN"</span>, <span class="co"># impose an unstructured variance-covariance of the study-specific random effects;</span></span>
<span id="cb43-29"><a href="#cb43-29" aria-hidden="true" tabindex="-1"></a>                <span class="at">test =</span> <span class="st">"t"</span>, <span class="co"># t distribution is specified to test the overall effect against the null hypothesis and construct confidence intervals;</span></span>
<span id="cb43-30"><a href="#cb43-30" aria-hidden="true" tabindex="-1"></a>                <span class="at">method =</span> <span class="st">"REML"</span>, <span class="co"># restricted likelihood maximum is assigned as the estimator for variance components as suggested;</span></span>
<span id="cb43-31"><a href="#cb43-31" aria-hidden="true" tabindex="-1"></a>                <span class="at">data =</span> dataset_bivariate, <span class="co"># the long format dataset</span></span>
<span id="cb43-32"><a href="#cb43-32" aria-hidden="true" tabindex="-1"></a>                <span class="at">sparse =</span> T</span>
<span id="cb43-33"><a href="#cb43-33" aria-hidden="true" tabindex="-1"></a>               )</span>
<span id="cb43-34"><a href="#cb43-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-35"><a href="#cb43-35" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(Bivariate_model, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"sensitivity_analysis/bivariate_model.rds"</span>))</span>
<span id="cb43-36"><a href="#cb43-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-37"><a href="#cb43-37" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(Bivariate_model)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Multivariate Meta-Analysis Model (k = 501; method: REML)

   logLik   Deviance        AIC        BIC       AICc   
-485.9593   971.9186   993.9186  1040.2573   994.4607   

Variance Components:

            estim    sqrt  nlvls  fixed          factor 
sigma^2.1  0.0000  0.0000     47     no  group_ID_coded 
sigma^2.2  0.0000  0.0000     19     no   population_ID 
sigma^2.3  0.0000  0.0000      7     no    bird_species 

outer factor: paper_ID   (nlvls = 26)
inner factor: ES_measure (nlvls = 2)

            estim    sqrt  k.lvl  fixed  level 
tau^2.1    0.0013  0.0359     26     no   lnRR 
tau^2.2    0.1561  0.3951     26     no   SMDH 

      rho.lnRR  rho.SMDH    lnRR  SMDH 
lnRR         1                 -    26 
SMDH   -1.0000         1      no     - 

outer factor: Observation_ID (nlvls = 258)
inner factor: ES_measure     (nlvls = 2)

              estim    sqrt  k.lvl  fixed  level 
gamma^2.1    0.1264  0.3555    243     no   lnRR 
gamma^2.2    1.2527  1.1192    258     no   SMDH 

      phi.lnRR  phi.SMDH    lnRR  SMDH 
lnRR         1    0.5215       -   243 
SMDH    0.5215         1      no     - 

Test for Residual Heterogeneity:
QE(df = 499) = 16205642539.1904, p-val &lt; .0001

Test of Moderators (coefficients 1:2):
F(df1 = 2, df2 = 499) = 1.7279, p-val = 0.1787

Model Results:

                estimate      se     tval   df    pval    ci.lb   ci.ub    
ES_measurelnRR   -0.0132  0.0301  -0.4392  499  0.6607  -0.0724  0.0460    
ES_measureSMDH    0.2404  0.1365   1.7616  499  0.0788  -0.0277  0.5085  . 

---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [11]:</pre></div><div class="cell" data-warning="false">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(ellipse)</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>ab.l <span class="ot">&lt;-</span> <span class="fu">matreg</span>(<span class="at">y=</span><span class="dv">2</span>, <span class="at">x=</span><span class="dv">1</span>, <span class="at">R=</span>Bivariate_model<span class="sc">$</span>G, <span class="at">cov=</span><span class="cn">TRUE</span>, <span class="at">means=</span><span class="fu">coef</span>(Bivariate_model), <span class="at">n=</span>Bivariate_model<span class="sc">$</span>g.levels.comb.k) <span class="co"># fit regression model</span></span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>xy <span class="ot">&lt;-</span> <span class="fu">ellipse</span>(Bivariate_model<span class="sc">$</span>G, <span class="at">centre=</span><span class="fu">coef</span>(Bivariate_model), <span class="at">level=</span><span class="fl">0.95</span>) <span class="co"># get 95% coverage region</span></span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a>ellipse_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">x =</span> xy[, <span class="dv">1</span>], <span class="at">y =</span> xy[, <span class="dv">2</span>]) <span class="co"># convert it int dataframe so that we can use ggplot to make a figure</span></span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a><span class="fu">pivot_wider</span>(dataset_bivariate, <span class="at">names_from =</span> ES_measure, <span class="at">values_from =</span> <span class="fu">c</span>(ES, V_ES)) <span class="sc">%&gt;%</span></span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span> </span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> ES_lnRR, <span class="at">y =</span> ES_SMDH, <span class="at">size =</span> <span class="dv">1</span><span class="sc">/</span><span class="fu">sqrt</span>(V_ES_lnRR)), <span class="at">color =</span> <span class="st">"#1B9E77"</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span> </span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">coef</span>(Bivariate_model)[<span class="dv">1</span>], <span class="at">y =</span> <span class="fu">coef</span>(Bivariate_model)[<span class="dv">2</span>]), <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span> </span>
<span id="cb45-11"><a href="#cb45-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_abline</span>(<span class="at">intercept =</span> ab.l<span class="sc">$</span>tab<span class="sc">$</span>beta[<span class="dv">1</span>], <span class="at">slope =</span> ab.l<span class="sc">$</span>tab<span class="sc">$</span>beta[<span class="dv">2</span>]) <span class="sc">+</span></span>
<span id="cb45-12"><a href="#cb45-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_path</span>(<span class="at">data =</span> ellipse_df, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y), <span class="at">color =</span> <span class="st">"gray50"</span>) <span class="sc">+</span></span>
<span id="cb45-13"><a href="#cb45-13" aria-hidden="true" tabindex="-1"></a>  <span class="co">#scale_x_continuous(limits = c(-2.5, 2.5)) + </span></span>
<span id="cb45-14"><a href="#cb45-14" aria-hidden="true" tabindex="-1"></a>  <span class="co">#scale_y_continuous(limits = c(-2.5, 2.5)) + </span></span>
<span id="cb45-15"><a href="#cb45-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">size =</span> <span class="st">"none"</span>) <span class="sc">+</span> </span>
<span id="cb45-16"><a href="#cb45-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"lnRR"</span>, <span class="at">y =</span> <span class="st">"SMD"</span>) <span class="sc">+</span> </span>
<span id="cb45-17"><a href="#cb45-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in geom_point(aes(x = coef(Bivariate_model)[1], y = coef(Bivariate_model)[2]), : All aesthetics have length 1, but the data has 258 rows.
ℹ Please consider using `annotate()` or provide this layer with data containing
  a single row.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 15 rows containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
</div>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="data_analysis_files/figure-html/unnamed-chunk-6-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-4"><img src="data_analysis_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
</div></div>
<p>There is a very high negative correlation between the two effect sizes. This probably is an artifact of the data and not a true correlation. If you look at the model summary, the <span class="math inline">\(tau^2\)</span> for lnRR, it is extremely small 0.0008, i.e., there is very little variance. This could also be stemming from the fact that the log values in the dataset could be causing this. For SMDH on the other hand, the <span class="math inline">\(tau^2\)</span> is 0.1681 that is much larger. This difference in the variance estimate itself could be causing this artifact of the model. I will only go ahead to present this model as a senstivity analysis because I am not completely convinced by it’s robustness.</p>
</section>
</section>
<section id="sensitivity-analysis" class="level1">
<h1>Sensitivity Analysis</h1>
<section id="sensitivity-analysis-with-flagged-proxies" class="level3">
<h3 class="anchored" data-anchor-id="sensitivity-analysis-with-flagged-proxies">Sensitivity Analysis with flagged proxies</h3>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [12]:</pre></div><div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Preparing new variance-covariance matrix</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>VCV_lnRR_flagged <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> lnRR_variance,</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> lnRR_con_flagged,</span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a><span class="do">## Running intercept only model</span></span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a>SA_lnRR_flagged<span class="ot">&lt;-</span> <span class="fu">rma.mv</span>(<span class="at">yi =</span> lnRR_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb48-14"><a href="#cb48-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_lnRR_flagged, <span class="co"># specify lnRR's sampling variance-covariance matrix;</span></span>
<span id="cb48-15"><a href="#cb48-15" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb48-16"><a href="#cb48-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb48-17"><a href="#cb48-17" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb48-18"><a href="#cb48-18" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb48-19"><a href="#cb48-19" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb48-20"><a href="#cb48-20" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb48-21"><a href="#cb48-21" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb48-22"><a href="#cb48-22" aria-hidden="true" tabindex="-1"></a>                    <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb48-23"><a href="#cb48-23" aria-hidden="true" tabindex="-1"></a>                    <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb48-24"><a href="#cb48-24" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> lnRR_con_flagged)</span>
<span id="cb48-25"><a href="#cb48-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-26"><a href="#cb48-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-27"><a href="#cb48-27" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the model</span></span>
<span id="cb48-28"><a href="#cb48-28" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SA_lnRR_flagged, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"sensitivity_analysis/SA_lnRR_flagged.rds"</span>))</span>
<span id="cb48-29"><a href="#cb48-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-30"><a href="#cb48-30" aria-hidden="true" tabindex="-1"></a><span class="co"># What is the effect now?</span></span>
<span id="cb48-31"><a href="#cb48-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-32"><a href="#cb48-32" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(SA_lnRR_flagged)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Multivariate Meta-Analysis Model (k = 264; method: REML)

  logLik  Deviance       AIC       BIC      AICc   
-20.7461   41.4922   53.4922   74.9252   53.8204   

Variance Components:

            estim    sqrt  nlvls  fixed          factor 
sigma^2.1  0.0000  0.0001     28     no        paper_ID 
sigma^2.2  0.0213  0.1458    264     no  Observation_ID 
sigma^2.3  0.0000  0.0003     52     no  group_ID_coded 
sigma^2.4  0.0000  0.0001     20     no   population_ID 
sigma^2.5  0.0000  0.0000      7     no    bird_species 

Test for Heterogeneity:
Q(df = 263) = 2402.2982, p-val &lt; .0001

Model Results:

estimate      se    tval   df    pval    ci.lb   ci.ub    
  0.0221  0.0143  1.5404  263  0.1247  -0.0061  0.0503    

---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">predict</span>(SA_lnRR_flagged)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
   pred     se   ci.lb  ci.ub   pi.lb  pi.ub 
 0.0221 0.0143 -0.0061 0.0503 -0.2664 0.3105 </code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Now for SMDH </span></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Preparing new variance-covariance matrix</span></span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>VCV_SMDH_flagged <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> SMDH_variance,</span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> SMDH_con_flagged,</span>
<span id="cb52-9"><a href="#cb52-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb52-10"><a href="#cb52-10" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb52-11"><a href="#cb52-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-12"><a href="#cb52-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-13"><a href="#cb52-13" aria-hidden="true" tabindex="-1"></a><span class="do">## Running intercept only model</span></span>
<span id="cb52-14"><a href="#cb52-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-15"><a href="#cb52-15" aria-hidden="true" tabindex="-1"></a>SA_SMDH_flagged<span class="ot">&lt;-</span> <span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb52-16"><a href="#cb52-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_SMDH_flagged, <span class="co"># specify lnRR's sampling variance-covariance matrix;</span></span>
<span id="cb52-17"><a href="#cb52-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb52-18"><a href="#cb52-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb52-19"><a href="#cb52-19" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb52-20"><a href="#cb52-20" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb52-21"><a href="#cb52-21" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb52-22"><a href="#cb52-22" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb52-23"><a href="#cb52-23" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb52-24"><a href="#cb52-24" aria-hidden="true" tabindex="-1"></a>                    <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb52-25"><a href="#cb52-25" aria-hidden="true" tabindex="-1"></a>                    <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb52-26"><a href="#cb52-26" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> SMDH_con_flagged)</span>
<span id="cb52-27"><a href="#cb52-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-28"><a href="#cb52-28" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the model</span></span>
<span id="cb52-29"><a href="#cb52-29" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SA_SMDH_flagged, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"sensitivity_analysis/SA_SMDH_flagged.rds"</span>))</span>
<span id="cb52-30"><a href="#cb52-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-31"><a href="#cb52-31" aria-hidden="true" tabindex="-1"></a><span class="co"># What is the effect now?</span></span>
<span id="cb52-32"><a href="#cb52-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-33"><a href="#cb52-33" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(SA_SMDH_flagged)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Multivariate Meta-Analysis Model (k = 279; method: REML)

   logLik   Deviance        AIC        BIC       AICc   
-348.4133   696.8266   708.8266   730.5924   709.1366   

Variance Components:

            estim    sqrt  nlvls  fixed          factor 
sigma^2.1  0.0000  0.0052     28     no        paper_ID 
sigma^2.2  0.2428  0.4927    279     no  Observation_ID 
sigma^2.3  0.0074  0.0862     53     no  group_ID_coded 
sigma^2.4  0.0000  0.0056     20     no   population_ID 
sigma^2.5  0.0000  0.0032      7     no    bird_species 

Test for Heterogeneity:
Q(df = 278) = 1194.9344, p-val &lt; .0001

Model Results:

estimate      se    tval   df    pval   ci.lb   ci.ub     
  0.1877  0.0650  2.8849  278  0.0042  0.0596  0.3157  ** 

---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">predict</span>(SA_SMDH_flagged)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
   pred     se  ci.lb  ci.ub   pi.lb  pi.ub 
 0.1877 0.0650 0.0596 0.3157 -0.8055 1.1808 </code></pre>
</div>
</div></div>
</section>
<section id="sensitivity-analysis-without-0-es" class="level3">
<h3 class="anchored" data-anchor-id="sensitivity-analysis-without-0-es">Sensitivity Analysis without 0 ES</h3>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [13]:</pre></div><div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Preparing new variance-covariance matrix</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>VCV_lnRR_no_missingES <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> lnRR_variance,</span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> lnRR_sin_missingES,</span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb56-8"><a href="#cb56-8" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb56-9"><a href="#cb56-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-10"><a href="#cb56-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-11"><a href="#cb56-11" aria-hidden="true" tabindex="-1"></a><span class="do">## Running intercept only model</span></span>
<span id="cb56-12"><a href="#cb56-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-13"><a href="#cb56-13" aria-hidden="true" tabindex="-1"></a>SA_lnRR_no_missingES<span class="ot">&lt;-</span> <span class="fu">rma.mv</span>(<span class="at">yi =</span> lnRR_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb56-14"><a href="#cb56-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_lnRR_no_missingES, <span class="co"># specify lnRR's sampling variance-covariance matrix;</span></span>
<span id="cb56-15"><a href="#cb56-15" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb56-16"><a href="#cb56-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb56-17"><a href="#cb56-17" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb56-18"><a href="#cb56-18" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb56-19"><a href="#cb56-19" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb56-20"><a href="#cb56-20" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb56-21"><a href="#cb56-21" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb56-22"><a href="#cb56-22" aria-hidden="true" tabindex="-1"></a>                    <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb56-23"><a href="#cb56-23" aria-hidden="true" tabindex="-1"></a>                    <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb56-24"><a href="#cb56-24" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> lnRR_sin_missingES)</span>
<span id="cb56-25"><a href="#cb56-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-26"><a href="#cb56-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-27"><a href="#cb56-27" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the model</span></span>
<span id="cb56-28"><a href="#cb56-28" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SA_lnRR_no_missingES, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"sensitivity_analysis/SA_lnRR_no_missingES.rds"</span>))</span>
<span id="cb56-29"><a href="#cb56-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-30"><a href="#cb56-30" aria-hidden="true" tabindex="-1"></a><span class="co"># What is the effect now?</span></span>
<span id="cb56-31"><a href="#cb56-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-32"><a href="#cb56-32" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(SA_lnRR_no_missingES)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Multivariate Meta-Analysis Model (k = 227; method: REML)

  logLik  Deviance       AIC       BIC      AICc   
-25.4123   50.8246   62.8246   83.3478   63.2082   

Variance Components:

            estim    sqrt  nlvls  fixed          factor 
sigma^2.1  0.0000  0.0000     26     no        paper_ID 
sigma^2.2  0.0230  0.1518    227     no  Observation_ID 
sigma^2.3  0.0000  0.0003     46     no  group_ID_coded 
sigma^2.4  0.0000  0.0000     19     no   population_ID 
sigma^2.5  0.0000  0.0000      7     no    bird_species 

Test for Heterogeneity:
Q(df = 226) = 2282.3305, p-val &lt; .0001

Model Results:

estimate      se    tval   df    pval    ci.lb   ci.ub    
  0.0176  0.0153  1.1527  226  0.2503  -0.0125  0.0478    

---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="fu">predict</span>(SA_lnRR_no_missingES)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
   pred     se   ci.lb  ci.ub   pi.lb  pi.ub 
 0.0176 0.0153 -0.0125 0.0478 -0.2830 0.3182 </code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Now we do this for SMDH</span></span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Preparing new variance-covariance matrix</span></span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a>VCV_SMDH_no_missingES <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> SMDH_variance,</span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb60-7"><a href="#cb60-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb60-8"><a href="#cb60-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> SMDH_sin_missingES,</span>
<span id="cb60-9"><a href="#cb60-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb60-10"><a href="#cb60-10" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb60-11"><a href="#cb60-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-12"><a href="#cb60-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-13"><a href="#cb60-13" aria-hidden="true" tabindex="-1"></a><span class="do">## Running intercept only model</span></span>
<span id="cb60-14"><a href="#cb60-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-15"><a href="#cb60-15" aria-hidden="true" tabindex="-1"></a>SA_SMDH_no_missingES<span class="ot">&lt;-</span> <span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb60-16"><a href="#cb60-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_SMDH_no_missingES, <span class="co"># specify lnRR's sampling variance-covariance matrix;</span></span>
<span id="cb60-17"><a href="#cb60-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb60-18"><a href="#cb60-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb60-19"><a href="#cb60-19" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb60-20"><a href="#cb60-20" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb60-21"><a href="#cb60-21" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb60-22"><a href="#cb60-22" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb60-23"><a href="#cb60-23" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb60-24"><a href="#cb60-24" aria-hidden="true" tabindex="-1"></a>                    <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb60-25"><a href="#cb60-25" aria-hidden="true" tabindex="-1"></a>                    <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb60-26"><a href="#cb60-26" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> SMDH_sin_missingES)</span>
<span id="cb60-27"><a href="#cb60-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-28"><a href="#cb60-28" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the model</span></span>
<span id="cb60-29"><a href="#cb60-29" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SA_SMDH_no_missingES, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"sensitivity_analysis/SA_SMDH_no_missingES.rds"</span>))</span>
<span id="cb60-30"><a href="#cb60-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-31"><a href="#cb60-31" aria-hidden="true" tabindex="-1"></a><span class="co"># What is the effect now?</span></span>
<span id="cb60-32"><a href="#cb60-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-33"><a href="#cb60-33" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(SA_SMDH_no_missingES)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Multivariate Meta-Analysis Model (k = 242; method: REML)

   logLik   Deviance        AIC        BIC       AICc   
-319.9149   639.8297   651.8297   672.7385   652.1887   

Variance Components:

            estim    sqrt  nlvls  fixed          factor 
sigma^2.1  0.0000  0.0012     26     no        paper_ID 
sigma^2.2  0.2917  0.5401    242     no  Observation_ID 
sigma^2.3  0.0000  0.0019     47     no  group_ID_coded 
sigma^2.4  0.0000  0.0045     19     no   population_ID 
sigma^2.5  0.0000  0.0002      7     no    bird_species 

Test for Heterogeneity:
Q(df = 241) = 1123.5497, p-val &lt; .0001

Model Results:

estimate      se    tval   df    pval   ci.lb   ci.ub    
  0.1781  0.0696  2.5576  241  0.0112  0.0409  0.3152  * 

---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="fu">predict</span>(SA_SMDH_no_missingES)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
   pred     se  ci.lb  ci.ub   pi.lb  pi.ub 
 0.1781 0.0696 0.0409 0.3152 -0.8946 1.2508 </code></pre>
</div>
</div></div>
</section>
<section id="sensitivity-analysis-with-flagged-proxies-and-no-missing-es" class="level3">
<h3 class="anchored" data-anchor-id="sensitivity-analysis-with-flagged-proxies-and-no-missing-es">Sensitivity Analysis with flagged proxies and no missing ES</h3>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [14]:</pre></div><div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Preparing new variance-covariance matrix</span></span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a>VCV_flagged_no_missingES <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> lnRR_variance,</span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb64-5"><a href="#cb64-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb64-6"><a href="#cb64-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> lnRR_flagged_no_missingES,</span>
<span id="cb64-7"><a href="#cb64-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb64-8"><a href="#cb64-8" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb64-9"><a href="#cb64-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-10"><a href="#cb64-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-11"><a href="#cb64-11" aria-hidden="true" tabindex="-1"></a><span class="do">## Running intercept only model</span></span>
<span id="cb64-12"><a href="#cb64-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-13"><a href="#cb64-13" aria-hidden="true" tabindex="-1"></a>SA_lnRR_flagged_no_missingES<span class="ot">&lt;-</span> <span class="fu">rma.mv</span>(<span class="at">yi =</span> lnRR_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb64-14"><a href="#cb64-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_flagged_no_missingES, <span class="co"># specify lnRR's sampling variance-covariance matrix;</span></span>
<span id="cb64-15"><a href="#cb64-15" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb64-16"><a href="#cb64-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb64-17"><a href="#cb64-17" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb64-18"><a href="#cb64-18" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb64-19"><a href="#cb64-19" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb64-20"><a href="#cb64-20" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb64-21"><a href="#cb64-21" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb64-22"><a href="#cb64-22" aria-hidden="true" tabindex="-1"></a>                    <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb64-23"><a href="#cb64-23" aria-hidden="true" tabindex="-1"></a>                    <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb64-24"><a href="#cb64-24" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> lnRR_flagged_no_missingES)</span>
<span id="cb64-25"><a href="#cb64-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-26"><a href="#cb64-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-27"><a href="#cb64-27" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the model</span></span>
<span id="cb64-28"><a href="#cb64-28" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SA_lnRR_flagged_no_missingES, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"sensitivity_analysis/SA_lnRR_flagged_no_missingES.rds"</span>))</span>
<span id="cb64-29"><a href="#cb64-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-30"><a href="#cb64-30" aria-hidden="true" tabindex="-1"></a><span class="co"># What is the effect now?</span></span>
<span id="cb64-31"><a href="#cb64-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-32"><a href="#cb64-32" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(SA_lnRR_flagged_no_missingES)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Multivariate Meta-Analysis Model (k = 248; method: REML)

  logLik  Deviance       AIC       BIC      AICc   
-24.5958   49.1917   61.1917   82.2480   61.5417   

Variance Components:

            estim    sqrt  nlvls  fixed          factor 
sigma^2.1  0.0000  0.0001     28     no        paper_ID 
sigma^2.2  0.0224  0.1497    248     no  Observation_ID 
sigma^2.3  0.0000  0.0003     52     no  group_ID_coded 
sigma^2.4  0.0000  0.0000     20     no   population_ID 
sigma^2.5  0.0000  0.0000      7     no    bird_species 

Test for Heterogeneity:
Q(df = 247) = 2399.8889, p-val &lt; .0001

Model Results:

estimate      se    tval   df    pval    ci.lb   ci.ub    
  0.0222  0.0147  1.5164  247  0.1307  -0.0066  0.0511    

---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="fu">predict</span>(SA_lnRR_flagged_no_missingES)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
   pred     se   ci.lb  ci.ub   pi.lb  pi.ub 
 0.0222 0.0147 -0.0066 0.0511 -0.2740 0.3185 </code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="do">## with SMDH</span></span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Preparing new variance-covariance matrix</span></span>
<span id="cb68-4"><a href="#cb68-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-5"><a href="#cb68-5" aria-hidden="true" tabindex="-1"></a>VCV_SMDH_flagged_no_missingES <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> SMDH_variance,</span>
<span id="cb68-6"><a href="#cb68-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb68-7"><a href="#cb68-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb68-8"><a href="#cb68-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> SMDH_flagged_no_missingES,</span>
<span id="cb68-9"><a href="#cb68-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb68-10"><a href="#cb68-10" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb68-11"><a href="#cb68-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-12"><a href="#cb68-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-13"><a href="#cb68-13" aria-hidden="true" tabindex="-1"></a><span class="do">## Running intercept only model</span></span>
<span id="cb68-14"><a href="#cb68-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-15"><a href="#cb68-15" aria-hidden="true" tabindex="-1"></a>SA_SMDH_flagged_no_missingES<span class="ot">&lt;-</span> <span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb68-16"><a href="#cb68-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_SMDH_flagged_no_missingES, <span class="co"># specify lnRR's sampling variance-covariance matrix;</span></span>
<span id="cb68-17"><a href="#cb68-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb68-18"><a href="#cb68-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb68-19"><a href="#cb68-19" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb68-20"><a href="#cb68-20" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb68-21"><a href="#cb68-21" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb68-22"><a href="#cb68-22" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb68-23"><a href="#cb68-23" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb68-24"><a href="#cb68-24" aria-hidden="true" tabindex="-1"></a>                    <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb68-25"><a href="#cb68-25" aria-hidden="true" tabindex="-1"></a>                    <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb68-26"><a href="#cb68-26" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> SMDH_flagged_no_missingES)</span>
<span id="cb68-27"><a href="#cb68-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-28"><a href="#cb68-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-29"><a href="#cb68-29" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the model</span></span>
<span id="cb68-30"><a href="#cb68-30" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SA_SMDH_flagged_no_missingES, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"sensitivity_analysis/SA_SMDH_flagged_no_missingES.rds"</span>))</span>
<span id="cb68-31"><a href="#cb68-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-32"><a href="#cb68-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-33"><a href="#cb68-33" aria-hidden="true" tabindex="-1"></a><span class="co"># What is the effect now?</span></span>
<span id="cb68-34"><a href="#cb68-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-35"><a href="#cb68-35" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(SA_SMDH_flagged_no_missingES)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Multivariate Meta-Analysis Model (k = 263; method: REML)

   logLik   Deviance        AIC        BIC       AICc   
-339.6297   679.2593   691.2593   712.6694   691.5887   

Variance Components:

            estim    sqrt  nlvls  fixed          factor 
sigma^2.1  0.0000  0.0002     28     no        paper_ID 
sigma^2.2  0.2772  0.5265    263     no  Observation_ID 
sigma^2.3  0.0001  0.0085     53     no  group_ID_coded 
sigma^2.4  0.0000  0.0002     20     no   population_ID 
sigma^2.5  0.0000  0.0000      7     no    bird_species 

Test for Heterogeneity:
Q(df = 262) = 1193.7544, p-val &lt; .0001

Model Results:

estimate      se    tval   df    pval   ci.lb   ci.ub     
  0.1914  0.0657  2.9124  262  0.0039  0.0620  0.3208  ** 

---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="fu">predict</span>(SA_SMDH_flagged_no_missingES)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
   pred     se  ci.lb  ci.ub   pi.lb  pi.ub 
 0.1914 0.0657 0.0620 0.3208 -0.8535 1.2363 </code></pre>
</div>
</div></div>
</section>
<section id="sensitivity-analysis-only-using-effect-sizes-calculated-from-means-sds-and-ns" class="level3">
<h3 class="anchored" data-anchor-id="sensitivity-analysis-only-using-effect-sizes-calculated-from-means-sds-and-ns">Sensitivity analysis only using effect sizes calculated from means, SDs, and ns</h3>
<p>(i.e., excluding any effect sizes calculated from inferential statistics)</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [15]:</pre></div><div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Preparing new variance-covariance matrix</span></span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a>VCV_SMDH_no_inferential <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> SMDH_variance,</span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb72-5"><a href="#cb72-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb72-6"><a href="#cb72-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> SMDH_no_inferential,</span>
<span id="cb72-7"><a href="#cb72-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb72-8"><a href="#cb72-8" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb72-9"><a href="#cb72-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-10"><a href="#cb72-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-11"><a href="#cb72-11" aria-hidden="true" tabindex="-1"></a><span class="do">## Running intercept only model</span></span>
<span id="cb72-12"><a href="#cb72-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-13"><a href="#cb72-13" aria-hidden="true" tabindex="-1"></a>SA_SMDH_no_inferential<span class="ot">&lt;-</span> <span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb72-14"><a href="#cb72-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_SMDH_no_inferential, <span class="co"># specify lnRR's sampling variance-covariance matrix;</span></span>
<span id="cb72-15"><a href="#cb72-15" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb72-16"><a href="#cb72-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb72-17"><a href="#cb72-17" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb72-18"><a href="#cb72-18" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb72-19"><a href="#cb72-19" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb72-20"><a href="#cb72-20" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb72-21"><a href="#cb72-21" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb72-22"><a href="#cb72-22" aria-hidden="true" tabindex="-1"></a>                    <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb72-23"><a href="#cb72-23" aria-hidden="true" tabindex="-1"></a>                    <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb72-24"><a href="#cb72-24" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> SMDH_no_inferential)</span>
<span id="cb72-25"><a href="#cb72-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-26"><a href="#cb72-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-27"><a href="#cb72-27" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the model</span></span>
<span id="cb72-28"><a href="#cb72-28" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SA_SMDH_no_inferential, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"sensitivity_analysis/SA_SMDH_no_inferential.rds"</span>))</span>
<span id="cb72-29"><a href="#cb72-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-30"><a href="#cb72-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-31"><a href="#cb72-31" aria-hidden="true" tabindex="-1"></a><span class="co"># What is the effect now?</span></span>
<span id="cb72-32"><a href="#cb72-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-33"><a href="#cb72-33" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(SA_SMDH_no_inferential)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Multivariate Meta-Analysis Model (k = 250; method: REML)

   logLik   Deviance        AIC        BIC       AICc   
-322.7054   645.4108   657.4108   678.5155   657.7579   

Variance Components:

            estim    sqrt  nlvls  fixed          factor 
sigma^2.1  0.0000  0.0048     26     no        paper_ID 
sigma^2.2  0.2623  0.5122    250     no  Observation_ID 
sigma^2.3  0.0001  0.0088     46     no  group_ID_coded 
sigma^2.4  0.0004  0.0204     19     no   population_ID 
sigma^2.5  0.0000  0.0013      7     no    bird_species 

Test for Heterogeneity:
Q(df = 249) = 1110.1810, p-val &lt; .0001

Model Results:

estimate      se    tval   df    pval   ci.lb   ci.ub    
  0.1721  0.0680  2.5298  249  0.0120  0.0381  0.3061  * 

---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">predict</span>(SA_SMDH_no_inferential)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
   pred     se  ci.lb  ci.ub   pi.lb  pi.ub 
 0.1721 0.0680 0.0381 0.3061 -0.8465 1.1907 </code></pre>
</div>
</div></div>
</section>
<section id="different-values-of-sampling-variance-correlation" class="level3">
<h3 class="anchored" data-anchor-id="different-values-of-sampling-variance-correlation">Different values of sampling variance correlation</h3>
<p><u><strong>Robust variance estimation to guard against arbitary value of sampling correlation</strong></u> - ‘robust()’ function from R package ‘metafor’ (Viechtbauer 2010) using ‘CR2’ (bias reduced linearization correction) from the R package ‘clubSandwich’ (Pustejovsky 2024) for small-sample size adjustment.</p>
<p>At the moment, this is not working..</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [16]:</pre></div><div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="co">#MLMA_lnRR_RVE&lt;- robust(MLMA_lnRR_VCV,</span></span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a><span class="co">#                        cluster = paper_ID,</span></span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a><span class="co">#                        clubSandwich = TRUE,</span></span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a><span class="co">#                        verbose=T)</span></span>
<span id="cb76-5"><a href="#cb76-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-6"><a href="#cb76-6" aria-hidden="true" tabindex="-1"></a><span class="co"># # summary(MLMA_lnRR_RVE)</span></span>
<span id="cb76-7"><a href="#cb76-7" aria-hidden="true" tabindex="-1"></a><span class="co"># I2_CI &lt;- i2_ml(MLMA_lnRR_RVE) # Relative heterogeneity</span></span>
<span id="cb76-8"><a href="#cb76-8" aria-hidden="true" tabindex="-1"></a><span class="co"># round(I2_CI,2) # absolute heterogeneity is reported in the summary of the model</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Important
</div>
</div>
<div class="callout-body-container callout-body">
<p>Sensitivity Analysis</p>
<p>We will also perform a sensitivity analysis using 0.3 and 0.7 correlation between the sampling variances from the same paper_ID and report it in the supplementary materials. If this sensitivity analysis shows large differences in conclusions for the intercept-only model, we will proceed to perform such sensitivity analysis for the meta-regressions.</p>
</div>
</div>
<p>First, set a series of ρs (i.e., 0.3, 0.5, 0.7) (we assume these values arbitrarily):</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [17]:</pre></div><div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="co"># rho_range &lt;- c(0.3, 0.5, 0.7)</span></span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb77-3"><a href="#cb77-3" aria-hidden="true" tabindex="-1"></a><span class="co"># MLMAlnRR_VCV_range &lt;- list() # repeatedly run the specified model with varying rho</span></span>
<span id="cb77-4"><a href="#cb77-4" aria-hidden="true" tabindex="-1"></a><span class="co"># for (i in 1:length(rho_range))</span></span>
<span id="cb77-5"><a href="#cb77-5" aria-hidden="true" tabindex="-1"></a><span class="co"># {VCV_range &lt;- vcalc(vi = lnRR_variance,</span></span>
<span id="cb77-6"><a href="#cb77-6" aria-hidden="true" tabindex="-1"></a><span class="co">#                   cluster = paper_ID,</span></span>
<span id="cb77-7"><a href="#cb77-7" aria-hidden="true" tabindex="-1"></a><span class="co">#                   subgroup = group_ID,</span></span>
<span id="cb77-8"><a href="#cb77-8" aria-hidden="true" tabindex="-1"></a><span class="co">#                   obs = Observation_ID,</span></span>
<span id="cb77-9"><a href="#cb77-9" aria-hidden="true" tabindex="-1"></a><span class="co">#                   data = dataset_lnRR,</span></span>
<span id="cb77-10"><a href="#cb77-10" aria-hidden="true" tabindex="-1"></a><span class="co">#                    rho = rho_range[i],</span></span>
<span id="cb77-11"><a href="#cb77-11" aria-hidden="true" tabindex="-1"></a><span class="co">#                     ) # impute sampling variance covariance matrix with varying rho</span></span>
<span id="cb77-12"><a href="#cb77-12" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb77-13"><a href="#cb77-13" aria-hidden="true" tabindex="-1"></a><span class="co"># MLMAlnRR_VCV_range[[i]] &lt;- rma.mv(yi = lnRR_sign,</span></span>
<span id="cb77-14"><a href="#cb77-14" aria-hidden="true" tabindex="-1"></a><span class="co">#                                   V = VCV_range, # sampling variance covariance matrix with varying values of rho.</span></span>
<span id="cb77-15"><a href="#cb77-15" aria-hidden="true" tabindex="-1"></a><span class="co">#                                  random = list(~ 1 | paper_ID,</span></span>
<span id="cb77-16"><a href="#cb77-16" aria-hidden="true" tabindex="-1"></a><span class="co">#                                   ~ 1 | Observation_ID,</span></span>
<span id="cb77-17"><a href="#cb77-17" aria-hidden="true" tabindex="-1"></a><span class="co">#                                   ~ 1 | experiment_ID,</span></span>
<span id="cb77-18"><a href="#cb77-18" aria-hidden="true" tabindex="-1"></a><span class="co">#                                   ~ 1 | group_ID,</span></span>
<span id="cb77-19"><a href="#cb77-19" aria-hidden="true" tabindex="-1"></a><span class="co">#                                   ~ 1 | repeated_trait_ID),</span></span>
<span id="cb77-20"><a href="#cb77-20" aria-hidden="true" tabindex="-1"></a><span class="co">#                                   method = "REML",</span></span>
<span id="cb77-21"><a href="#cb77-21" aria-hidden="true" tabindex="-1"></a><span class="co">#                                   data = dataset_lnRR # run the model with varying rho</span></span>
<span id="cb77-22"><a href="#cb77-22" aria-hidden="true" tabindex="-1"></a><span class="co"># )</span></span>
<span id="cb77-23"><a href="#cb77-23" aria-hidden="true" tabindex="-1"></a><span class="co"># }</span></span>
<span id="cb77-24"><a href="#cb77-24" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb77-25"><a href="#cb77-25" aria-hidden="true" tabindex="-1"></a><span class="co"># save(VCV_range, file = here::here("GNM/figures/VCV_range.Rdata"))</span></span>
<span id="cb77-26"><a href="#cb77-26" aria-hidden="true" tabindex="-1"></a><span class="co"># save(MLMAlnRR_VCV_range, file = here::here("GNM/figures/MLMAlnRR_VCV_range.Rdata"))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [18]:</pre></div><div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="co"># t4 &lt;- data.frame(rho  = rho_range,</span></span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a><span class="co">#                  "overall effect"  = sapply(MLMAlnRR_VCV_range, function(x) coef(x)),</span></span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a><span class="co">#                  "standard error" = sapply(MLMAlnRR_VCV_range, function(x) x$se),</span></span>
<span id="cb78-4"><a href="#cb78-4" aria-hidden="true" tabindex="-1"></a><span class="co">#                  "p-value" = sapply(MLMAlnRR_VCV_range, function(x) x$pval),</span></span>
<span id="cb78-5"><a href="#cb78-5" aria-hidden="true" tabindex="-1"></a><span class="co">#                  "Lower CI" = sapply(MLMAlnRR_VCV_range, function(x) x$ci.lb),</span></span>
<span id="cb78-6"><a href="#cb78-6" aria-hidden="true" tabindex="-1"></a><span class="co">#                  "Upper CI" = sapply(MLMAlnRR_VCV_range, function(x) x$ci.ub),</span></span>
<span id="cb78-7"><a href="#cb78-7" aria-hidden="true" tabindex="-1"></a><span class="co">#                  "Log-likehood" = sapply(MLMAlnRR_VCV_range, function(x) fitstats(x)[1,1]),</span></span>
<span id="cb78-8"><a href="#cb78-8" aria-hidden="true" tabindex="-1"></a><span class="co">#                  "AIC" = sapply(MLMAlnRR_VCV_range, function(x) fitstats(x)[3,1]),</span></span>
<span id="cb78-9"><a href="#cb78-9" aria-hidden="true" tabindex="-1"></a><span class="co">#                  "BIC" = sapply(MLMAlnRR_VCV_range, function(x) fitstats(x)[4,1]),</span></span>
<span id="cb78-10"><a href="#cb78-10" aria-hidden="true" tabindex="-1"></a><span class="co">#                  "AICc" = sapply(MLMAlnRR_VCV_range, function(x) fitstats(x)[5,1]))</span></span>
<span id="cb78-11"><a href="#cb78-11" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb78-12"><a href="#cb78-12" aria-hidden="true" tabindex="-1"></a><span class="co"># colnames(t4) &lt;- c("Sampling correlation", "Overall effect (lnRR)", "Standard error", "p-value", "Lower CI", "Upper CI", "Log-likehood", "AIC", "BIC", "AICc")</span></span>
<span id="cb78-13"><a href="#cb78-13" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb78-14"><a href="#cb78-14" aria-hidden="true" tabindex="-1"></a><span class="co"># t4 %&gt;% dfround(4) %&gt;% DT::datatable()</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
</section>
</section>
<section id="mechanistic-hypothesis" class="level1">
<h1>Mechanistic Hypothesis</h1>
<p>To address our secondary questions “Does the addition of green nest material by birds increase their reproductive success? (Courtship hypothesis)” and “Does the addition of green nest material have protective effects and health benefits for the nestlings? (Parental care hypothesis)”, we will make use of the unimoderator multilevel meta-regressions hypothesis type (levels: CH, PCH, Both) as the moderator.</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [19]:</pre></div><div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a>lnRR_hypothesis <span class="ot">&lt;-</span></span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> lnRR_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_lnRR,<span class="co"># specify lnRR's sampling variance;</span></span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="sc">-</span> <span class="dv">1</span> <span class="sc">+</span> Hypothesis,</span>
<span id="cb79-5"><a href="#cb79-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb79-6"><a href="#cb79-6" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb79-7"><a href="#cb79-7" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb79-8"><a href="#cb79-8" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb79-9"><a href="#cb79-9" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb79-10"><a href="#cb79-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb79-11"><a href="#cb79-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb79-12"><a href="#cb79-12" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb79-13"><a href="#cb79-13" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_lnRR)</span>
<span id="cb79-14"><a href="#cb79-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-15"><a href="#cb79-15" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(lnRR_hypothesis, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/lnRR_hypothesis.rds"</span>))</span>
<span id="cb79-16"><a href="#cb79-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-17"><a href="#cb79-17" aria-hidden="true" tabindex="-1"></a>lnRR_hypothesis_R2 <span class="ot">&lt;-</span> <span class="fu">r2_ml</span>(lnRR_hypothesis)</span>
<span id="cb79-18"><a href="#cb79-18" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(lnRR_hypothesis_R2 <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>   R2_marginal R2_conditional 
           1.7            1.7 </code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a>SMDH_hypothesis <span class="ot">&lt;-</span></span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb81-3"><a href="#cb81-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_SMDH,<span class="co"># specify lnRR's sampling variance;</span></span>
<span id="cb81-4"><a href="#cb81-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="sc">-</span> <span class="dv">1</span> <span class="sc">+</span> Hypothesis,</span>
<span id="cb81-5"><a href="#cb81-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb81-6"><a href="#cb81-6" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb81-7"><a href="#cb81-7" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb81-8"><a href="#cb81-8" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb81-9"><a href="#cb81-9" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb81-10"><a href="#cb81-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb81-11"><a href="#cb81-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb81-12"><a href="#cb81-12" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb81-13"><a href="#cb81-13" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_SMDH)</span>
<span id="cb81-14"><a href="#cb81-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-15"><a href="#cb81-15" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SMDH_hypothesis, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/SMDH_hypothesis.rds"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
</section>
<section id="exploratory-analysis" class="level1">
<h1>Exploratory Analysis</h1>
<p>For our exploratory analyses (see section Exploratory Analyses), we will run unimoderator multilevel meta-regressions using the following moderators:</p>
<section id="type-of-parasites" class="level4">
<h4 class="anchored" data-anchor-id="type-of-parasites">Type of parasites</h4>
<p>(levels: arthropods, micro-organisms)&nbsp;</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [20]:</pre></div><div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="co"># This analysis only includes dataset that has a proxy relating to parasite or pathogen</span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>dataset_lnRR_parasite <span class="ot">&lt;-</span> dataset_lnRR<span class="sc">%&gt;%</span></span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(parasite_type))</span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-5"><a href="#cb82-5" aria-hidden="true" tabindex="-1"></a>VCV_lnRR_parasite <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> lnRR_variance,</span>
<span id="cb82-6"><a href="#cb82-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb82-7"><a href="#cb82-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb82-8"><a href="#cb82-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_lnRR_parasite,</span>
<span id="cb82-9"><a href="#cb82-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb82-10"><a href="#cb82-10" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb82-11"><a href="#cb82-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-12"><a href="#cb82-12" aria-hidden="true" tabindex="-1"></a>lnRR_parasite <span class="ot">&lt;-</span></span>
<span id="cb82-13"><a href="#cb82-13" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> lnRR_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb82-14"><a href="#cb82-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_lnRR_parasite,<span class="co"># specify lnRR's sampling variance;</span></span>
<span id="cb82-15"><a href="#cb82-15" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="sc">-</span> <span class="dv">1</span> <span class="sc">+</span> parasite_type,</span>
<span id="cb82-16"><a href="#cb82-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb82-17"><a href="#cb82-17" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb82-18"><a href="#cb82-18" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb82-19"><a href="#cb82-19" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb82-20"><a href="#cb82-20" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb82-21"><a href="#cb82-21" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb82-22"><a href="#cb82-22" aria-hidden="true" tabindex="-1"></a>                   <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb82-23"><a href="#cb82-23" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb82-24"><a href="#cb82-24" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_lnRR_parasite)</span>
<span id="cb82-25"><a href="#cb82-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-26"><a href="#cb82-26" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>( lnRR_parasite, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/lnRR_parasite.rds"</span>))</span>
<span id="cb82-27"><a href="#cb82-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-28"><a href="#cb82-28" aria-hidden="true" tabindex="-1"></a><span class="do">### Similiarly for SMDH as well selecting the dataset with parasite type</span></span>
<span id="cb82-29"><a href="#cb82-29" aria-hidden="true" tabindex="-1"></a><span class="co"># This analysis only includes dataset that has a proxy relating to parasite or pathogen</span></span>
<span id="cb82-30"><a href="#cb82-30" aria-hidden="true" tabindex="-1"></a>dataset_SMDH_parasite <span class="ot">&lt;-</span> dataset_SMDH<span class="sc">%&gt;%</span></span>
<span id="cb82-31"><a href="#cb82-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(parasite_type))</span>
<span id="cb82-32"><a href="#cb82-32" aria-hidden="true" tabindex="-1"></a>VCV_SMDH_parasite <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> SMDH_variance,</span>
<span id="cb82-33"><a href="#cb82-33" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb82-34"><a href="#cb82-34" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb82-35"><a href="#cb82-35" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_SMDH_parasite,</span>
<span id="cb82-36"><a href="#cb82-36" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb82-37"><a href="#cb82-37" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb82-38"><a href="#cb82-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-39"><a href="#cb82-39" aria-hidden="true" tabindex="-1"></a>SMDH_parasite <span class="ot">&lt;-</span></span>
<span id="cb82-40"><a href="#cb82-40" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb82-41"><a href="#cb82-41" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_SMDH_parasite,<span class="co"># specify lnRR's sampling variance;</span></span>
<span id="cb82-42"><a href="#cb82-42" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="sc">-</span> <span class="dv">1</span> <span class="sc">+</span> parasite_type,</span>
<span id="cb82-43"><a href="#cb82-43" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb82-44"><a href="#cb82-44" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb82-45"><a href="#cb82-45" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb82-46"><a href="#cb82-46" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb82-47"><a href="#cb82-47" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb82-48"><a href="#cb82-48" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb82-49"><a href="#cb82-49" aria-hidden="true" tabindex="-1"></a>                   <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb82-50"><a href="#cb82-50" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb82-51"><a href="#cb82-51" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_SMDH_parasite)</span>
<span id="cb82-52"><a href="#cb82-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-53"><a href="#cb82-53" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb82-54"><a href="#cb82-54" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SMDH_parasite, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/SMDH_parasite.rds"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
</section>
<section id="time-of-addition-of-green-nest-material" class="level4">
<h4 class="anchored" data-anchor-id="time-of-addition-of-green-nest-material">Time of addition of green nest material</h4>
<p>(levels: before egg hatching, after egg hatching, continuously throughout the nesting phase).</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [21]:</pre></div><div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="do">## We use the complete dataset in this case</span></span>
<span id="cb83-2"><a href="#cb83-2" aria-hidden="true" tabindex="-1"></a>lnRR_time_gnm_addition <span class="ot">&lt;-</span></span>
<span id="cb83-3"><a href="#cb83-3" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> lnRR_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb83-4"><a href="#cb83-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_lnRR,<span class="co"># specify lnRR's sampling variance using the variance-covariance matrix we created for the intercept-only model since the whole dataset is being used in this case as well</span></span>
<span id="cb83-5"><a href="#cb83-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="sc">-</span> <span class="dv">1</span> <span class="sc">+</span> time_of_gnm_addition,</span>
<span id="cb83-6"><a href="#cb83-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb83-7"><a href="#cb83-7" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb83-8"><a href="#cb83-8" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb83-9"><a href="#cb83-9" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb83-10"><a href="#cb83-10" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb83-11"><a href="#cb83-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb83-12"><a href="#cb83-12" aria-hidden="true" tabindex="-1"></a>                  <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb83-13"><a href="#cb83-13" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb83-14"><a href="#cb83-14" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_lnRR)</span>
<span id="cb83-15"><a href="#cb83-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-16"><a href="#cb83-16" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(lnRR_time_gnm_addition, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/lnRR_time_gnm_addition.rds"</span>))</span>
<span id="cb83-17"><a href="#cb83-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-18"><a href="#cb83-18" aria-hidden="true" tabindex="-1"></a><span class="do">## Now similarly for the SMDH effect size using the complete dataset</span></span>
<span id="cb83-19"><a href="#cb83-19" aria-hidden="true" tabindex="-1"></a>SMDH_time_gnm_addition <span class="ot">&lt;-</span></span>
<span id="cb83-20"><a href="#cb83-20" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb83-21"><a href="#cb83-21" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_SMDH,<span class="co"># specify lnRR's sampling variance using the variance-covariance matrix we created for the intercept-only model since the whole dataset is being used in this case as well</span></span>
<span id="cb83-22"><a href="#cb83-22" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="sc">-</span> <span class="dv">1</span> <span class="sc">+</span> time_of_gnm_addition,</span>
<span id="cb83-23"><a href="#cb83-23" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb83-24"><a href="#cb83-24" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb83-25"><a href="#cb83-25" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb83-26"><a href="#cb83-26" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb83-27"><a href="#cb83-27" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb83-28"><a href="#cb83-28" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb83-29"><a href="#cb83-29" aria-hidden="true" tabindex="-1"></a>                  <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb83-30"><a href="#cb83-30" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb83-31"><a href="#cb83-31" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_SMDH)</span>
<span id="cb83-32"><a href="#cb83-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-33"><a href="#cb83-33" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SMDH_time_gnm_addition, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/SMDH_time_gnm_addition.rds"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
</section>
<section id="type-of-experimental-design" class="level4">
<h4 class="anchored" data-anchor-id="type-of-experimental-design">Type of experimental design</h4>
<p>(levels: 1 = non-aromatic vs.&nbsp;aromatic, 2 = no added material vs.&nbsp;aromatic, 3 = no added material vs.&nbsp;non-aromatic).</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [22]:</pre></div><div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="co"># This analysis only includes dataset that has a proxy relating to comaparision type</span></span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a>dataset_lnRR_design <span class="ot">&lt;-</span> dataset_lnRR <span class="sc">%&gt;%</span></span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(comparision_type))<span class="sc">%&gt;%</span></span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">comparision_type=</span> <span class="fu">as.factor</span>(comparision_type))</span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a>VCV_lnRR_design <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> lnRR_variance,</span>
<span id="cb84-7"><a href="#cb84-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb84-8"><a href="#cb84-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb84-9"><a href="#cb84-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_lnRR_design,</span>
<span id="cb84-10"><a href="#cb84-10" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb84-11"><a href="#cb84-11" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb84-12"><a href="#cb84-12" aria-hidden="true" tabindex="-1"></a>lnRR_design <span class="ot">&lt;-</span></span>
<span id="cb84-13"><a href="#cb84-13" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> lnRR_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb84-14"><a href="#cb84-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_lnRR_design ,<span class="co"># specify lnRR's sampling variance;</span></span>
<span id="cb84-15"><a href="#cb84-15" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="sc">-</span> <span class="dv">1</span> <span class="sc">+</span> comparision_type,</span>
<span id="cb84-16"><a href="#cb84-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb84-17"><a href="#cb84-17" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb84-18"><a href="#cb84-18" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb84-19"><a href="#cb84-19" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb84-20"><a href="#cb84-20" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb84-21"><a href="#cb84-21" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb84-22"><a href="#cb84-22" aria-hidden="true" tabindex="-1"></a>                  <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb84-23"><a href="#cb84-23" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb84-24"><a href="#cb84-24" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_lnRR_design )</span>
<span id="cb84-25"><a href="#cb84-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-26"><a href="#cb84-26" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(lnRR_design, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/lnRR_design.rds"</span>))</span>
<span id="cb84-27"><a href="#cb84-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-28"><a href="#cb84-28" aria-hidden="true" tabindex="-1"></a><span class="co"># This analysis only includes dataset that has a proxy relating to comaparision type</span></span>
<span id="cb84-29"><a href="#cb84-29" aria-hidden="true" tabindex="-1"></a>dataset_SMDH_design <span class="ot">&lt;-</span> dataset_SMDH <span class="sc">%&gt;%</span></span>
<span id="cb84-30"><a href="#cb84-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(comparision_type))<span class="sc">%&gt;%</span></span>
<span id="cb84-31"><a href="#cb84-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">comparision_type=</span> <span class="fu">as.factor</span>(comparision_type))</span>
<span id="cb84-32"><a href="#cb84-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-33"><a href="#cb84-33" aria-hidden="true" tabindex="-1"></a>VCV_SMDH_design <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> SMDH_variance,</span>
<span id="cb84-34"><a href="#cb84-34" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb84-35"><a href="#cb84-35" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb84-36"><a href="#cb84-36" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_SMDH_design,</span>
<span id="cb84-37"><a href="#cb84-37" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb84-38"><a href="#cb84-38" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb84-39"><a href="#cb84-39" aria-hidden="true" tabindex="-1"></a>SMDH_design <span class="ot">&lt;-</span></span>
<span id="cb84-40"><a href="#cb84-40" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb84-41"><a href="#cb84-41" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_SMDH_design ,<span class="co"># specify lnRR's sampling variance;</span></span>
<span id="cb84-42"><a href="#cb84-42" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="sc">-</span> <span class="dv">1</span> <span class="sc">+</span> comparision_type,</span>
<span id="cb84-43"><a href="#cb84-43" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb84-44"><a href="#cb84-44" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb84-45"><a href="#cb84-45" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb84-46"><a href="#cb84-46" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb84-47"><a href="#cb84-47" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb84-48"><a href="#cb84-48" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb84-49"><a href="#cb84-49" aria-hidden="true" tabindex="-1"></a>                  <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb84-50"><a href="#cb84-50" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb84-51"><a href="#cb84-51" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_SMDH_design )</span>
<span id="cb84-52"><a href="#cb84-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-53"><a href="#cb84-53" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SMDH_design, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/SMDH_design.rds"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<p>Let me try to do a sensitivity analysis to see what happens when I remove the control condition of 3 in the overall model since that is expected to not have an effect according to the authors as well.. i.e.&nbsp;it is a double control..</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [23]:</pre></div><div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a>dataset_lnRR_remove3<span class="ot">&lt;-</span>dataset_lnRR<span class="sc">%&gt;%</span><span class="fu">filter</span>(<span class="fu">is.na</span>(comparision_type) <span class="sc">|</span> comparision_type<span class="sc">!=</span><span class="st">"3"</span>)</span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-3"><a href="#cb85-3" aria-hidden="true" tabindex="-1"></a>VCV_lnRR_remove3 <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> lnRR_variance,</span>
<span id="cb85-4"><a href="#cb85-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb85-5"><a href="#cb85-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb85-6"><a href="#cb85-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_lnRR_remove3,</span>
<span id="cb85-7"><a href="#cb85-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb85-8"><a href="#cb85-8" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb85-9"><a href="#cb85-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-10"><a href="#cb85-10" aria-hidden="true" tabindex="-1"></a>lnRR_remove3 <span class="ot">&lt;-</span></span>
<span id="cb85-11"><a href="#cb85-11" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> lnRR_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb85-12"><a href="#cb85-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_lnRR_remove3 ,<span class="co"># specify lnRR's sampling variance;</span></span>
<span id="cb85-13"><a href="#cb85-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb85-14"><a href="#cb85-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb85-15"><a href="#cb85-15" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb85-16"><a href="#cb85-16" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb85-17"><a href="#cb85-17" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb85-18"><a href="#cb85-18" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb85-19"><a href="#cb85-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb85-20"><a href="#cb85-20" aria-hidden="true" tabindex="-1"></a>                  <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb85-21"><a href="#cb85-21" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb85-22"><a href="#cb85-22" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_lnRR_remove3 )</span>
<span id="cb85-23"><a href="#cb85-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-24"><a href="#cb85-24" aria-hidden="true" tabindex="-1"></a><span class="do">## Now for SMDH</span></span>
<span id="cb85-25"><a href="#cb85-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-26"><a href="#cb85-26" aria-hidden="true" tabindex="-1"></a>dataset_SMDH_remove3<span class="ot">&lt;-</span>dataset_SMDH<span class="sc">%&gt;%</span><span class="fu">filter</span>(<span class="fu">is.na</span>(comparision_type) <span class="sc">|</span> comparision_type<span class="sc">!=</span><span class="st">"3"</span>)</span>
<span id="cb85-27"><a href="#cb85-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-28"><a href="#cb85-28" aria-hidden="true" tabindex="-1"></a>VCV_SMDH_remove3 <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> SMDH_variance,</span>
<span id="cb85-29"><a href="#cb85-29" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb85-30"><a href="#cb85-30" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb85-31"><a href="#cb85-31" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_SMDH_remove3,</span>
<span id="cb85-32"><a href="#cb85-32" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb85-33"><a href="#cb85-33" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb85-34"><a href="#cb85-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-35"><a href="#cb85-35" aria-hidden="true" tabindex="-1"></a>SMDH_remove3 <span class="ot">&lt;-</span></span>
<span id="cb85-36"><a href="#cb85-36" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb85-37"><a href="#cb85-37" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_SMDH_remove3 ,<span class="co"># specify lnRR's sampling variance;</span></span>
<span id="cb85-38"><a href="#cb85-38" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb85-39"><a href="#cb85-39" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb85-40"><a href="#cb85-40" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb85-41"><a href="#cb85-41" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb85-42"><a href="#cb85-42" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb85-43"><a href="#cb85-43" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb85-44"><a href="#cb85-44" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb85-45"><a href="#cb85-45" aria-hidden="true" tabindex="-1"></a>                  <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb85-46"><a href="#cb85-46" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb85-47"><a href="#cb85-47" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_SMDH_remove3 )</span>
<span id="cb85-48"><a href="#cb85-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-49"><a href="#cb85-49" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(SMDH_remove3)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Multivariate Meta-Analysis Model (k = 246; method: REML)

   logLik   Deviance        AIC        BIC       AICc   
-287.7625   575.5250   587.5250   608.5326   587.8779   

Variance Components:

            estim    sqrt  nlvls  fixed          factor 
sigma^2.1  0.0000  0.0002     26     no        paper_ID 
sigma^2.2  0.2338  0.4836    246     no  Observation_ID 
sigma^2.3  0.0000  0.0008     47     no  group_ID_coded 
sigma^2.4  0.0000  0.0002     19     no   population_ID 
sigma^2.5  0.0000  0.0000      7     no    bird_species 

Test for Heterogeneity:
Q(df = 245) = 1048.0795, p-val &lt; .0001

Model Results:

estimate      se    tval   df    pval   ci.lb   ci.ub    
  0.1415  0.0654  2.1618  245  0.0316  0.0126  0.2703  * 

---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div></div>
<p>Results remain practically the same irrespective..</p>
</section>
<section id="bird-species" class="level4">
<h4 class="anchored" data-anchor-id="bird-species">Bird species</h4>
<p>(levels: Cyanistes caeruleus, Sturnus unicolor, Tachycineta bicolor, Sturnus vulgaris; note that these levels reflect the list of species studied in our current database, which may increase after updating our search and/or receiving unpublished data from authors)</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [24]:</pre></div><div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a>lnRR_birds <span class="ot">&lt;-</span></span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> lnRR_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb87-3"><a href="#cb87-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_lnRR,<span class="co"># specify lnRR's sampling variance;</span></span>
<span id="cb87-4"><a href="#cb87-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="sc">-</span> <span class="dv">1</span> <span class="sc">+</span> bird_species,</span>
<span id="cb87-5"><a href="#cb87-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb87-6"><a href="#cb87-6" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb87-7"><a href="#cb87-7" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb87-8"><a href="#cb87-8" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb87-9"><a href="#cb87-9" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb87-10"><a href="#cb87-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb87-11"><a href="#cb87-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb87-12"><a href="#cb87-12" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb87-13"><a href="#cb87-13" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_lnRR)</span>
<span id="cb87-14"><a href="#cb87-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-15"><a href="#cb87-15" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(lnRR_birds, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/lnRR_birds.rds"</span>))</span>
<span id="cb87-16"><a href="#cb87-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-17"><a href="#cb87-17" aria-hidden="true" tabindex="-1"></a><span class="do">## Now doing this for SMDH</span></span>
<span id="cb87-18"><a href="#cb87-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-19"><a href="#cb87-19" aria-hidden="true" tabindex="-1"></a>SMDH_birds <span class="ot">&lt;-</span></span>
<span id="cb87-20"><a href="#cb87-20" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb87-21"><a href="#cb87-21" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_SMDH,<span class="co"># specify lnRR's sampling variance;</span></span>
<span id="cb87-22"><a href="#cb87-22" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="sc">-</span> <span class="dv">1</span> <span class="sc">+</span> bird_species,</span>
<span id="cb87-23"><a href="#cb87-23" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb87-24"><a href="#cb87-24" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb87-25"><a href="#cb87-25" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb87-26"><a href="#cb87-26" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb87-27"><a href="#cb87-27" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb87-28"><a href="#cb87-28" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb87-29"><a href="#cb87-29" aria-hidden="true" tabindex="-1"></a>                  <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb87-30"><a href="#cb87-30" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb87-31"><a href="#cb87-31" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_SMDH)</span>
<span id="cb87-32"><a href="#cb87-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-33"><a href="#cb87-33" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SMDH_birds, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/SMDH_birds.rds"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
</section>
<section id="type-of-trait-studied" class="level4">
<h4 class="anchored" data-anchor-id="type-of-trait-studied">Type of trait studied</h4>
<p>(levels:physiology, morphology, reproduction, behaviour, parasite and pathogenic load, phenology)</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [25]:</pre></div><div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a>lnRR_trait <span class="ot">&lt;-</span></span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> lnRR_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_lnRR,<span class="co"># specify lnRR's sampling variance;</span></span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="sc">-</span> <span class="dv">1</span> <span class="sc">+</span> trait_type,</span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb88-6"><a href="#cb88-6" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb88-7"><a href="#cb88-7" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb88-8"><a href="#cb88-8" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb88-9"><a href="#cb88-9" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb88-10"><a href="#cb88-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb88-11"><a href="#cb88-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb88-12"><a href="#cb88-12" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb88-13"><a href="#cb88-13" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_lnRR)</span>
<span id="cb88-14"><a href="#cb88-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-15"><a href="#cb88-15" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(lnRR_trait, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/lnRR_trait.rds"</span>))</span>
<span id="cb88-16"><a href="#cb88-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-17"><a href="#cb88-17" aria-hidden="true" tabindex="-1"></a><span class="do">## Now similarly for SMDH</span></span>
<span id="cb88-18"><a href="#cb88-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-19"><a href="#cb88-19" aria-hidden="true" tabindex="-1"></a>SMDH_trait <span class="ot">&lt;-</span></span>
<span id="cb88-20"><a href="#cb88-20" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb88-21"><a href="#cb88-21" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_SMDH,<span class="co"># specify lnRR's sampling variance;</span></span>
<span id="cb88-22"><a href="#cb88-22" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="sc">-</span> <span class="dv">1</span> <span class="sc">+</span> trait_type,</span>
<span id="cb88-23"><a href="#cb88-23" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb88-24"><a href="#cb88-24" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb88-25"><a href="#cb88-25" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb88-26"><a href="#cb88-26" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb88-27"><a href="#cb88-27" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb88-28"><a href="#cb88-28" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb88-29"><a href="#cb88-29" aria-hidden="true" tabindex="-1"></a>                  <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb88-30"><a href="#cb88-30" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb88-31"><a href="#cb88-31" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_SMDH)</span>
<span id="cb88-32"><a href="#cb88-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-33"><a href="#cb88-33" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SMDH_trait, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/SMDH_trait.rds"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
</section>
</section>
<section id="sensitivity-analyses-robustness-checks" class="level1">
<h1><strong>Sensitivity analyses / robustness checks</strong></h1>
<p>We will run several sensitivity analyses using the same random effect structure as described above aimed at ensuring the robustness of our results.&nbsp;</p>
<ol start="3" type="1">
<li><p>We will re-run our intercept-only multivariate meta-analysis (with both lnRR and SMDH as effect sizes) using 0.3 and 0.7 correlation (ρs) between sampling variances from the same paper_ID. If we find large differences in conclusions depending on ρs, we will run the corresponding sensitivity analyses for all our meta-regressions as well.</p></li>
<li><p>If during our analyses we notice the need to run additional sensitivity analyses to ensure the robustness of our results, we will label those analyses as non-pre-registered and report them in the supplementary materials of the eventual manuscript.</p></li>
</ol>
<section id="risk-of-bias" class="level3">
<h3 class="anchored" data-anchor-id="risk-of-bias">Risk of Bias</h3>
<p>6. Blinding: (Levels: yes, no(</p>
<p>7. Random Assignment: (Levels: yes, no)</p>
<p>8. Missing Data: (Levels: yes, no)</p>
<p>Note that the specific subset of data for each meta-regression may differ based on the data availability (e.g., information about the type of parasites is only available for effect sizes based on parasite/pathogen related fitness proxy). We will conduct meta-regression on suitable subsets of data for each analysis, provided we have sufficient effect sizes per moderator level (i.e., at least 5 effect sizes per moderator level).</p>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Phylogenetic Effects
</div>
</div>
<div class="callout-body-container callout-body">
<p>Note also that if we obtain additional effect sizes for other bird species after we update our search and/or contact authors, our models may need to account for phylogenetic nonindependence via the inclusion of two additional random effects: “species” and “phylogeny” following Cinar et al.&nbsp;(2022).</p>
</div>
</div>
</section>
<section id="other-analysis" class="level2">
<h2 class="anchored" data-anchor-id="other-analysis">Other Analysis</h2>
<section id="risk-of-bias-assessment" class="level3">
<h3 class="anchored" data-anchor-id="risk-of-bias-assessment">Risk of Bias Assessment</h3>
<ul>
<li>RoB - The outcome will be presented in the form of a <u><strong>summary table</strong></u>. We will conduct uni-moderator meta-regressions using these variables.</li>
</ul>
</section>
</section>
<section id="heterogeneity" class="level2">
<h2 class="anchored" data-anchor-id="heterogeneity">Heterogeneity</h2>
<p>To understand the sources of variation or heterogeneity,</p>
<section id="for-the-intercept-only-models" class="level3">
<h3 class="anchored" data-anchor-id="for-the-intercept-only-models">For the intercept-only models</h3>
<p>we will use the pluralistic approach recently suggested by Yang et al.&nbsp;(2023),</p>
<p>which consists of calculating four metrics of heterogeneity (σ2, I2, CV, M) to comprehensively understand both total and random-effect specific heterogeneity.</p>
</section>
<section id="for-the-meta-regressions" class="level3">
<h3 class="anchored" data-anchor-id="for-the-meta-regressions">For the meta-regressions,</h3>
<p>we will calculate Rmarginal2 to quantify the heterogeneity explained by the moderator(s) in proportion to the total variation (Nakagawa &amp; Schielzeth, 2013).</p>
</section>
</section>
<section id="publication-bias" class="level2">
<h2 class="anchored" data-anchor-id="publication-bias">Publication Bias</h2>
<p>To explore evidence of publication bias, we will run multilevel meta-regressions with the same random effects structure as described above. We will follow the approach recommended in section 4.3 from Nakagawa et al.&nbsp;(2022) for both lnRR and SMDH, which consists of the following steps:</p>
<p>First, run a uni-moderator meta-regression to test for evidence of small-study effects for both lnRR and SMDH separately, where the moderator is the square root of the inverse of the ‘effective sample size’ (see equation 27 in Nakagawa et al.&nbsp;2022). The slope of this meta-regression provides information about whether funnel plot asymmetry exists, and the intercept corresponds to the overall effect size adjusted by small-study effects. However, if this intercept is statistically significantly different from zero,&nbsp;</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [26]:</pre></div><div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a>dataset_SMDH<span class="ot">&lt;-</span>dataset_SMDH<span class="sc">%&gt;%</span></span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">effective_n_inv=</span><span class="dv">1</span><span class="sc">/</span>((<span class="dv">4</span><span class="sc">*</span>effective_n_experiment<span class="sc">*</span>effective_n_control)<span class="sc">/</span>(effective_n_experiment<span class="sc">+</span>effective_n_control)))<span class="sc">%&gt;%</span></span>
<span id="cb89-3"><a href="#cb89-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sqrt_effective_n_inv=</span><span class="fu">sqrt</span>(effective_n_inv))</span>
<span id="cb89-4"><a href="#cb89-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-5"><a href="#cb89-5" aria-hidden="true" tabindex="-1"></a>MR_SMDH_small_study <span class="ot">&lt;-</span></span>
<span id="cb89-6"><a href="#cb89-6" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb89-7"><a href="#cb89-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> SMDH_variance,<span class="co"># specify lnRR's sampling variance;</span></span>
<span id="cb89-8"><a href="#cb89-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span>sqrt_effective_n_inv,</span>
<span id="cb89-9"><a href="#cb89-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb89-10"><a href="#cb89-10" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb89-11"><a href="#cb89-11" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb89-12"><a href="#cb89-12" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb89-13"><a href="#cb89-13" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb89-14"><a href="#cb89-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb89-15"><a href="#cb89-15" aria-hidden="true" tabindex="-1"></a>                  <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb89-16"><a href="#cb89-16" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb89-17"><a href="#cb89-17" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_SMDH)</span>
<span id="cb89-18"><a href="#cb89-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-19"><a href="#cb89-19" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(MR_SMDH_small_study)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Multivariate Meta-Analysis Model (k = 258; method: REML)

   logLik   Deviance        AIC        BIC       AICc   
-295.9602   591.9205   605.9205   630.7367   606.3721   

Variance Components:

            estim    sqrt  nlvls  fixed          factor 
sigma^2.1  0.0000  0.0000     26     no        paper_ID 
sigma^2.2  0.1283  0.3582    258     no  Observation_ID 
sigma^2.3  0.0000  0.0006     47     no  group_ID_coded 
sigma^2.4  0.0228  0.1510     19     no   population_ID 
sigma^2.5  0.0000  0.0030      7     no    bird_species 

Test for Residual Heterogeneity:
QE(df = 256) = 642.1391, p-val &lt; .0001

Test of Moderators (coefficient 2):
F(df1 = 1, df2 = 256) = 0.2720, p-val = 0.6025

Model Results:

                      estimate      se     tval   df    pval    ci.lb   ci.ub 
intrcpt                 0.1450  0.1134   1.2782  256  0.2024  -0.0784  0.3684 
sqrt_effective_n_inv   -0.2598  0.4981  -0.5215  256  0.6025  -1.2407  0.7212 
                        
intrcpt                 
sqrt_effective_n_inv    

---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a>SMDH_small_study_bubble<span class="ot">&lt;-</span>orchaRd<span class="sc">::</span><span class="fu">mod_results</span>(MR_SMDH_small_study,</span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">mod=</span><span class="st">"sqrt_effective_n_inv"</span>,</span>
<span id="cb91-3"><a href="#cb91-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group=</span><span class="st">"paper_ID"</span>,</span>
<span id="cb91-4"><a href="#cb91-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">weights=</span><span class="st">"prop"</span>)</span>
<span id="cb91-5"><a href="#cb91-5" aria-hidden="true" tabindex="-1"></a>orchaRd<span class="sc">::</span><span class="fu">bubble_plot</span>(SMDH_small_study_bubble,</span>
<span id="cb91-6"><a href="#cb91-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">mod=</span><span class="st">"sqrt_effective_n_inv"</span>,</span>
<span id="cb91-7"><a href="#cb91-7" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group=</span><span class="st">"paper_ID"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="data_analysis_files/figure-html/unnamed-chunk-18-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-5"><img src="data_analysis_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(<span class="fu">r2_ml</span>(MR_SMDH_small_study)<span class="sc">*</span><span class="dv">100</span>,<span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>   R2_marginal R2_conditional 
          0.54          15.55 </code></pre>
</div>
</div></div>
<p>Then, run the following uni-moderator meta-regression for both lnRR and SMDH separately which provides a less biased adjusted overall effect. The moderator here is simply the inverse of the ‘effective sample size’ (see equation 28 in Nakagawa et al.&nbsp;2022). The slope of this meta-regression provides information about whether funnel plot asymmetry exists and the intercept corresponds to a less biased overall effect size adjusted by small-study effects.</p>
<p>Next, run a uni-moderator meta-regression to test for evidence of decline effects (also known as time-lag bias) which includes year of publication (mean-centered) as the only moderator, and whose slope would provide information about whether the overall effect size has changed (decline) over time (Nakagawa et al.&nbsp;2022; Sánchez-Tójar et al.&nbsp;2018).</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [27]:</pre></div><div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a>dataset_SMDH<span class="ot">&lt;-</span>dataset_SMDH<span class="sc">%&gt;%</span></span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(year_publication<span class="sc">!=</span><span class="st">"Unpublished"</span>)<span class="sc">%&gt;%</span></span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">year_MeanCenter=</span><span class="fu">scale</span>(<span class="fu">as.numeric</span>(year_publication),<span class="at">scale=</span>F)[,<span class="dv">1</span>])</span>
<span id="cb94-4"><a href="#cb94-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-5"><a href="#cb94-5" aria-hidden="true" tabindex="-1"></a>MR_SMDH_decline_eff <span class="ot">&lt;-</span></span>
<span id="cb94-6"><a href="#cb94-6" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb94-7"><a href="#cb94-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> SMDH_variance,<span class="co"># specify lnRR's sampling variance;</span></span>
<span id="cb94-8"><a href="#cb94-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> year_MeanCenter,</span>
<span id="cb94-9"><a href="#cb94-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb94-10"><a href="#cb94-10" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb94-11"><a href="#cb94-11" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb94-12"><a href="#cb94-12" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb94-13"><a href="#cb94-13" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb94-14"><a href="#cb94-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb94-15"><a href="#cb94-15" aria-hidden="true" tabindex="-1"></a>                  <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb94-16"><a href="#cb94-16" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb94-17"><a href="#cb94-17" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_SMDH)</span>
<span id="cb94-18"><a href="#cb94-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-19"><a href="#cb94-19" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(MR_SMDH_decline_eff)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Multivariate Meta-Analysis Model (k = 240; method: REML)

   logLik   Deviance        AIC        BIC       AICc   
-280.4498   560.8996   574.8996   599.2055   575.3866   

Variance Components:

            estim    sqrt  nlvls  fixed          factor 
sigma^2.1  0.0001  0.0097     24     no        paper_ID 
sigma^2.2  0.1340  0.3661    240     no  Observation_ID 
sigma^2.3  0.0002  0.0133     45     no  group_ID_coded 
sigma^2.4  0.0365  0.1909     17     no   population_ID 
sigma^2.5  0.0000  0.0001      6     no    bird_species 

Test for Residual Heterogeneity:
QE(df = 238) = 626.8624, p-val &lt; .0001

Test of Moderators (coefficient 2):
F(df1 = 1, df2 = 238) = 0.0574, p-val = 0.8109

Model Results:

                 estimate      se    tval   df    pval    ci.lb   ci.ub    
intrcpt            0.1117  0.0624  1.7896  238  0.0748  -0.0113  0.2346  . 
year_MeanCenter    0.0014  0.0058  0.2395  238  0.8109  -0.0101  0.0129    

---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(<span class="fu">r2_ml</span>(MR_SMDH_decline_eff)<span class="sc">*</span><span class="dv">100</span>,<span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>   R2_marginal R2_conditional 
          0.14          21.61 </code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a>MR_SMDH_decline_effect_bubble<span class="ot">&lt;-</span>orchaRd<span class="sc">::</span><span class="fu">mod_results</span>(MR_SMDH_decline_eff,</span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">mod=</span><span class="st">"year_MeanCenter"</span>,</span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group=</span><span class="st">"paper_ID"</span>,</span>
<span id="cb98-4"><a href="#cb98-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">weights=</span><span class="st">"prop"</span>)</span>
<span id="cb98-5"><a href="#cb98-5" aria-hidden="true" tabindex="-1"></a>orchaRd<span class="sc">::</span><span class="fu">bubble_plot</span>(MR_SMDH_decline_effect_bubble,</span>
<span id="cb98-6"><a href="#cb98-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">mod=</span><span class="st">"year_MeanCenter"</span>,</span>
<span id="cb98-7"><a href="#cb98-7" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group=</span><span class="st">"paper_ID"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="data_analysis_files/figure-html/unnamed-chunk-19-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-6"><img src="data_analysis_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
</div></div>
<p>Last, run an all-in meta-regression including both the (square root of the) inverse of the ‘effective sample size’ and the mean-centered year of publication as well as all the other moderators tested in our study to estimate how much heterogeneity is explained by all moderators combined as well as to explore whether evidence remains similar after accounting for all the other moderators (see equation 29 in Nakagawa et al.&nbsp;2022 for more information about this approach).&nbsp;</p>
</section>
</section>
     </main>
<!-- /main column -->  <script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>  </div> <!-- /content -->  <script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script> 
  
</body></html>