<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
    <meta charset="utf-8">
    <meta name="generator" content="quarto-1.7.32">

    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

    <meta name="author" content="Shreya D.">
    <meta name="dcterms.date" content="2025-07-07">

    <title>Data Analysis</title>
    <style>
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      div.columns{display: flex; gap: min(4vw, 1.5em);}
      div.column{flex: auto; overflow-x: auto;}
      div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
      ul.task-list{list-style: none;}
      ul.task-list li input[type="checkbox"] {
        width: 0.8em;
        margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
        vertical-align: middle;
      }
      /* CSS for syntax highlighting */
      html { -webkit-text-size-adjust: 100%; }
      pre > code.sourceCode { white-space: pre; position: relative; }
      pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
      pre > code.sourceCode > span:empty { height: 1.2em; }
      .sourceCode { overflow: visible; }
      code.sourceCode > span { color: inherit; text-decoration: inherit; }
      div.sourceCode { margin: 1em 0; }
      pre.sourceCode { margin: 0; }
      @media screen {
      div.sourceCode { overflow: auto; }
      }
      @media print {
      pre > code.sourceCode { white-space: pre-wrap; }
      pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
      }
      pre.numberSource code
        { counter-reset: source-line 0; }
      pre.numberSource code > span
        { position: relative; left: -4em; counter-increment: source-line; }
      pre.numberSource code > span > a:first-child::before
        { content: counter(source-line);
          position: relative; left: -1em; text-align: right; vertical-align: baseline;
          border: none; display: inline-block;
          -webkit-touch-callout: none; -webkit-user-select: none;
          -khtml-user-select: none; -moz-user-select: none;
          -ms-user-select: none; user-select: none;
          padding: 0 4px; width: 4em;
        }
      pre.numberSource { margin-left: 3em;  padding-left: 4px; }
      div.sourceCode
        {   }
      @media screen {
      pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
      }
    </style>

    <style>
      body.hypothesis-enabled #quarto-embed-header {
        padding-right: 36px;
      }

      #quarto-embed-header {
        height: 3em;
        width: 100%;
        display: flex;
        justify-content: space-between;
        align-items: center;
        border-bottom: solid 1px;
      }

      #quarto-embed-header h6 {
        font-size: 1.1em;
        padding-top: 0.6em;
        margin-left: 1em;
        margin-right: 1em;
        font-weight: 400;
      }

      #quarto-embed-header a.quarto-back-link,
      #quarto-embed-header a.quarto-download-embed {
        font-size: 0.8em;
        margin-top: 1em;
        margin-bottom: 1em;
        margin-left: 1em;
        margin-right: 1em;
      }

      .quarto-back-container {
        padding-left: 0.5em;
        display: flex;
      }

      .headroom {
          will-change: transform;
          transition: transform 200ms linear;
      }

      .headroom--pinned {
          transform: translateY(0%);
      }

      .headroom--unpinned {
          transform: translateY(-100%);
      }      
    </style>

    <script>
    window.document.addEventListener("DOMContentLoaded", function () {

      var header = window.document.querySelector("#quarto-embed-header");
      const titleBannerEl = window.document.querySelector("body > #title-block-header");
      if (titleBannerEl) {
        titleBannerEl.style.paddingTop = header.clientHeight + "px";
      }
      const contentEl = window.document.getElementById('quarto-content');
      for (const child of contentEl.children) {
        child.style.paddingTop = header.clientHeight + "px";
        child.style.marginTop = "1em";
      }

      // Use the article root if the `back` call doesn't work. This isn't perfect
      // but should typically work
      window.quartoBackToArticle = () => {
        var currentUrl = window.location.href;
        window.history.back();
        setTimeout(() => {
            // if location was not changed in 100 ms, then there is no history back
            if(currentUrl === window.location.href){              
                // redirect to site root
                window.location.href = "../index.html";
            }
        }, 100);
      }

      const headroom = new window.Headroom(header, {
        tolerance: 5,
        onPin: function () {
        },
        onUnpin: function () {
        },
      });
      headroom.init();
    });
    </script>

    
<script src="../site_libs/manuscript-notebook/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-8a894fcc3cb5e37eb5bf30def131be2f.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
     <script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>   <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script> 
      </head>

  <body class="quarto-notebook quarto-light">
    <div id="quarto-embed-header" class="headroom fixed-top bg-primary">
      
      <a onclick="window.quartoBackToArticle(); return false;" class="btn btn-primary quarto-back-link" href=""><i class="bi bi-caret-left"></i> Back to Article</a>
      <h6><i class="bi bi-journal-code"></i> Data Analysis</h6>

            <a href="../code/07_data_analysis.Rmd" class="btn btn-primary quarto-download-embed" download="07_data_analysis.Rmd">Download Notebook</a>
          </div>

     <div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#loading-data-for-analysis" id="toc-loading-data-for-analysis" class="nav-link active" data-scroll-target="#loading-data-for-analysis">Loading Data for analysis</a>
  <ul class="collapse">
  <li><a href="#creating-phylogeny-for-the-plot" id="toc-creating-phylogeny-for-the-plot" class="nav-link" data-scroll-target="#creating-phylogeny-for-the-plot">Creating Phylogeny for the plot</a></li>
  <li><a href="#visualizing-random-effects-structure" id="toc-visualizing-random-effects-structure" class="nav-link" data-scroll-target="#visualizing-random-effects-structure">Visualizing random effects structure</a></li>
  </ul></li>
  <li><a href="#preparing-datasets" id="toc-preparing-datasets" class="nav-link" data-scroll-target="#preparing-datasets">Preparing datasets</a></li>
  <li><a href="#overall-effect-intercept-only-meta-analytic-model" id="toc-overall-effect-intercept-only-meta-analytic-model" class="nav-link" data-scroll-target="#overall-effect-intercept-only-meta-analytic-model">Overall effect : Intercept-only meta-analytic model</a>
  <ul class="collapse">
  <li><a href="#using-lnrr" id="toc-using-lnrr" class="nav-link" data-scroll-target="#using-lnrr">using lnRR</a></li>
  <li><a href="#using-smdh" id="toc-using-smdh" class="nav-link" data-scroll-target="#using-smdh">using SMDH</a></li>
  <li><a href="#smdh-heterogeneity" id="toc-smdh-heterogeneity" class="nav-link" data-scroll-target="#smdh-heterogeneity">SMDH-heterogeneity</a></li>
  <li><a href="#bivariate-meta-analytical-model" id="toc-bivariate-meta-analytical-model" class="nav-link" data-scroll-target="#bivariate-meta-analytical-model">Bivariate Meta-analytical model</a></li>
  </ul></li>
  <li><a href="#sensitivity-analysis" id="toc-sensitivity-analysis" class="nav-link" data-scroll-target="#sensitivity-analysis">Sensitivity Analysis</a>
  <ul class="collapse">
  <li><a href="#with-published-data-only" id="toc-with-published-data-only" class="nav-link" data-scroll-target="#with-published-data-only">With published data only</a></li>
  <li><a href="#with-flagged-proxies" id="toc-with-flagged-proxies" class="nav-link" data-scroll-target="#with-flagged-proxies">With flagged proxies</a></li>
  <li><a href="#without-0-es" id="toc-without-0-es" class="nav-link" data-scroll-target="#without-0-es">Without 0 ES</a></li>
  <li><a href="#with-flagged-proxies-and-no-missing-es" id="toc-with-flagged-proxies-and-no-missing-es" class="nav-link" data-scroll-target="#with-flagged-proxies-and-no-missing-es">With flagged proxies and no missing ES</a></li>
  <li><a href="#no-inferential-stats-for-smdh-effect-sizes-calculated-from-means-sds-and-sample-size" id="toc-no-inferential-stats-for-smdh-effect-sizes-calculated-from-means-sds-and-sample-size" class="nav-link" data-scroll-target="#no-inferential-stats-for-smdh-effect-sizes-calculated-from-means-sds-and-sample-size">No inferential stats for SMDH: effect sizes calculated from means, SDs, and sample size</a></li>
  <li><a href="#exluding-failed-gearys-test-for-lnrr" id="toc-exluding-failed-gearys-test-for-lnrr" class="nav-link" data-scroll-target="#exluding-failed-gearys-test-for-lnrr">Exluding failed geary’s test for lnRR</a></li>
  <li><a href="#excluding-the-parasite-and-pathogen-load-fitness-proxies" id="toc-excluding-the-parasite-and-pathogen-load-fitness-proxies" class="nav-link" data-scroll-target="#excluding-the-parasite-and-pathogen-load-fitness-proxies">Excluding the parasite and pathogen load fitness proxies</a></li>
  <li><a href="#different-values-of-sampling-variance-correlation" id="toc-different-values-of-sampling-variance-correlation" class="nav-link" data-scroll-target="#different-values-of-sampling-variance-correlation">Different values of sampling variance correlation</a></li>
  </ul></li>
  <li><a href="#mechanistic-hypothesis" id="toc-mechanistic-hypothesis" class="nav-link" data-scroll-target="#mechanistic-hypothesis">Mechanistic Hypothesis</a></li>
  <li><a href="#meta-regressions" id="toc-meta-regressions" class="nav-link" data-scroll-target="#meta-regressions">Meta-regressions</a>
  <ul class="collapse">
  <li><a href="#type-of-experimental-design" id="toc-type-of-experimental-design" class="nav-link" data-scroll-target="#type-of-experimental-design">Type of experimental design</a>
  <ul class="collapse">
  <li><a href="#excluding-non-aromatic-and-blank-comparison-i.e.-the-two-control-group-comparison" id="toc-excluding-non-aromatic-and-blank-comparison-i.e.-the-two-control-group-comparison" class="nav-link" data-scroll-target="#excluding-non-aromatic-and-blank-comparison-i.e.-the-two-control-group-comparison">Excluding non-aromatic and blank comparison (i.e.&nbsp;the two control group comparison)</a></li>
  </ul></li>
  <li><a href="#time-of-addition-of-green-nest-material" id="toc-time-of-addition-of-green-nest-material" class="nav-link" data-scroll-target="#time-of-addition-of-green-nest-material">Time of addition of green nest material</a></li>
  <li><a href="#type-of-parasites" id="toc-type-of-parasites" class="nav-link" data-scroll-target="#type-of-parasites">Type of parasites</a></li>
  </ul></li>
  <li><a href="#risk-of-bias" id="toc-risk-of-bias" class="nav-link" data-scroll-target="#risk-of-bias">Risk of Bias</a>
  <ul class="collapse">
  <li><a href="#other-analysis" id="toc-other-analysis" class="nav-link" data-scroll-target="#other-analysis">Other Analysis</a>
  <ul class="collapse">
  <li><a href="#risk-of-bias-assessment" id="toc-risk-of-bias-assessment" class="nav-link" data-scroll-target="#risk-of-bias-assessment">Risk of Bias Assessment</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#publication-bias" id="toc-publication-bias" class="nav-link" data-scroll-target="#publication-bias">Publication Bias</a>
  <ul class="collapse">
  <li><a href="#all-in-model-for-heterogeneity" id="toc-all-in-model-for-heterogeneity" class="nav-link" data-scroll-target="#all-in-model-for-heterogeneity">All-in Model for heterogeneity</a></li>
  </ul></li>
  <li><a href="#pair-wise-comparison-for-meta-regression" id="toc-pair-wise-comparison-for-meta-regression" class="nav-link" data-scroll-target="#pair-wise-comparison-for-meta-regression">Pair-wise comparison for meta-regression</a></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content" id="quarto-document-content">      <header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Data Analysis</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Shreya D. </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">July 7, 2025</p>
    </div>
  </div>
  
    
  </div>
  


</header>     <section id="loading-data-for-analysis" class="level1">
<h1>Loading Data for analysis</h1>
<div class="cell-container"><div class="cell-decorator"><pre>In [1]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Setting up workspace</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co"># For plotting trees, and although not strictly necessary, we may need the `ggtree` R package, which you should be able to install by running:</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a> <span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"BiocManager"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a> <span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a> BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"ggtree"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Bioconductor version 3.21 (BiocManager 1.30.26), R 4.5.1 (2025-06-13)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package(s) not installed when version(s) same as or greater than current; use
  `force = TRUE` to re-install: 'ggtree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Old packages: 'aplot', 'doBy', 'ggpubr', 'pillar', 'Rcpp', 'shiny'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"orchaRd"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">"daniel1noble/orchaRd"</span>, <span class="at">ref =</span> <span class="st">"main"</span>, <span class="at">force =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Loading the 'orchaRd' package (version 2.0). For an
introduction and vignette to the package please see: https://daniel1noble.github.io/orchaRd/</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(here,tidyverse,metafor,ggpubr,orchaRd,ggpubfigs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'ggpubfigs' is not available for this version of R

A version of this package for your version of R might be available elsewhere,
see the ideas at
https://cran.r-project.org/doc/manuals/r-patched/R-admin.html#Installing-packages</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in p_install(package, character.only = TRUE, ...):</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in library(package, lib.loc = lib.loc, character.only = TRUE,
logical.return = TRUE, : there is no package called 'ggpubfigs'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in pacman::p_load(here, tidyverse, metafor, ggpubr, orchaRd, ggpubfigs): Failed to install/load:
ggpubfigs</code></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Packages needed</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list=</span><span class="fu">ls</span>()) <span class="do">## cleaning up</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">430</span>) <span class="co">#set seed for reproducibility (for random number generation)</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Functions needed</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a><span class="co"># function to estimate typical sampling error variance obtained from </span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a><span class="co"># https://github.com/Yefeng0920/heterogeneity_ecoevo/tree/main/function</span></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>sigma2_v <span class="ot">&lt;-</span> <span class="cf">function</span>(mod){</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>  sigma2_v <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="dv">1</span> <span class="sc">/</span> mod<span class="sc">$</span>vi) <span class="sc">*</span> (mod<span class="sc">$</span>k <span class="sc">-</span> <span class="dv">1</span>) <span class="sc">/</span></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>    (<span class="fu">sum</span>(<span class="dv">1</span> <span class="sc">/</span> mod<span class="sc">$</span>vi)<span class="sc">^</span><span class="dv">2</span> <span class="sc">-</span> <span class="fu">sum</span>((<span class="dv">1</span> <span class="sc">/</span> mod<span class="sc">$</span>vi)<span class="sc">^</span><span class="dv">2</span>))</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(sigma2_v)</span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a><span class="co"># ------------------------------------------------------------</span></span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Loading cleaned dataset</span></span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a><span class="co"># ------------------------------------------------------------</span></span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a>dataset_analysis<span class="ot">&lt;-</span><span class="fu">read_csv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/04_data_analysis/dataset_analysis.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 283 Columns: 71</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (35): paper_ID, fulltext_screening, variable_note, authors, population_l...
dbl (35): year_publication, Observation_ID, experiment_ID, group_ID, repeate...
lgl  (1): fulltext_notes

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>dataset_lnRR<span class="ot">&lt;-</span><span class="fu">read_csv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/04_data_analysis/dataset_lnRR.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 268 Columns: 71
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (34): paper_ID, fulltext_screening, variable_note, authors, population_l...
dbl (32): year_publication, Observation_ID, experiment_ID, group_ID, repeate...
lgl  (5): fulltext_notes, test_statistics_type, sign_relationship, total_sam...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>dataset_SMDH<span class="ot">&lt;-</span><span class="fu">read_csv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/04_data_analysis/dataset_SMDH.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 274 Columns: 71
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (35): paper_ID, fulltext_screening, variable_note, authors, population_l...
dbl (35): year_publication, Observation_ID, experiment_ID, group_ID, repeate...
lgl  (1): fulltext_notes

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
</div></div>
<section id="creating-phylogeny-for-the-plot" class="level3">
<h3 class="anchored" data-anchor-id="creating-phylogeny-for-the-plot">Creating Phylogeny for the plot</h3>
<div class="cell-container"><div class="cell-decorator"><pre>In [2]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages("pacman")</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="co"># load packages</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(rotl,</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>               ape,</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>               ggtree)</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a><span class="co"># We have 7 bird species in the dataset</span></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>bird_species <span class="ot">&lt;-</span> <span class="fu">unique</span>(dataset_analysis<span class="sc">$</span>bird_species)</span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a><span class="co"># extracting taxonomic information</span></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>taxa <span class="ot">&lt;-</span> rotl<span class="sc">::</span><span class="fu">tnrs_match_names</span>(<span class="at">names =</span> bird_species)</span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a>taxa</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>        search_string         unique_name approximate_match score  ott_id
1 cyanistes caeruleus Cyanistes caeruleus             FALSE     1  746120
2    sturnus unicolor    Sturnus unicolor             FALSE     1  366470
3 tachycineta bicolor Tachycineta bicolor             FALSE     1  136028
4    sturnus vulgaris    Sturnus vulgaris             FALSE     1  565813
5         parus major         Parus major             FALSE     1  515143
6  passer cinnamomeus  Passer cinnamomeus             FALSE     1 7661508
7         buteo buteo         Buteo buteo             FALSE     1  119214
  is_synonym          flags number_matches
1      FALSE                             1
2      FALSE sibling_higher              1
3      FALSE                             1
4      FALSE sibling_higher              1
5      FALSE                             1
6      FALSE                             1
7      FALSE sibling_higher              1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># check approximate matches</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>taxa[taxa<span class="sc">$</span>approximate_match<span class="sc">==</span><span class="cn">TRUE</span> <span class="sc">&amp;</span> <span class="sc">!</span>(<span class="fu">is.na</span>(taxa<span class="sc">$</span>approximate_match)),]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] search_string     unique_name       approximate_match score            
[5] ott_id            is_synonym        flags             number_matches   
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
</div>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># phylogenetic information for our species from the Open Tree of Life</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>tree <span class="ot">&lt;-</span> rotl<span class="sc">::</span><span class="fu">tol_induced_subtree</span>(<span class="at">ott_ids =</span> taxa[[<span class="st">"ott_id"</span>]], <span class="at">label_format =</span> <span class="st">"name"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Progress [----------------------------------] 0/99 (  0%) ?s
Progress [================================] 99/99 (100%)  0s
                                                            </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in collapse_singles(tr, show_progress): Dropping singleton nodes with
labels: mrcaott246ott2907, mrcaott246ott3600042, mrcaott246ott47588,
mrcaott246ott7113, Passeriformes, mrcaott246ott3212, mrcaott246ott428578,
mrcaott246ott44866, mrcaott246ott5929, mrcaott246ott32658, mrcaott246ott4820,
mrcaott246ott310390, mrcaott246ott176461, mrcaott246ott22325,
mrcaott246ott10351, mrcaott246ott193904, mrcaott246ott5935, mrcaott246ott1566,
mrcaott1566ott3598440, mrcaott1566ott496009, mrcaott1566ott2175,
mrcaott2175ott600902, mrcaott2175ott259082, mrcaott2175ott59905,
mrcaott2175ott2224, mrcaott2224ott366470, mrcaott3364ott310375,
mrcaott3364ott3370, mrcaott4083ott35042, mrcaott4083ott370807,
mrcaott4083ott469177, mrcaott4083ott18592, mrcaott4083ott4088,
mrcaott4088ott9416, mrcaott9416ott840030, mrcaott9416ott96148,
mrcaott9416ott7068473, Passer, mrcaott9416ott7661508, mrcaott1488ott72472,
mrcaott1488ott63797, mrcaott1488ott284404, mrcaott1488ott107463,
mrcaott1488ott17016, mrcaott1488ott107476, mrcaott1488ott104673,
mrcaott1488ott16185, mrcaott16185ott274931, Hirundinidae, mrcaott16185ott67916,
mrcaott16185ott16194, mrcaott16185ott136028, mrcaott136028ott1050413,
Tachycineta, mrcaott136028ott621794, mrcaott2375ott73144, mrcaott2375ott124085,
mrcaott2375ott71358, mrcaott2375ott814750, mrcaott2375ott61147,
mrcaott84656ott325811, mrcaott84656ott325806, mrcaott84656ott875992, Parus,
mrcaott84656ott3598691, mrcaott123763ott995871, mrcaott123763ott258794,
Accipitriformes, mrcaott1858ott1036186, mrcaott1858ott806938, Accipitridae,
Accipitrinae, mrcaott1858ott47576, mrcaott1858ott103122, mrcaott1858ott8285,
mrcaott8285ott34284, mrcaott8285ott11582, mrcaott8285ott919198,
mrcaott8285ott68001, mrcaott8285ott243416, mrcaott8285ott327100,
mrcaott8285ott376409, mrcaott8285ott33798, mrcaott8285ott33791,
mrcaott8285ott1008547, mrcaott8285ott1034654, mrcaott8285ott33792,
mrcaott33792ott458092, mrcaott33792ott120833, mrcaott33792ott34339,
mrcaott33792ott458099, mrcaott33792ott33796, mrcaott33796ott34349,
mrcaott33796ott33797, mrcaott33796ott554772, mrcaott33796ott215782,
mrcaott215782ott236917, mrcaott215782ott433865, mrcaott215782ott805051</code></pre>
</div>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>ape<span class="sc">::</span><span class="fu">is.binary</span>(tree)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># here are the species included in the tree</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sort</span>(tree<span class="sc">$</span>tip.label)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Buteo_buteo"         "Cyanistes_caeruleus" "Parus_major"        
[4] "Passer_cinnamomeus"  "Sturnus_unicolor"    "Sturnus_vulgaris"   
[7] "Tachycineta_bicolor"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># removing the underscore "_" from the tree tip.label</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>tree<span class="sc">$</span>tip.label <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_"</span>,<span class="st">" "</span>, tree<span class="sc">$</span>tip.label)</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="fu">sort</span>(tree<span class="sc">$</span>tip.label)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Buteo buteo"         "Cyanistes caeruleus" "Parus major"        
[4] "Passer cinnamomeus"  "Sturnus unicolor"    "Sturnus vulgaris"   
[7] "Tachycineta bicolor"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># compute branch lengths of tree</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>phylo_branch <span class="ot">&lt;-</span> <span class="fu">compute.brlen</span>(tree, <span class="at">method =</span> <span class="st">"Grafen"</span>, <span class="at">power =</span> <span class="dv">1</span>)</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a><span class="co"># check tree is ultrametric</span></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a><span class="fu">is.ultrametric</span>(phylo_branch) <span class="co"># TRUE</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co"># matrix to be included in the models</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>bird_phylo <span class="ot">&lt;-</span> <span class="fu">vcv</span>(phylo_branch, <span class="at">cor =</span> T)</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>bird_phylo</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                    Sturnus unicolor Sturnus vulgaris Passer cinnamomeus
Sturnus unicolor           1.0000000        0.8333333          0.6666667
Sturnus vulgaris           0.8333333        1.0000000          0.6666667
Passer cinnamomeus         0.6666667        0.6666667          1.0000000
Tachycineta bicolor        0.1666667        0.1666667          0.1666667
Parus major                0.1666667        0.1666667          0.1666667
Cyanistes caeruleus        0.1666667        0.1666667          0.1666667
Buteo buteo                0.0000000        0.0000000          0.0000000
                    Tachycineta bicolor Parus major Cyanistes caeruleus
Sturnus unicolor              0.1666667   0.1666667           0.1666667
Sturnus vulgaris              0.1666667   0.1666667           0.1666667
Passer cinnamomeus            0.1666667   0.1666667           0.1666667
Tachycineta bicolor           1.0000000   0.6666667           0.6666667
Parus major                   0.6666667   1.0000000           0.8333333
Cyanistes caeruleus           0.6666667   0.8333333           1.0000000
Buteo buteo                   0.0000000   0.0000000           0.0000000
                    Buteo buteo
Sturnus unicolor              0
Sturnus vulgaris              0
Passer cinnamomeus            0
Tachycineta bicolor           0
Parus major                   0
Cyanistes caeruleus           0
Buteo buteo                   1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(tree) <span class="co">#check: https://www.rdocumentation.org/packages/ape/versions/5.3/topics/plot.phylo</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="07_data_analysis_files/figure-html/unnamed-chunk-1-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1"><img src="07_data_analysis_files/figure-html/unnamed-chunk-1-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
</div></div>
</section>
<section id="visualizing-random-effects-structure" class="level3">
<h3 class="anchored" data-anchor-id="visualizing-random-effects-structure">Visualizing random effects structure</h3>
<div class="cell-container"><div class="cell-decorator"><pre>In [3]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages("devtools")</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">"davidsjoberg/ggsankey"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Using GitHub PAT from the git credential store.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Skipping install of 'ggsankey' from a github remote, the SHA1 (b675d0d5) has not changed since last install.
  Use `force = TRUE` to force installation</code></pre>
</div>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggsankey)</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>sankey_data <span class="ot">&lt;-</span> dataset_analysis <span class="sc">%&gt;%</span></span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Observation_ID, repeated_trait_ID_coded, paper_ID, experiment_ID_coded, group_ID_coded)<span class="sc">%&gt;%</span></span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(Observation_ID, repeated_trait_ID_coded, paper_ID, experiment_ID_coded, group_ID_coded), as.factor)) <span class="sc">%&gt;%</span></span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">make_long</span>(Observation_ID, repeated_trait_ID_coded, paper_ID, experiment_ID_coded, group_ID_coded)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: attributes are not identical across measure variables; they will be
dropped</code></pre>
</div>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>sankey_data <span class="ot">&lt;-</span> dataset_analysis <span class="sc">%&gt;%</span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(paper_ID, bird_species, population_ID)<span class="sc">%&gt;%</span></span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(paper_ID, bird_species, population_ID), as.factor)) <span class="sc">%&gt;%</span></span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">make_long</span>(paper_ID, bird_species, population_ID)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: attributes are not identical across measure variables; they will be
dropped</code></pre>
</div>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(sankey_data, <span class="fu">aes</span>(<span class="at">x =</span> x, </span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">next_x =</span> next_x, </span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">node =</span> node, </span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">next_node =</span> next_node,</span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">fill =</span> <span class="fu">factor</span>(node))) <span class="sc">+</span></span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_sankey</span>() <span class="sc">+</span></span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_discrete</span>(<span class="at">drop=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb44-8"><a href="#cb44-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="07_data_analysis_files/figure-html/unnamed-chunk-2-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-2"><img src="07_data_analysis_files/figure-html/unnamed-chunk-2-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(sankey_data, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">next_x =</span> next_x, <span class="at">node =</span> node, <span class="at">next_node =</span> next_node, <span class="at">fill =</span> <span class="fu">factor</span>(node))) <span class="sc">+</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_alluvial</span>(<span class="at">flow.alpha =</span> .<span class="dv">6</span>) <span class="sc">+</span></span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis_d</span>(<span class="at">drop =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_alluvial</span>(<span class="at">base_size =</span> <span class="dv">18</span>) <span class="sc">+</span></span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> .<span class="dv">5</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="07_data_analysis_files/figure-html/unnamed-chunk-2-2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-3"><img src="07_data_analysis_files/figure-html/unnamed-chunk-2-2.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
</div></div>
</section>
</section>
<section id="preparing-datasets" class="level1">
<h1>Preparing datasets</h1>
<div class="cell-container"><div class="cell-decorator"><pre>In [4]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Let's just quickly ensure all the important variables are </span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a><span class="co"># in the correct data type</span></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>dataset_analysis<span class="ot">&lt;-</span> dataset_analysis<span class="sc">%&gt;%</span><span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(paper_ID, Observation_ID, repeated_trait_ID_coded, paper_ID, experiment_ID_coded, group_ID_coded, comparision_type,Hypothesis,trait_type,parasite_type,time_of_gnm_addition, blinding, random_assignment, missing_data, population_ID,bird_species), as.factor))</span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a>dataset_lnRR<span class="ot">&lt;-</span> dataset_lnRR<span class="sc">%&gt;%</span><span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(paper_ID, Observation_ID, repeated_trait_ID_coded, paper_ID, experiment_ID_coded, group_ID_coded, comparision_type,Hypothesis,trait_type,parasite_type,time_of_gnm_addition, blinding, random_assignment, missing_data, population_ID, bird_species), as.factor))</span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-9"><a href="#cb46-9" aria-hidden="true" tabindex="-1"></a>dataset_SMDH<span class="ot">&lt;-</span> dataset_SMDH<span class="sc">%&gt;%</span><span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(paper_ID, Observation_ID, repeated_trait_ID_coded, paper_ID, experiment_ID_coded, group_ID_coded, comparision_type,Hypothesis,trait_type,parasite_type,time_of_gnm_addition, blinding, random_assignment, missing_data, population_ID,bird_species), as.factor))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [5]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="do">## For lnRR (or simply load dataset as provided)</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a><span class="co"># dataset_lnRR&lt;-dataset_analysis%&gt;%</span></span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a><span class="co"># filter(lnRR_sign != "NA" | lnRR_variance != "NA")</span></span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a><span class="co"># selecting dataset which are not flagged</span></span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>dataset_lnRR<span class="ot">&lt;-</span>dataset_lnRR<span class="sc">%&gt;%</span></span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(proxy_decision <span class="sc">==</span> <span class="st">"include"</span>)</span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a><span class="co"># selecting only published dataset</span></span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a>dataset_lnRR_published<span class="ot">&lt;-</span>dataset_lnRR<span class="sc">%&gt;%</span></span>
<span id="cb47-11"><a href="#cb47-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(year_publication))</span>
<span id="cb47-12"><a href="#cb47-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-13"><a href="#cb47-13" aria-hidden="true" tabindex="-1"></a><span class="co"># selecting dataset which are flagged</span></span>
<span id="cb47-14"><a href="#cb47-14" aria-hidden="true" tabindex="-1"></a>lnRR_con_flagged<span class="ot">&lt;-</span>dataset_analysis<span class="sc">%&gt;%</span></span>
<span id="cb47-15"><a href="#cb47-15" aria-hidden="true" tabindex="-1"></a>   <span class="fu">filter</span>(lnRR_sign <span class="sc">!=</span> <span class="st">"NA"</span> <span class="sc">|</span> lnRR_variance <span class="sc">!=</span> <span class="st">"NA"</span>)<span class="sc">%&gt;%</span></span>
<span id="cb47-16"><a href="#cb47-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(proxy_decision <span class="sc">!=</span> <span class="st">"SMDH only"</span> <span class="sc">|</span> proxy_decision <span class="sc">!=</span> <span class="st">"contingency table"</span> )</span>
<span id="cb47-17"><a href="#cb47-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-18"><a href="#cb47-18" aria-hidden="true" tabindex="-1"></a><span class="co"># selecting dataset without those using 0 as ES because authors reported no effect</span></span>
<span id="cb47-19"><a href="#cb47-19" aria-hidden="true" tabindex="-1"></a>lnRR_sin_missingES <span class="ot">&lt;-</span> dataset_lnRR <span class="sc">%&gt;%</span> </span>
<span id="cb47-20"><a href="#cb47-20" aria-hidden="true" tabindex="-1"></a>   <span class="fu">filter</span>(<span class="sc">!</span>proxy_comment <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"use 0 as ES"</span>))</span>
<span id="cb47-21"><a href="#cb47-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-22"><a href="#cb47-22" aria-hidden="true" tabindex="-1"></a>lnRR_flagged_no_missingES<span class="ot">&lt;-</span> lnRR_con_flagged <span class="sc">%&gt;%</span> </span>
<span id="cb47-23"><a href="#cb47-23" aria-hidden="true" tabindex="-1"></a>   <span class="fu">filter</span>(<span class="sc">!</span>proxy_comment <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"use 0 as ES"</span>))</span>
<span id="cb47-24"><a href="#cb47-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-25"><a href="#cb47-25" aria-hidden="true" tabindex="-1"></a><span class="co"># selecting only dataset that passes geary's test</span></span>
<span id="cb47-26"><a href="#cb47-26" aria-hidden="true" tabindex="-1"></a>lnRR_geary_passed<span class="ot">&lt;-</span>dataset_lnRR<span class="sc">%&gt;%</span><span class="fu">filter</span>(geary_test<span class="sc">==</span><span class="st">"pass"</span>)</span>
<span id="cb47-27"><a href="#cb47-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-28"><a href="#cb47-28" aria-hidden="true" tabindex="-1"></a><span class="do">## For SMDH</span></span>
<span id="cb47-29"><a href="#cb47-29" aria-hidden="true" tabindex="-1"></a><span class="co"># selecting dataset which are not flagged</span></span>
<span id="cb47-30"><a href="#cb47-30" aria-hidden="true" tabindex="-1"></a>dataset_SMDH<span class="ot">&lt;-</span>dataset_analysis<span class="sc">%&gt;%</span></span>
<span id="cb47-31"><a href="#cb47-31" aria-hidden="true" tabindex="-1"></a>   <span class="fu">filter</span>(proxy_decision <span class="sc">==</span> <span class="st">"include"</span><span class="sc">|</span> proxy_decision <span class="sc">==</span> <span class="st">"SMDH only"</span><span class="sc">|</span>proxy_decision <span class="sc">==</span><span class="st">"contingency table"</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(SMDH_sign <span class="sc">!=</span> <span class="st">"NA"</span>)</span>
<span id="cb47-32"><a href="#cb47-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-33"><a href="#cb47-33" aria-hidden="true" tabindex="-1"></a><span class="co"># selecting only published dataset</span></span>
<span id="cb47-34"><a href="#cb47-34" aria-hidden="true" tabindex="-1"></a>dataset_SMDH_published<span class="ot">&lt;-</span>dataset_SMDH<span class="sc">%&gt;%</span></span>
<span id="cb47-35"><a href="#cb47-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(year_publication))</span>
<span id="cb47-36"><a href="#cb47-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-37"><a href="#cb47-37" aria-hidden="true" tabindex="-1"></a><span class="co"># selecting dataset which are flagged</span></span>
<span id="cb47-38"><a href="#cb47-38" aria-hidden="true" tabindex="-1"></a>SMDH_con_flagged<span class="ot">&lt;-</span>dataset_analysis<span class="sc">%&gt;%</span></span>
<span id="cb47-39"><a href="#cb47-39" aria-hidden="true" tabindex="-1"></a>   <span class="fu">filter</span>(SMDH_sign <span class="sc">!=</span> <span class="st">"NA"</span>)</span>
<span id="cb47-40"><a href="#cb47-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-41"><a href="#cb47-41" aria-hidden="true" tabindex="-1"></a><span class="co"># selecting dataset without those using 0 as ES because authors reported no effect</span></span>
<span id="cb47-42"><a href="#cb47-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-43"><a href="#cb47-43" aria-hidden="true" tabindex="-1"></a>SMDH_sin_missingES <span class="ot">&lt;-</span> dataset_SMDH <span class="sc">%&gt;%</span> </span>
<span id="cb47-44"><a href="#cb47-44" aria-hidden="true" tabindex="-1"></a>   <span class="fu">filter</span>(<span class="sc">!</span>proxy_comment <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"use 0 as ES"</span>))</span>
<span id="cb47-45"><a href="#cb47-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-46"><a href="#cb47-46" aria-hidden="true" tabindex="-1"></a>SMDH_flagged_no_missingES<span class="ot">&lt;-</span> SMDH_con_flagged <span class="sc">%&gt;%</span> </span>
<span id="cb47-47"><a href="#cb47-47" aria-hidden="true" tabindex="-1"></a>   <span class="fu">filter</span>(<span class="sc">!</span>proxy_comment <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"use 0 as ES"</span>))</span>
<span id="cb47-48"><a href="#cb47-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-49"><a href="#cb47-49" aria-hidden="true" tabindex="-1"></a><span class="co"># selecting dataset which are not flagged but without the SMDH calculated from inferential statistics </span></span>
<span id="cb47-50"><a href="#cb47-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-51"><a href="#cb47-51" aria-hidden="true" tabindex="-1"></a>SMDH_no_inferential<span class="ot">&lt;-</span>dataset_analysis<span class="sc">%&gt;%</span></span>
<span id="cb47-52"><a href="#cb47-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(proxy_decision <span class="sc">==</span> <span class="st">"include"</span>)<span class="sc">%&gt;%</span> <span class="fu">filter</span>(SMDH_sign <span class="sc">!=</span> <span class="st">"NA"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
<section id="overall-effect-intercept-only-meta-analytic-model" class="level1">
<h1>Overall effect : Intercept-only meta-analytic model</h1>
<p><em>“Does adding green material to the nest have an adaptive function across the bird species that perform this behaviour?”</em></p>
<section id="using-lnrr" class="level2">
<h2 class="anchored" data-anchor-id="using-lnrr">using lnRR</h2>
<div class="cell-container"><div class="cell-decorator"><pre>In [6]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>VCV_lnRR<span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> lnRR_variance,</span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_lnRR,</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a>intercept_lnRR<span class="ot">&lt;-</span> <span class="fu">rma.mv</span>(<span class="at">yi =</span> lnRR_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_lnRR, <span class="co"># specify lnRR's sampling variance-covariance matrix;</span></span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb48-14"><a href="#cb48-14" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb48-15"><a href="#cb48-15" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb48-16"><a href="#cb48-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb48-17"><a href="#cb48-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb48-18"><a href="#cb48-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb48-19"><a href="#cb48-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> dataset_lnRR)</span>
<span id="cb48-20"><a href="#cb48-20" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the model</span></span>
<span id="cb48-21"><a href="#cb48-21" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(intercept_lnRR, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/intercept_lnRR.rds"</span>))</span>
<span id="cb48-22"><a href="#cb48-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-23"><a href="#cb48-23" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the dataset for this model</span></span>
<span id="cb48-24"><a href="#cb48-24" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(dataset_lnRR,here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/04_data_analysis/intercept_lnRR.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<section id="exploring-heterogeneity-in-intercept-only-model" class="level4">
<h4 class="anchored" data-anchor-id="exploring-heterogeneity-in-intercept-only-model">Exploring heterogeneity in intercept only model</h4>
<p>This is the function from Yefeng’s Github for interpreting heterogeneity. I have to download the data used for this function to be able to use it.</p>
<p>I am using the values from the table directly..</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [7]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ## Function is from the Pluralistic approach to heterogeneity from Yefeng's Pre-print repository from GitHub</span></span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a><span class="co"># #-----------------------heterogeneity interpretation-----------------------#</span></span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a><span class="co"># het_interpret &lt;- function(observed_value, het_type, es_type, data) {</span></span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a><span class="co">#   if (!het_type %in% c("I2", "CVH", "M", "sigma2", "V_bar")) {</span></span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a><span class="co">#     stop("Invalid heterogeneity type. Choose from 'I2', 'CVH', 'M', 'sigma2', or 'V_bar'.")</span></span>
<span id="cb49-8"><a href="#cb49-8" aria-hidden="true" tabindex="-1"></a><span class="co">#   }</span></span>
<span id="cb49-9"><a href="#cb49-9" aria-hidden="true" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb49-10"><a href="#cb49-10" aria-hidden="true" tabindex="-1"></a><span class="co">#   if (!es_type %in% unique(data$es.type)) {</span></span>
<span id="cb49-11"><a href="#cb49-11" aria-hidden="true" tabindex="-1"></a><span class="co">#     stop("Invalid effect size type. Check the levels of 'es.type'.")</span></span>
<span id="cb49-12"><a href="#cb49-12" aria-hidden="true" tabindex="-1"></a><span class="co">#   }</span></span>
<span id="cb49-13"><a href="#cb49-13" aria-hidden="true" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb49-14"><a href="#cb49-14" aria-hidden="true" tabindex="-1"></a><span class="co">#   filtered_data &lt;- data %&gt;% filter(es.type == es_type)</span></span>
<span id="cb49-15"><a href="#cb49-15" aria-hidden="true" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb49-16"><a href="#cb49-16" aria-hidden="true" tabindex="-1"></a><span class="co">#   if (nrow(filtered_data) == 0) {</span></span>
<span id="cb49-17"><a href="#cb49-17" aria-hidden="true" tabindex="-1"></a><span class="co">#     stop("No data available for the selected effect size type.")</span></span>
<span id="cb49-18"><a href="#cb49-18" aria-hidden="true" tabindex="-1"></a><span class="co">#   }</span></span>
<span id="cb49-19"><a href="#cb49-19" aria-hidden="true" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb49-20"><a href="#cb49-20" aria-hidden="true" tabindex="-1"></a><span class="co">#   het_values &lt;- filtered_data[[het_type]]</span></span>
<span id="cb49-21"><a href="#cb49-21" aria-hidden="true" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb49-22"><a href="#cb49-22" aria-hidden="true" tabindex="-1"></a><span class="co">#   percentiles &lt;- quantile(het_values, probs = seq(0, 1, by = 0.05), na.rm = TRUE)</span></span>
<span id="cb49-23"><a href="#cb49-23" aria-hidden="true" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb49-24"><a href="#cb49-24" aria-hidden="true" tabindex="-1"></a><span class="co">#   lower_bound &lt;- max(percentiles[percentiles &lt;= observed_value], na.rm = TRUE)</span></span>
<span id="cb49-25"><a href="#cb49-25" aria-hidden="true" tabindex="-1"></a><span class="co">#   upper_bound &lt;- min(percentiles[percentiles &gt;= observed_value], na.rm = TRUE)</span></span>
<span id="cb49-26"><a href="#cb49-26" aria-hidden="true" tabindex="-1"></a><span class="co">#   lower_percentile &lt;- names(percentiles)[percentiles == lower_bound]</span></span>
<span id="cb49-27"><a href="#cb49-27" aria-hidden="true" tabindex="-1"></a><span class="co">#   upper_percentile &lt;- names(percentiles)[percentiles == upper_bound]</span></span>
<span id="cb49-28"><a href="#cb49-28" aria-hidden="true" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb49-29"><a href="#cb49-29" aria-hidden="true" tabindex="-1"></a><span class="co">#   percentile_range &lt;- paste0(lower_percentile, "-", upper_percentile, "th percentile")</span></span>
<span id="cb49-30"><a href="#cb49-30" aria-hidden="true" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb49-31"><a href="#cb49-31" aria-hidden="true" tabindex="-1"></a><span class="co">#   # return the results</span></span>
<span id="cb49-32"><a href="#cb49-32" aria-hidden="true" tabindex="-1"></a><span class="co">#   return(tibble::tibble(</span></span>
<span id="cb49-33"><a href="#cb49-33" aria-hidden="true" tabindex="-1"></a><span class="co">#     observed_value = observed_value,</span></span>
<span id="cb49-34"><a href="#cb49-34" aria-hidden="true" tabindex="-1"></a><span class="co">#     het_type = het_type,</span></span>
<span id="cb49-35"><a href="#cb49-35" aria-hidden="true" tabindex="-1"></a><span class="co">#     es_type = es_type,</span></span>
<span id="cb49-36"><a href="#cb49-36" aria-hidden="true" tabindex="-1"></a><span class="co">#     percentile_range = percentile_range</span></span>
<span id="cb49-37"><a href="#cb49-37" aria-hidden="true" tabindex="-1"></a><span class="co">#   ))</span></span>
<span id="cb49-38"><a href="#cb49-38" aria-hidden="true" tabindex="-1"></a><span class="co"># }</span></span>
<span id="cb49-39"><a href="#cb49-39" aria-hidden="true" tabindex="-1"></a><span class="co"># x&lt;-het_interpret(observed_value= I2_main_model_vector[1], </span></span>
<span id="cb49-40"><a href="#cb49-40" aria-hidden="true" tabindex="-1"></a><span class="co">#                  het_type = "I2", </span></span>
<span id="cb49-41"><a href="#cb49-41" aria-hidden="true" tabindex="-1"></a><span class="co">#                  es_type = "lnRR_sign", </span></span>
<span id="cb49-42"><a href="#cb49-42" aria-hidden="true" tabindex="-1"></a><span class="co">#                  data= intercept_lnRR)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<p>Let us look at the heterogeneity in lnRR intercept-only model</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [8]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Then also:</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="co"># # Typical sampling variance (which captures the statistical noise of the data, </span></span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a><span class="co"># # but is rarely reported in the current meta-analytic practice):</span></span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a><span class="fu">sigma2_v</span>(intercept_lnRR)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.0003392512</code></pre>
</div>
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="co"># To calculate heterogeneity, these are the functions from orchaRd:</span></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a><span class="co"># # I2, CV and M</span></span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>sigma2_main_model_vector <span class="ot">&lt;-</span> <span class="fu">sum</span>(intercept_lnRR<span class="sc">$</span>sigma2)</span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a><span class="co"># 0.11, 0.27, and 0.57 for lnRR</span></span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a>I2_main_model_vector <span class="ot">&lt;-</span> orchaRd<span class="sc">::</span><span class="fu">i2_ml</span>(intercept_lnRR)</span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-9"><a href="#cb52-9" aria-hidden="true" tabindex="-1"></a><span class="co"># 0.88, 0.95, and 0.99 for lnRR</span></span>
<span id="cb52-10"><a href="#cb52-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-11"><a href="#cb52-11" aria-hidden="true" tabindex="-1"></a>CV2_main_model_vector <span class="ot">&lt;-</span> orchaRd<span class="sc">::</span><span class="fu">cvh2_ml</span>(intercept_lnRR)</span>
<span id="cb52-12"><a href="#cb52-12" aria-hidden="true" tabindex="-1"></a>CV1_main_model_vector <span class="ot">&lt;-</span> orchaRd<span class="sc">::</span><span class="fu">cvh1_ml</span>(intercept_lnRR)</span>
<span id="cb52-13"><a href="#cb52-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-14"><a href="#cb52-14" aria-hidden="true" tabindex="-1"></a><span class="co"># CVH2 (CVH1) - 1.36 (1.16), 3.76 (1.94), and 12.1 (3.48) for lnRR</span></span>
<span id="cb52-15"><a href="#cb52-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-16"><a href="#cb52-16" aria-hidden="true" tabindex="-1"></a>M2_main_model_vector <span class="ot">&lt;-</span> orchaRd<span class="sc">::</span><span class="fu">m2_ml</span>(intercept_lnRR)</span>
<span id="cb52-17"><a href="#cb52-17" aria-hidden="true" tabindex="-1"></a>M1_main_model_vector <span class="ot">&lt;-</span> orchaRd<span class="sc">::</span><span class="fu">m1_ml</span>(intercept_lnRR)</span>
<span id="cb52-18"><a href="#cb52-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-19"><a href="#cb52-19" aria-hidden="true" tabindex="-1"></a><span class="co"># M2(M1) - 0.58 (0.54), 0.79 (0.66), and 0.78 for lnRR;</span></span>
<span id="cb52-20"><a href="#cb52-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-21"><a href="#cb52-21" aria-hidden="true" tabindex="-1"></a><span class="co"># # # Total unstandardized raw variance (i.e. total heterogeneity)</span></span>
<span id="cb52-22"><a href="#cb52-22" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(<span class="fu">sum</span>(intercept_lnRR<span class="sc">$</span>sigma2), <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.023</code></pre>
</div>
</div></div>
</section>
</section>
<section id="using-smdh" class="level2">
<h2 class="anchored" data-anchor-id="using-smdh">using SMDH</h2>
<p>We are only running these analysis for the fitness proxy labelled “included”. Some proxies have been flagged and we will run them in a sensitivity analysis only</p>
<p>The sampling variances will be fitted as variance–covariance matrices assuming a 0.5 correlation (ρs) between sampling variances from the same paper_ID (Noble et al.&nbsp;2017).</p>
<p>The MLMA using SMDH as the effect size measure incorporate the imputed sampling variance</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [9]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>VCV_SMDH <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> SMDH_variance,</span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_SMDH,</span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb54-7"><a href="#cb54-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-8"><a href="#cb54-8" aria-hidden="true" tabindex="-1"></a>intercept_SMDH<span class="ot">&lt;-</span> <span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify SMDH as the effect size measure;</span></span>
<span id="cb54-9"><a href="#cb54-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_SMDH, <span class="co"># specify SMDH's sampling variance-covariance matrix;</span></span>
<span id="cb54-10"><a href="#cb54-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb54-11"><a href="#cb54-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb54-12"><a href="#cb54-12" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb54-13"><a href="#cb54-13" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb54-14"><a href="#cb54-14" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb54-15"><a href="#cb54-15" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb54-16"><a href="#cb54-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb54-17"><a href="#cb54-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb54-18"><a href="#cb54-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb54-19"><a href="#cb54-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> dataset_SMDH)</span>
<span id="cb54-20"><a href="#cb54-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-21"><a href="#cb54-21" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the model</span></span>
<span id="cb54-22"><a href="#cb54-22" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(intercept_SMDH, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/intercept_SMDH.rds"</span>))</span>
<span id="cb54-23"><a href="#cb54-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-24"><a href="#cb54-24" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the dataset for this model</span></span>
<span id="cb54-25"><a href="#cb54-25" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(dataset_SMDH,here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/04_data_analysis/intercept_SMDH.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
<section id="smdh-heterogeneity" class="level2">
<h2 class="anchored" data-anchor-id="smdh-heterogeneity">SMDH-heterogeneity</h2>
<p>Now looking at the intercept only model</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [10]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(intercept_SMDH)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Multivariate Meta-Analysis Model (k = 253; method: REML)

   logLik   Deviance        AIC        BIC       AICc   
-324.4948   648.9896   660.9896   682.1662   661.3325   

Variance Components:

            estim    sqrt  nlvls  fixed          factor 
sigma^2.1  0.0000  0.0008     26     no        paper_ID 
sigma^2.2  0.2578  0.5077    253     no  Observation_ID 
sigma^2.3  0.0000  0.0013     47     no  group_ID_coded 
sigma^2.4  0.0000  0.0004     16     no   population_ID 
sigma^2.5  0.0000  0.0001      7     no    bird_species 

Test for Heterogeneity:
Q(df = 252) = 1114.7203, p-val &lt; .0001

Model Results:

estimate      se    tval   df    pval   ci.lb   ci.ub     
  0.1793  0.0665  2.6948  252  0.0075  0.0483  0.3104  ** 

---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">predict</span>(intercept_SMDH)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   pred     se  ci.lb  ci.ub   pi.lb  pi.ub 
 0.1793 0.0665 0.0483 0.3104 -0.8292 1.1879 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Then also:</span></span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a><span class="co"># # Typical sampling variance (which captures the statistical noise of the data, </span></span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a><span class="co"># # but is rarely reported in the current meta-analytic practice):</span></span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a><span class="fu">sigma2_v</span>(intercept_SMDH)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.1319841</code></pre>
</div>
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="co"># To calculate heterogeneity, these are the functions from orchaRd:</span></span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a><span class="co"># # I2, CV and M</span></span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a>sigma2_main_model_vector <span class="ot">&lt;-</span> <span class="fu">sum</span>(intercept_SMDH<span class="sc">$</span>sigma2)</span>
<span id="cb61-4"><a href="#cb61-4" aria-hidden="true" tabindex="-1"></a><span class="co"># 0.54, 1.25, and 3.03 for SMD;</span></span>
<span id="cb61-5"><a href="#cb61-5" aria-hidden="true" tabindex="-1"></a>I2_main_model_vector <span class="ot">&lt;-</span> orchaRd<span class="sc">::</span><span class="fu">i2_ml</span>(intercept_SMDH)</span>
<span id="cb61-6"><a href="#cb61-6" aria-hidden="true" tabindex="-1"></a><span class="co"># 0.78, 0.89, and 0.96 for SMD</span></span>
<span id="cb61-7"><a href="#cb61-7" aria-hidden="true" tabindex="-1"></a><span class="co"># 0.88, 0.95, and 0.99 for lnRR</span></span>
<span id="cb61-8"><a href="#cb61-8" aria-hidden="true" tabindex="-1"></a>CV2_main_model_vector <span class="ot">&lt;-</span> orchaRd<span class="sc">::</span><span class="fu">cvh2_ml</span>(intercept_SMDH)</span>
<span id="cb61-9"><a href="#cb61-9" aria-hidden="true" tabindex="-1"></a>CV1_main_model_vector <span class="ot">&lt;-</span> orchaRd<span class="sc">::</span><span class="fu">cvh1_ml</span>(intercept_SMDH)</span>
<span id="cb61-10"><a href="#cb61-10" aria-hidden="true" tabindex="-1"></a><span class="co"># CVH2 (CVH1) - 1.36 (1.16), 3.76 (1.94), and 12.1 (3.48) for lnRR</span></span>
<span id="cb61-11"><a href="#cb61-11" aria-hidden="true" tabindex="-1"></a>M2_main_model_vector <span class="ot">&lt;-</span> orchaRd<span class="sc">::</span><span class="fu">m2_ml</span>(intercept_SMDH)</span>
<span id="cb61-12"><a href="#cb61-12" aria-hidden="true" tabindex="-1"></a>M1_main_model_vector <span class="ot">&lt;-</span> orchaRd<span class="sc">::</span><span class="fu">m1_ml</span>(intercept_SMDH)</span>
<span id="cb61-13"><a href="#cb61-13" aria-hidden="true" tabindex="-1"></a><span class="co"># M2 (M1) - 0.58 (0.54), 0.79 (0.66), and 0.78 for lnRR</span></span>
<span id="cb61-14"><a href="#cb61-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-15"><a href="#cb61-15" aria-hidden="true" tabindex="-1"></a><span class="co"># # # Total unstandardized raw variance (i.e. total heterogeneity)</span></span>
<span id="cb61-16"><a href="#cb61-16" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(<span class="fu">sum</span>(intercept_SMDH<span class="sc">$</span>sigma2), <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.258</code></pre>
</div>
</div></div>
</section>
<section id="bivariate-meta-analytical-model" class="level2">
<h2 class="anchored" data-anchor-id="bivariate-meta-analytical-model">Bivariate Meta-analytical model</h2>
<p>For the overall effect size (i.e., the meta-analytical mean), we will run the bivariate multilevel meta-analytic model suggested by Yang et al.&nbsp;(2024), which models both lnRR and SMD(H) simultaneously (i.e., including them both as response variables and estimating their correlation; random).</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [11]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a>dataset_bivariate <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">paper_ID =</span> <span class="fu">c</span>(dataset_lnRR<span class="sc">$</span>paper_ID, dataset_SMDH<span class="sc">$</span>paper_ID),</span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>                      <span class="at">Observation_ID =</span> <span class="fu">c</span>(dataset_lnRR<span class="sc">$</span>Observation_ID, dataset_SMDH<span class="sc">$</span>Observation_ID),</span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">group_ID_coded =</span> <span class="fu">c</span>(dataset_lnRR<span class="sc">$</span>group_ID_coded, dataset_SMDH<span class="sc">$</span>group_ID_coded),</span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">population_ID =</span> <span class="fu">c</span>(dataset_lnRR<span class="sc">$</span>population_ID, dataset_SMDH<span class="sc">$</span>population_ID),</span>
<span id="cb63-5"><a href="#cb63-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">bird_species =</span> <span class="fu">c</span>(dataset_lnRR<span class="sc">$</span>bird_species, dataset_SMDH<span class="sc">$</span>bird_species),</span>
<span id="cb63-6"><a href="#cb63-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">ES =</span> <span class="fu">c</span>(dataset_lnRR<span class="sc">$</span>lnRR_sign, dataset_SMDH<span class="sc">$</span>SMDH_sign),</span>
<span id="cb63-7"><a href="#cb63-7" aria-hidden="true" tabindex="-1"></a>                      <span class="at">V_ES =</span> <span class="fu">c</span>(dataset_lnRR<span class="sc">$</span>lnRR_variance, dataset_SMDH<span class="sc">$</span>SMDH_variance),</span>
<span id="cb63-8"><a href="#cb63-8" aria-hidden="true" tabindex="-1"></a>                      <span class="at">ES_measure =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"lnRR"</span>, <span class="fu">nrow</span>(dataset_lnRR)), <span class="fu">rep</span>(<span class="st">"SMDH"</span>, <span class="fu">nrow</span>(dataset_SMDH)))</span>
<span id="cb63-9"><a href="#cb63-9" aria-hidden="true" tabindex="-1"></a>                      )</span>
<span id="cb63-10"><a href="#cb63-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-11"><a href="#cb63-11" aria-hidden="true" tabindex="-1"></a>VCV_bivariate <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> V_ES, <span class="co"># sampling variances of lnRR and SMD that are correlated within the same study;</span></span>
<span id="cb63-12"><a href="#cb63-12" aria-hidden="true" tabindex="-1"></a>             <span class="at">cluster =</span> paper_ID, <span class="co"># study identity;</span></span>
<span id="cb63-13"><a href="#cb63-13" aria-hidden="true" tabindex="-1"></a>             <span class="at">type =</span> ES_measure, <span class="co"># different types of effect size measures underlying the observed effect sizes;</span></span>
<span id="cb63-14"><a href="#cb63-14" aria-hidden="true" tabindex="-1"></a>             <span class="at">data =</span> dataset_bivariate, <span class="co"># the long format data frame;</span></span>
<span id="cb63-15"><a href="#cb63-15" aria-hidden="true" tabindex="-1"></a>             <span class="at">rho =</span> <span class="fl">0.2</span>, <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5.</span></span>
<span id="cb63-16"><a href="#cb63-16" aria-hidden="true" tabindex="-1"></a>             <span class="at">nearpd =</span> <span class="cn">TRUE</span> <span class="co"># specify whether impose the non positive definite VCV matrix (not invertible) to the nearest positive semi-definite matrix </span></span>
<span id="cb63-17"><a href="#cb63-17" aria-hidden="true" tabindex="-1"></a>             )</span>
<span id="cb63-18"><a href="#cb63-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-19"><a href="#cb63-19" aria-hidden="true" tabindex="-1"></a>Bivariate_model <span class="ot">&lt;-</span> <span class="fu">rma.mv</span>(<span class="at">yi =</span> ES, <span class="co"># specify the effect size estimate (the variable ES in our case);</span></span>
<span id="cb63-20"><a href="#cb63-20" aria-hidden="true" tabindex="-1"></a>                <span class="at">V =</span> VCV_bivariate, <span class="co"># specify the imputed variance-covariance matrix;</span></span>
<span id="cb63-21"><a href="#cb63-21" aria-hidden="true" tabindex="-1"></a>                <span class="at">mods =</span> <span class="sc">~</span> ES_measure <span class="sc">-</span> <span class="dv">1</span>, <span class="co"># specify the variable "ES_measure" indicating the types of effect size measures;</span></span>
<span id="cb63-22"><a href="#cb63-22" aria-hidden="true" tabindex="-1"></a>                <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> ES_measure <span class="sc">|</span> paper_ID, </span>
<span id="cb63-23"><a href="#cb63-23" aria-hidden="true" tabindex="-1"></a>                              <span class="sc">~</span> ES_measure <span class="sc">|</span> Observation_ID,</span>
<span id="cb63-24"><a href="#cb63-24" aria-hidden="true" tabindex="-1"></a>                              <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb63-25"><a href="#cb63-25" aria-hidden="true" tabindex="-1"></a>                              <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb63-26"><a href="#cb63-26" aria-hidden="true" tabindex="-1"></a>                              <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species</span>
<span id="cb63-27"><a href="#cb63-27" aria-hidden="true" tabindex="-1"></a>                              ), <span class="co"># add correlated random effects corresponding to the lnRR and SMD parameters in the same study;</span></span>
<span id="cb63-28"><a href="#cb63-28" aria-hidden="true" tabindex="-1"></a>                <span class="at">struct =</span> <span class="st">"UN"</span>, <span class="co"># impose an unstructured variance-covariance of the study-specific random effects;</span></span>
<span id="cb63-29"><a href="#cb63-29" aria-hidden="true" tabindex="-1"></a>                <span class="at">test =</span> <span class="st">"t"</span>, <span class="co"># t distribution is specified to test the overall effect against the null hypothesis and construct confidence intervals;</span></span>
<span id="cb63-30"><a href="#cb63-30" aria-hidden="true" tabindex="-1"></a>                <span class="at">method =</span> <span class="st">"REML"</span>, <span class="co"># restricted likelihood maximum is assigned as the estimator for variance components as suggested;</span></span>
<span id="cb63-31"><a href="#cb63-31" aria-hidden="true" tabindex="-1"></a>                <span class="at">data =</span> dataset_bivariate, <span class="co"># the long format dataset</span></span>
<span id="cb63-32"><a href="#cb63-32" aria-hidden="true" tabindex="-1"></a>                <span class="at">sparse =</span> T</span>
<span id="cb63-33"><a href="#cb63-33" aria-hidden="true" tabindex="-1"></a>               )</span>
<span id="cb63-34"><a href="#cb63-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-35"><a href="#cb63-35" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the model</span></span>
<span id="cb63-36"><a href="#cb63-36" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(Bivariate_model, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/sensitivity_analysis/bivariate_model.rds"</span>))</span>
<span id="cb63-37"><a href="#cb63-37" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the dataset for this model</span></span>
<span id="cb63-38"><a href="#cb63-38" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(dataset_bivariate,here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/05_data_sensitivity_analysis/dataset_bivariate.csv"</span>))</span>
<span id="cb63-39"><a href="#cb63-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-40"><a href="#cb63-40" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(Bivariate_model)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Multivariate Meta-Analysis Model (k = 500; method: REML)

   logLik   Deviance        AIC        BIC       AICc   
-481.2569   962.5139   984.5139  1030.8305   985.0571   

Variance Components:

            estim    sqrt  nlvls  fixed          factor 
sigma^2.1  0.0000  0.0000     47     no  group_ID_coded 
sigma^2.2  0.0000  0.0000     16     no   population_ID 
sigma^2.3  0.0000  0.0000      7     no    bird_species 

outer factor: paper_ID   (nlvls = 26)
inner factor: ES_measure (nlvls = 2)

            estim    sqrt  k.lvl  fixed  level 
tau^2.1    0.0008  0.0283     26     no   lnRR 
tau^2.2    0.1645  0.4055     26     no   SMDH 

      rho.lnRR  rho.SMDH    lnRR  SMDH 
lnRR         1                 -    26 
SMDH   -1.0000         1      no     - 

outer factor: Observation_ID (nlvls = 262)
inner factor: ES_measure     (nlvls = 2)

              estim    sqrt  k.lvl  fixed  level 
gamma^2.1    0.1258  0.3546    247     no   lnRR 
gamma^2.2    1.2645  1.1245    253     no   SMDH 

      phi.lnRR  phi.SMDH    lnRR  SMDH 
lnRR         1    0.5184       -   238 
SMDH    0.5184         1      no     - 

Test for Residual Heterogeneity:
QE(df = 498) = 15908426971.3013, p-val &lt; .0001

Test of Moderators (coefficients 1:2):
F(df1 = 2, df2 = 498) = 1.9036, p-val = 0.1501

Model Results:

                estimate      se     tval   df    pval    ci.lb   ci.ub    
ES_measurelnRR   -0.0132  0.0298  -0.4435  498  0.6576  -0.0718  0.0454    
ES_measureSMDH    0.2544  0.1383   1.8397  498  0.0664  -0.0173  0.5261  . 

---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [12]:</pre></div><div class="cell" data-warning="false">
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(ellipse)</span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a>ab.l <span class="ot">&lt;-</span> <span class="fu">matreg</span>(<span class="at">y=</span><span class="dv">2</span>, <span class="at">x=</span><span class="dv">1</span>, <span class="at">R=</span>Bivariate_model<span class="sc">$</span>G, <span class="at">cov=</span><span class="cn">TRUE</span>, <span class="at">means=</span><span class="fu">coef</span>(Bivariate_model), <span class="at">n=</span>Bivariate_model<span class="sc">$</span>g.levels.comb.k) <span class="co"># fit regression model</span></span>
<span id="cb65-3"><a href="#cb65-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-4"><a href="#cb65-4" aria-hidden="true" tabindex="-1"></a>xy <span class="ot">&lt;-</span> <span class="fu">ellipse</span>(Bivariate_model<span class="sc">$</span>G, <span class="at">centre=</span><span class="fu">coef</span>(Bivariate_model), <span class="at">level=</span><span class="fl">0.95</span>) <span class="co"># get 95% coverage region</span></span>
<span id="cb65-5"><a href="#cb65-5" aria-hidden="true" tabindex="-1"></a>ellipse_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">x =</span> xy[, <span class="dv">1</span>], <span class="at">y =</span> xy[, <span class="dv">2</span>]) <span class="co"># convert it int dataframe so that we can use ggplot to make a figure</span></span>
<span id="cb65-6"><a href="#cb65-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-7"><a href="#cb65-7" aria-hidden="true" tabindex="-1"></a><span class="fu">pivot_wider</span>(dataset_bivariate, <span class="at">names_from =</span> ES_measure, <span class="at">values_from =</span> <span class="fu">c</span>(ES, V_ES)) <span class="sc">%&gt;%</span></span>
<span id="cb65-8"><a href="#cb65-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span> </span>
<span id="cb65-9"><a href="#cb65-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> ES_lnRR, <span class="at">y =</span> ES_SMDH, <span class="at">size =</span> <span class="dv">1</span><span class="sc">/</span><span class="fu">sqrt</span>(V_ES_lnRR)), <span class="at">color =</span> <span class="st">"#1B9E77"</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span> </span>
<span id="cb65-10"><a href="#cb65-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">coef</span>(Bivariate_model)[<span class="dv">1</span>], <span class="at">y =</span> <span class="fu">coef</span>(Bivariate_model)[<span class="dv">2</span>]), <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span> </span>
<span id="cb65-11"><a href="#cb65-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_abline</span>(<span class="at">intercept =</span> ab.l<span class="sc">$</span>tab<span class="sc">$</span>beta[<span class="dv">1</span>], <span class="at">slope =</span> ab.l<span class="sc">$</span>tab<span class="sc">$</span>beta[<span class="dv">2</span>]) <span class="sc">+</span></span>
<span id="cb65-12"><a href="#cb65-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_path</span>(<span class="at">data =</span> ellipse_df, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y), <span class="at">color =</span> <span class="st">"gray50"</span>) <span class="sc">+</span></span>
<span id="cb65-13"><a href="#cb65-13" aria-hidden="true" tabindex="-1"></a>  <span class="co">#scale_x_continuous(limits = c(-2.5, 2.5)) + </span></span>
<span id="cb65-14"><a href="#cb65-14" aria-hidden="true" tabindex="-1"></a>  <span class="co">#scale_y_continuous(limits = c(-2.5, 2.5)) + </span></span>
<span id="cb65-15"><a href="#cb65-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">size =</span> <span class="st">"none"</span>) <span class="sc">+</span> </span>
<span id="cb65-16"><a href="#cb65-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"lnRR"</span>, <span class="at">y =</span> <span class="st">"SMD"</span>) <span class="sc">+</span> </span>
<span id="cb65-17"><a href="#cb65-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in geom_point(aes(x = coef(Bivariate_model)[1], y = coef(Bivariate_model)[2]), : All aesthetics have length 1, but the data has 262 rows.
ℹ Please consider using `annotate()` or provide this layer with data containing
  a single row.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 24 rows containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
</div>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="07_data_analysis_files/figure-html/unnamed-chunk-7-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-4"><img src="07_data_analysis_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
</div></div>
<p>There is a very high negative correlation between the two effect sizes. This probably is an artifact of the data and not a true correlation. If you look at the model summary, the <span class="math inline">\(tau^2\)</span> for lnRR, it is extremely small 0.0008, i.e., there is very little variance. This could also be stemming from the fact that the log values in the dataset could be causing this. For SMDH on the other hand, the <span class="math inline">\(tau^2\)</span> is 0.1681 that is much larger. This difference in the variance estimate itself could be causing this artifact of the model. I will only go ahead to present this model as a senstivity analysis because I am not completely convinced by it’s robustness.</p>
</section>
</section>
<section id="sensitivity-analysis" class="level1">
<h1>Sensitivity Analysis</h1>
<section id="with-published-data-only" class="level3">
<h3 class="anchored" data-anchor-id="with-published-data-only">With published data only</h3>
<div class="cell-container"><div class="cell-decorator"><pre>In [13]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="co">#----------------------------------------------------------------------------</span></span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a><span class="co"># for lnRR as Effect Size</span></span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a><span class="co">#----------------------------------------------------------------------------</span></span>
<span id="cb68-4"><a href="#cb68-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-5"><a href="#cb68-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Preparing new variance-covariance matrix</span></span>
<span id="cb68-6"><a href="#cb68-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-7"><a href="#cb68-7" aria-hidden="true" tabindex="-1"></a>VCV_lnRR_published <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> lnRR_variance,</span>
<span id="cb68-8"><a href="#cb68-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb68-9"><a href="#cb68-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb68-10"><a href="#cb68-10" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_lnRR_published,</span>
<span id="cb68-11"><a href="#cb68-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb68-12"><a href="#cb68-12" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb68-13"><a href="#cb68-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-14"><a href="#cb68-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-15"><a href="#cb68-15" aria-hidden="true" tabindex="-1"></a><span class="do">## Running intercept only model</span></span>
<span id="cb68-16"><a href="#cb68-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-17"><a href="#cb68-17" aria-hidden="true" tabindex="-1"></a>SA_lnRR_published<span class="ot">&lt;-</span> <span class="fu">rma.mv</span>(<span class="at">yi =</span> lnRR_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb68-18"><a href="#cb68-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_lnRR_published, <span class="co"># specify lnRR's sampling variance-covariance matrix;</span></span>
<span id="cb68-19"><a href="#cb68-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb68-20"><a href="#cb68-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb68-21"><a href="#cb68-21" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb68-22"><a href="#cb68-22" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb68-23"><a href="#cb68-23" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb68-24"><a href="#cb68-24" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb68-25"><a href="#cb68-25" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb68-26"><a href="#cb68-26" aria-hidden="true" tabindex="-1"></a>                    <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb68-27"><a href="#cb68-27" aria-hidden="true" tabindex="-1"></a>                    <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb68-28"><a href="#cb68-28" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> dataset_lnRR_published)</span>
<span id="cb68-29"><a href="#cb68-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-30"><a href="#cb68-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-31"><a href="#cb68-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the model</span></span>
<span id="cb68-32"><a href="#cb68-32" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SA_lnRR_published, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/sensitivity_analysis/SA_lnRR_published.rds"</span>))</span>
<span id="cb68-33"><a href="#cb68-33" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the dataset for this model</span></span>
<span id="cb68-34"><a href="#cb68-34" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(dataset_lnRR_published,here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/05_data_sensitivity_analysis/dataset_lnRR_published.csv"</span>))</span>
<span id="cb68-35"><a href="#cb68-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-36"><a href="#cb68-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-37"><a href="#cb68-37" aria-hidden="true" tabindex="-1"></a><span class="co"># What is the effect now?</span></span>
<span id="cb68-38"><a href="#cb68-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-39"><a href="#cb68-39" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(SA_lnRR_published)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Multivariate Meta-Analysis Model (k = 230; method: REML)

  logLik  Deviance       AIC       BIC      AICc   
-22.7153   45.4307   57.4307   78.0330   57.8090   

Variance Components:

            estim    sqrt  nlvls  fixed          factor 
sigma^2.1  0.0000  0.0001     24     no        paper_ID 
sigma^2.2  0.0244  0.1564    230     no  Observation_ID 
sigma^2.3  0.0000  0.0002     44     no  group_ID_coded 
sigma^2.4  0.0000  0.0000     14     no   population_ID 
sigma^2.5  0.0000  0.0000      6     no    bird_species 

Test for Heterogeneity:
Q(df = 229) = 2318.5698, p-val &lt; .0001

Model Results:

estimate      se    tval   df    pval    ci.lb   ci.ub    
  0.0168  0.0161  1.0455  229  0.2969  -0.0149  0.0486    

---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="fu">predict</span>(SA_lnRR_published)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   pred     se   ci.lb  ci.ub   pi.lb  pi.ub 
 0.0168 0.0161 -0.0149 0.0486 -0.2929 0.3265 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="co">#----------------------------------------------------------------------------</span></span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a><span class="co"># for SMDH as Effect Size</span></span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a><span class="co">#----------------------------------------------------------------------------</span></span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-5"><a href="#cb72-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Preparing new variance-covariance matrix</span></span>
<span id="cb72-6"><a href="#cb72-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-7"><a href="#cb72-7" aria-hidden="true" tabindex="-1"></a>VCV_SMDH_published <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> SMDH_variance,</span>
<span id="cb72-8"><a href="#cb72-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb72-9"><a href="#cb72-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb72-10"><a href="#cb72-10" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_SMDH_published,</span>
<span id="cb72-11"><a href="#cb72-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb72-12"><a href="#cb72-12" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb72-13"><a href="#cb72-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-14"><a href="#cb72-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-15"><a href="#cb72-15" aria-hidden="true" tabindex="-1"></a><span class="do">## Running intercept only model</span></span>
<span id="cb72-16"><a href="#cb72-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-17"><a href="#cb72-17" aria-hidden="true" tabindex="-1"></a>SA_SMDH_published<span class="ot">&lt;-</span> <span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb72-18"><a href="#cb72-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_SMDH_published, <span class="co"># specify lnRR's sampling variance-covariance matrix;</span></span>
<span id="cb72-19"><a href="#cb72-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb72-20"><a href="#cb72-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb72-21"><a href="#cb72-21" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb72-22"><a href="#cb72-22" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb72-23"><a href="#cb72-23" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb72-24"><a href="#cb72-24" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb72-25"><a href="#cb72-25" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb72-26"><a href="#cb72-26" aria-hidden="true" tabindex="-1"></a>                    <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb72-27"><a href="#cb72-27" aria-hidden="true" tabindex="-1"></a>                    <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb72-28"><a href="#cb72-28" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> dataset_SMDH_published)</span>
<span id="cb72-29"><a href="#cb72-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-30"><a href="#cb72-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-31"><a href="#cb72-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the model</span></span>
<span id="cb72-32"><a href="#cb72-32" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SA_SMDH_published, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/sensitivity_analysis/SA_SMDH_published.rds"</span>))</span>
<span id="cb72-33"><a href="#cb72-33" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the dataset for this model</span></span>
<span id="cb72-34"><a href="#cb72-34" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(dataset_SMDH_published,here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/05_data_sensitivity_analysis/dataset_SMDH_published.csv"</span>))</span>
<span id="cb72-35"><a href="#cb72-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-36"><a href="#cb72-36" aria-hidden="true" tabindex="-1"></a><span class="co"># What is the effect now?</span></span>
<span id="cb72-37"><a href="#cb72-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-38"><a href="#cb72-38" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(SA_SMDH_published)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Multivariate Meta-Analysis Model (k = 235; method: REML)

   logLik   Deviance        AIC        BIC       AICc   
-308.4661   616.9322   628.9322   649.6641   629.3022   

Variance Components:

            estim    sqrt  nlvls  fixed          factor 
sigma^2.1  0.0000  0.0010     24     no        paper_ID 
sigma^2.2  0.2740  0.5235    235     no  Observation_ID 
sigma^2.3  0.0000  0.0014     45     no  group_ID_coded 
sigma^2.4  0.0000  0.0006     14     no   population_ID 
sigma^2.5  0.0000  0.0001      6     no    bird_species 

Test for Heterogeneity:
Q(df = 234) = 1092.0602, p-val &lt; .0001

Model Results:

estimate      se    tval   df    pval   ci.lb   ci.ub     
  0.1916  0.0688  2.7849  234  0.0058  0.0561  0.3271  ** 

---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">predict</span>(SA_SMDH_published)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   pred     se  ci.lb  ci.ub   pi.lb  pi.ub 
 0.1916 0.0688 0.0561 0.3271 -0.8486 1.2318 </code></pre>
</div>
</div></div>
</section>
<section id="with-flagged-proxies" class="level3">
<h3 class="anchored" data-anchor-id="with-flagged-proxies">With flagged proxies</h3>
<p>Some traits for which the relationship with fitness is unclear or highly debated in the literature</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [14]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="co">#----------------------------------------------------------------------------</span></span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a><span class="co"># for lnRR as Effect Size</span></span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a><span class="co">#----------------------------------------------------------------------------</span></span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-5"><a href="#cb76-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Preparing new variance-covariance matrix</span></span>
<span id="cb76-6"><a href="#cb76-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-7"><a href="#cb76-7" aria-hidden="true" tabindex="-1"></a>VCV_lnRR_flagged <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> lnRR_variance,</span>
<span id="cb76-8"><a href="#cb76-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb76-9"><a href="#cb76-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb76-10"><a href="#cb76-10" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> lnRR_con_flagged,</span>
<span id="cb76-11"><a href="#cb76-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb76-12"><a href="#cb76-12" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb76-13"><a href="#cb76-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-14"><a href="#cb76-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-15"><a href="#cb76-15" aria-hidden="true" tabindex="-1"></a><span class="do">## Running intercept only model</span></span>
<span id="cb76-16"><a href="#cb76-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-17"><a href="#cb76-17" aria-hidden="true" tabindex="-1"></a>SA_lnRR_flagged<span class="ot">&lt;-</span> <span class="fu">rma.mv</span>(<span class="at">yi =</span> lnRR_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb76-18"><a href="#cb76-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_lnRR_flagged, <span class="co"># specify lnRR's sampling variance-covariance matrix;</span></span>
<span id="cb76-19"><a href="#cb76-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb76-20"><a href="#cb76-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb76-21"><a href="#cb76-21" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb76-22"><a href="#cb76-22" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb76-23"><a href="#cb76-23" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb76-24"><a href="#cb76-24" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb76-25"><a href="#cb76-25" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb76-26"><a href="#cb76-26" aria-hidden="true" tabindex="-1"></a>                    <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb76-27"><a href="#cb76-27" aria-hidden="true" tabindex="-1"></a>                    <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb76-28"><a href="#cb76-28" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> lnRR_con_flagged)</span>
<span id="cb76-29"><a href="#cb76-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-30"><a href="#cb76-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-31"><a href="#cb76-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-32"><a href="#cb76-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the model</span></span>
<span id="cb76-33"><a href="#cb76-33" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SA_lnRR_flagged, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/sensitivity_analysis/SA_lnRR_flagged.rds"</span>))</span>
<span id="cb76-34"><a href="#cb76-34" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the dataset for this model</span></span>
<span id="cb76-35"><a href="#cb76-35" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(lnRR_con_flagged,here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/05_data_sensitivity_analysis/dataset_lnRR_con_flagged.csv"</span>))</span>
<span id="cb76-36"><a href="#cb76-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-37"><a href="#cb76-37" aria-hidden="true" tabindex="-1"></a><span class="co"># What is the effect now?</span></span>
<span id="cb76-38"><a href="#cb76-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-39"><a href="#cb76-39" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(SA_lnRR_flagged)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Multivariate Meta-Analysis Model (k = 268; method: REML)

  logLik  Deviance       AIC       BIC      AICc   
-25.3757   50.7514   62.7514   84.2749   63.0744   

Variance Components:

            estim    sqrt  nlvls  fixed          factor 
sigma^2.1  0.0000  0.0003     28     no        paper_ID 
sigma^2.2  0.0224  0.1495    268     no  Observation_ID 
sigma^2.3  0.0000  0.0001     52     no  group_ID_coded 
sigma^2.4  0.0000  0.0001     16     no   population_ID 
sigma^2.5  0.0000  0.0000      7     no    bird_species 

Test for Heterogeneity:
Q(df = 267) = 2460.7756, p-val &lt; .0001

Model Results:

estimate      se    tval   df    pval    ci.lb   ci.ub    
  0.0210  0.0147  1.4279  267  0.1545  -0.0080  0.0500    

---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">predict</span>(SA_lnRR_flagged)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   pred     se   ci.lb  ci.ub   pi.lb  pi.ub 
 0.0210 0.0147 -0.0080 0.0500 -0.2748 0.3169 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="co">#----------------------------------------------------------------------------</span></span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a><span class="co"># for SMDH as Effect Size</span></span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a><span class="co">#----------------------------------------------------------------------------</span></span>
<span id="cb80-4"><a href="#cb80-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-5"><a href="#cb80-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Preparing new variance-covariance matrix</span></span>
<span id="cb80-6"><a href="#cb80-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-7"><a href="#cb80-7" aria-hidden="true" tabindex="-1"></a>VCV_SMDH_flagged <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> SMDH_variance,</span>
<span id="cb80-8"><a href="#cb80-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb80-9"><a href="#cb80-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb80-10"><a href="#cb80-10" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> SMDH_con_flagged,</span>
<span id="cb80-11"><a href="#cb80-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb80-12"><a href="#cb80-12" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb80-13"><a href="#cb80-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-14"><a href="#cb80-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-15"><a href="#cb80-15" aria-hidden="true" tabindex="-1"></a><span class="do">## Running intercept only model</span></span>
<span id="cb80-16"><a href="#cb80-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-17"><a href="#cb80-17" aria-hidden="true" tabindex="-1"></a>SA_SMDH_flagged<span class="ot">&lt;-</span> <span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb80-18"><a href="#cb80-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_SMDH_flagged, <span class="co"># specify lnRR's sampling variance-covariance matrix;</span></span>
<span id="cb80-19"><a href="#cb80-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb80-20"><a href="#cb80-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb80-21"><a href="#cb80-21" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb80-22"><a href="#cb80-22" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb80-23"><a href="#cb80-23" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb80-24"><a href="#cb80-24" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb80-25"><a href="#cb80-25" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb80-26"><a href="#cb80-26" aria-hidden="true" tabindex="-1"></a>                    <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb80-27"><a href="#cb80-27" aria-hidden="true" tabindex="-1"></a>                    <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb80-28"><a href="#cb80-28" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> SMDH_con_flagged)</span>
<span id="cb80-29"><a href="#cb80-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-30"><a href="#cb80-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-31"><a href="#cb80-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the model</span></span>
<span id="cb80-32"><a href="#cb80-32" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SA_SMDH_flagged, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/sensitivity_analysis/SA_SMDH_flagged.rds"</span>))</span>
<span id="cb80-33"><a href="#cb80-33" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the dataset for this model</span></span>
<span id="cb80-34"><a href="#cb80-34" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(SMDH_con_flagged,here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/05_data_sensitivity_analysis/dataset_SMDH_con_flagged.csv"</span>))</span>
<span id="cb80-35"><a href="#cb80-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-36"><a href="#cb80-36" aria-hidden="true" tabindex="-1"></a><span class="co"># What is the effect now?</span></span>
<span id="cb80-37"><a href="#cb80-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-38"><a href="#cb80-38" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(SA_SMDH_flagged)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Multivariate Meta-Analysis Model (k = 274; method: REML)

   logLik   Deviance        AIC        BIC       AICc   
-344.0150   688.0299   700.0299   721.6868   700.3457   

Variance Components:

            estim    sqrt  nlvls  fixed          factor 
sigma^2.1  0.0000  0.0015     28     no        paper_ID 
sigma^2.2  0.2487  0.4987    274     no  Observation_ID 
sigma^2.3  0.0000  0.0025     53     no  group_ID_coded 
sigma^2.4  0.0000  0.0002     16     no   population_ID 
sigma^2.5  0.0000  0.0000      7     no    bird_species 

Test for Heterogeneity:
Q(df = 273) = 1184.8136, p-val &lt; .0001

Model Results:

estimate      se    tval   df    pval   ci.lb   ci.ub     
  0.1911  0.0631  3.0269  273  0.0027  0.0668  0.3154  ** 

---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="fu">predict</span>(SA_SMDH_flagged)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   pred     se  ci.lb  ci.ub   pi.lb  pi.ub 
 0.1911 0.0631 0.0668 0.3154 -0.7985 1.1807 </code></pre>
</div>
</div></div>
</section>
<section id="without-0-es" class="level3">
<h3 class="anchored" data-anchor-id="without-0-es">Without 0 ES</h3>
<div class="cell-container"><div class="cell-decorator"><pre>In [15]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="co">#----------------------------------------------------------------------------</span></span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a><span class="co"># for lnRR as Effect Size</span></span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a><span class="co">#----------------------------------------------------------------------------</span></span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Preparing new variance-covariance matrix</span></span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-7"><a href="#cb84-7" aria-hidden="true" tabindex="-1"></a>VCV_lnRR_no_missingES <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> lnRR_variance,</span>
<span id="cb84-8"><a href="#cb84-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb84-9"><a href="#cb84-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb84-10"><a href="#cb84-10" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> lnRR_sin_missingES,</span>
<span id="cb84-11"><a href="#cb84-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb84-12"><a href="#cb84-12" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb84-13"><a href="#cb84-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-14"><a href="#cb84-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-15"><a href="#cb84-15" aria-hidden="true" tabindex="-1"></a><span class="do">## Running intercept only model</span></span>
<span id="cb84-16"><a href="#cb84-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-17"><a href="#cb84-17" aria-hidden="true" tabindex="-1"></a>SA_lnRR_no_missingES<span class="ot">&lt;-</span> <span class="fu">rma.mv</span>(<span class="at">yi =</span> lnRR_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb84-18"><a href="#cb84-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_lnRR_no_missingES, <span class="co"># specify lnRR's sampling variance-covariance matrix;</span></span>
<span id="cb84-19"><a href="#cb84-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb84-20"><a href="#cb84-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb84-21"><a href="#cb84-21" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb84-22"><a href="#cb84-22" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb84-23"><a href="#cb84-23" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb84-24"><a href="#cb84-24" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb84-25"><a href="#cb84-25" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb84-26"><a href="#cb84-26" aria-hidden="true" tabindex="-1"></a>                    <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb84-27"><a href="#cb84-27" aria-hidden="true" tabindex="-1"></a>                    <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb84-28"><a href="#cb84-28" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> lnRR_sin_missingES)</span>
<span id="cb84-29"><a href="#cb84-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-30"><a href="#cb84-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-31"><a href="#cb84-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-32"><a href="#cb84-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the model</span></span>
<span id="cb84-33"><a href="#cb84-33" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SA_lnRR_no_missingES, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/sensitivity_analysis/SA_lnRR_no_missingES.rds"</span>))</span>
<span id="cb84-34"><a href="#cb84-34" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the dataset for this model</span></span>
<span id="cb84-35"><a href="#cb84-35" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(lnRR_sin_missingES,here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/05_data_sensitivity_analysis/dataset_lnRR_sin_missingES.csv"</span>))</span>
<span id="cb84-36"><a href="#cb84-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-37"><a href="#cb84-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-38"><a href="#cb84-38" aria-hidden="true" tabindex="-1"></a><span class="co"># What is the effect now?</span></span>
<span id="cb84-39"><a href="#cb84-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-40"><a href="#cb84-40" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(SA_lnRR_no_missingES)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Multivariate Meta-Analysis Model (k = 231; method: REML)

  logLik  Deviance       AIC       BIC      AICc   
-30.0909   60.1817   72.1817   92.8102   72.5584   

Variance Components:

            estim    sqrt  nlvls  fixed          factor 
sigma^2.1  0.0000  0.0002     26     no        paper_ID 
sigma^2.2  0.0244  0.1563    231     no  Observation_ID 
sigma^2.3  0.0000  0.0001     46     no  group_ID_coded 
sigma^2.4  0.0000  0.0000     16     no   population_ID 
sigma^2.5  0.0000  0.0000      7     no    bird_species 

Test for Heterogeneity:
Q(df = 230) = 2340.8098, p-val &lt; .0001

Model Results:

estimate      se    tval   df    pval    ci.lb   ci.ub    
  0.0162  0.0158  1.0289  230  0.3046  -0.0148  0.0473    

---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="fu">predict</span>(SA_lnRR_no_missingES)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   pred     se   ci.lb  ci.ub   pi.lb  pi.ub 
 0.0162 0.0158 -0.0148 0.0473 -0.2934 0.3258 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="co">#----------------------------------------------------------------------------</span></span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a><span class="co"># for SMDH as Effect Size</span></span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a><span class="co">#----------------------------------------------------------------------------</span></span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Preparing new variance-covariance matrix</span></span>
<span id="cb88-6"><a href="#cb88-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-7"><a href="#cb88-7" aria-hidden="true" tabindex="-1"></a>VCV_SMDH_no_missingES <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> SMDH_variance,</span>
<span id="cb88-8"><a href="#cb88-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb88-9"><a href="#cb88-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb88-10"><a href="#cb88-10" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> SMDH_sin_missingES,</span>
<span id="cb88-11"><a href="#cb88-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb88-12"><a href="#cb88-12" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb88-13"><a href="#cb88-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-14"><a href="#cb88-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-15"><a href="#cb88-15" aria-hidden="true" tabindex="-1"></a><span class="do">## Running intercept only model</span></span>
<span id="cb88-16"><a href="#cb88-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-17"><a href="#cb88-17" aria-hidden="true" tabindex="-1"></a>SA_SMDH_no_missingES<span class="ot">&lt;-</span> <span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb88-18"><a href="#cb88-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_SMDH_no_missingES, <span class="co"># specify lnRR's sampling variance-covariance matrix;</span></span>
<span id="cb88-19"><a href="#cb88-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb88-20"><a href="#cb88-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb88-21"><a href="#cb88-21" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb88-22"><a href="#cb88-22" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb88-23"><a href="#cb88-23" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb88-24"><a href="#cb88-24" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb88-25"><a href="#cb88-25" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb88-26"><a href="#cb88-26" aria-hidden="true" tabindex="-1"></a>                    <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb88-27"><a href="#cb88-27" aria-hidden="true" tabindex="-1"></a>                    <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb88-28"><a href="#cb88-28" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> SMDH_sin_missingES)</span>
<span id="cb88-29"><a href="#cb88-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-30"><a href="#cb88-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-31"><a href="#cb88-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the model</span></span>
<span id="cb88-32"><a href="#cb88-32" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SA_SMDH_no_missingES, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/sensitivity_analysis/SA_SMDH_no_missingES.rds"</span>))</span>
<span id="cb88-33"><a href="#cb88-33" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the dataset for this model</span></span>
<span id="cb88-34"><a href="#cb88-34" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(SMDH_sin_missingES,here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/05_data_sensitivity_analysis/dataset_SMDH_sin_missingES.csv"</span>))</span>
<span id="cb88-35"><a href="#cb88-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-36"><a href="#cb88-36" aria-hidden="true" tabindex="-1"></a><span class="co"># What is the effect now?</span></span>
<span id="cb88-37"><a href="#cb88-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-38"><a href="#cb88-38" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(SA_SMDH_no_missingES)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Multivariate Meta-Analysis Model (k = 237; method: REML)

   logLik   Deviance        AIC        BIC       AICc   
-315.4239   630.8479   642.8479   663.6309   643.2147   

Variance Components:

            estim    sqrt  nlvls  fixed          factor 
sigma^2.1  0.0000  0.0008     26     no        paper_ID 
sigma^2.2  0.2967  0.5447    237     no  Observation_ID 
sigma^2.3  0.0000  0.0012     47     no  group_ID_coded 
sigma^2.4  0.0000  0.0004     16     no   population_ID 
sigma^2.5  0.0000  0.0001      7     no    bird_species 

Test for Heterogeneity:
Q(df = 236) = 1113.5021, p-val &lt; .0001

Model Results:

estimate      se    tval   df    pval   ci.lb   ci.ub     
  0.1860  0.0701  2.6545  236  0.0085  0.0480  0.3240  ** 

---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="fu">predict</span>(SA_SMDH_no_missingES)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   pred     se  ci.lb  ci.ub   pi.lb  pi.ub 
 0.1860 0.0701 0.0480 0.3240 -0.8960 1.2680 </code></pre>
</div>
</div></div>
</section>
<section id="with-flagged-proxies-and-no-missing-es" class="level3">
<h3 class="anchored" data-anchor-id="with-flagged-proxies-and-no-missing-es">With flagged proxies and no missing ES</h3>
<div class="cell-container"><div class="cell-decorator"><pre>In [16]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="co">#----------------------------------------------------------------------------</span></span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a><span class="co"># for lnRR as Effect Size</span></span>
<span id="cb92-3"><a href="#cb92-3" aria-hidden="true" tabindex="-1"></a><span class="co">#----------------------------------------------------------------------------</span></span>
<span id="cb92-4"><a href="#cb92-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-5"><a href="#cb92-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Preparing new variance-covariance matrix</span></span>
<span id="cb92-6"><a href="#cb92-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-7"><a href="#cb92-7" aria-hidden="true" tabindex="-1"></a>VCV_flagged_no_missingES <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> lnRR_variance,</span>
<span id="cb92-8"><a href="#cb92-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb92-9"><a href="#cb92-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb92-10"><a href="#cb92-10" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> lnRR_flagged_no_missingES,</span>
<span id="cb92-11"><a href="#cb92-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb92-12"><a href="#cb92-12" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb92-13"><a href="#cb92-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-14"><a href="#cb92-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-15"><a href="#cb92-15" aria-hidden="true" tabindex="-1"></a><span class="do">## Running intercept only model</span></span>
<span id="cb92-16"><a href="#cb92-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-17"><a href="#cb92-17" aria-hidden="true" tabindex="-1"></a>SA_lnRR_flagged_no_missingES<span class="ot">&lt;-</span> <span class="fu">rma.mv</span>(<span class="at">yi =</span> lnRR_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb92-18"><a href="#cb92-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_flagged_no_missingES, <span class="co"># specify lnRR's sampling variance-covariance matrix;</span></span>
<span id="cb92-19"><a href="#cb92-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb92-20"><a href="#cb92-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb92-21"><a href="#cb92-21" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb92-22"><a href="#cb92-22" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb92-23"><a href="#cb92-23" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb92-24"><a href="#cb92-24" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb92-25"><a href="#cb92-25" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb92-26"><a href="#cb92-26" aria-hidden="true" tabindex="-1"></a>                    <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb92-27"><a href="#cb92-27" aria-hidden="true" tabindex="-1"></a>                    <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb92-28"><a href="#cb92-28" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> lnRR_flagged_no_missingES)</span>
<span id="cb92-29"><a href="#cb92-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-30"><a href="#cb92-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-31"><a href="#cb92-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-32"><a href="#cb92-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the model</span></span>
<span id="cb92-33"><a href="#cb92-33" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SA_lnRR_flagged_no_missingES, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/sensitivity_analysis/SA_lnRR_flagged_no_missingES.rds"</span>))</span>
<span id="cb92-34"><a href="#cb92-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-35"><a href="#cb92-35" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the dataset for this model</span></span>
<span id="cb92-36"><a href="#cb92-36" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(lnRR_flagged_no_missingES,here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/05_data_sensitivity_analysis/dataset_lnRR_flagged_no_missingES.csv"</span>))</span>
<span id="cb92-37"><a href="#cb92-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-38"><a href="#cb92-38" aria-hidden="true" tabindex="-1"></a><span class="co"># What is the effect now?</span></span>
<span id="cb92-39"><a href="#cb92-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-40"><a href="#cb92-40" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(SA_lnRR_flagged_no_missingES)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Multivariate Meta-Analysis Model (k = 252; method: REML)

  logLik  Deviance       AIC       BIC      AICc   
-29.3634   58.7268   70.7268   91.8795   71.0710   

Variance Components:

            estim    sqrt  nlvls  fixed          factor 
sigma^2.1  0.0000  0.0003     28     no        paper_ID 
sigma^2.2  0.0237  0.1538    252     no  Observation_ID 
sigma^2.3  0.0000  0.0001     52     no  group_ID_coded 
sigma^2.4  0.0000  0.0001     16     no   population_ID 
sigma^2.5  0.0000  0.0000      7     no    bird_species 

Test for Heterogeneity:
Q(df = 251) = 2458.3386, p-val &lt; .0001

Model Results:

estimate      se    tval   df    pval    ci.lb   ci.ub    
  0.0211  0.0151  1.4013  251  0.1624  -0.0086  0.0509    

---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="fu">predict</span>(SA_lnRR_flagged_no_missingES)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   pred     se   ci.lb  ci.ub   pi.lb  pi.ub 
 0.0211 0.0151 -0.0086 0.0509 -0.2832 0.3255 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="co">#----------------------------------------------------------------------------</span></span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a><span class="co"># for SMDH as Effect Size</span></span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a><span class="co">#----------------------------------------------------------------------------</span></span>
<span id="cb96-4"><a href="#cb96-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-5"><a href="#cb96-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Preparing new variance-covariance matrix</span></span>
<span id="cb96-6"><a href="#cb96-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-7"><a href="#cb96-7" aria-hidden="true" tabindex="-1"></a>VCV_SMDH_flagged_no_missingES <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> SMDH_variance,</span>
<span id="cb96-8"><a href="#cb96-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb96-9"><a href="#cb96-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb96-10"><a href="#cb96-10" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> SMDH_flagged_no_missingES,</span>
<span id="cb96-11"><a href="#cb96-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb96-12"><a href="#cb96-12" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb96-13"><a href="#cb96-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-14"><a href="#cb96-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-15"><a href="#cb96-15" aria-hidden="true" tabindex="-1"></a><span class="do">## Running intercept only model</span></span>
<span id="cb96-16"><a href="#cb96-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-17"><a href="#cb96-17" aria-hidden="true" tabindex="-1"></a>SA_SMDH_flagged_no_missingES<span class="ot">&lt;-</span> <span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb96-18"><a href="#cb96-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_SMDH_flagged_no_missingES, <span class="co"># specify lnRR's sampling variance-covariance matrix;</span></span>
<span id="cb96-19"><a href="#cb96-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb96-20"><a href="#cb96-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb96-21"><a href="#cb96-21" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb96-22"><a href="#cb96-22" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb96-23"><a href="#cb96-23" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb96-24"><a href="#cb96-24" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb96-25"><a href="#cb96-25" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb96-26"><a href="#cb96-26" aria-hidden="true" tabindex="-1"></a>                    <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb96-27"><a href="#cb96-27" aria-hidden="true" tabindex="-1"></a>                    <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb96-28"><a href="#cb96-28" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> SMDH_flagged_no_missingES)</span>
<span id="cb96-29"><a href="#cb96-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-30"><a href="#cb96-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-31"><a href="#cb96-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the model</span></span>
<span id="cb96-32"><a href="#cb96-32" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SA_SMDH_flagged_no_missingES, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/sensitivity_analysis/SA_SMDH_flagged_no_missingES.rds"</span>))</span>
<span id="cb96-33"><a href="#cb96-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-34"><a href="#cb96-34" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the dataset for this model</span></span>
<span id="cb96-35"><a href="#cb96-35" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(SMDH_flagged_no_missingES,here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/05_data_sensitivity_analysis/dataset_SMDH_flagged_no_missingES.csv"</span>))</span>
<span id="cb96-36"><a href="#cb96-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-37"><a href="#cb96-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-38"><a href="#cb96-38" aria-hidden="true" tabindex="-1"></a><span class="co"># What is the effect now?</span></span>
<span id="cb96-39"><a href="#cb96-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-40"><a href="#cb96-40" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(SA_SMDH_flagged_no_missingES)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Multivariate Meta-Analysis Model (k = 258; method: REML)

   logLik   Deviance        AIC        BIC       AICc   
-335.1461   670.2922   682.2922   703.5867   682.6282   

Variance Components:

            estim    sqrt  nlvls  fixed          factor 
sigma^2.1  0.0000  0.0004     28     no        paper_ID 
sigma^2.2  0.2812  0.5303    258     no  Observation_ID 
sigma^2.3  0.0001  0.0097     53     no  group_ID_coded 
sigma^2.4  0.0000  0.0001     16     no   population_ID 
sigma^2.5  0.0000  0.0000      7     no    bird_species 

Test for Heterogeneity:
Q(df = 257) = 1183.6500, p-val &lt; .0001

Model Results:

estimate      se    tval   df    pval   ci.lb   ci.ub     
  0.1988  0.0661  3.0075  257  0.0029  0.0686  0.3289  ** 

---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="fu">predict</span>(SA_SMDH_flagged_no_missingES)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   pred     se  ci.lb  ci.ub   pi.lb  pi.ub 
 0.1988 0.0661 0.0686 0.3289 -0.8538 1.2514 </code></pre>
</div>
</div></div>
</section>
<section id="no-inferential-stats-for-smdh-effect-sizes-calculated-from-means-sds-and-sample-size" class="level3">
<h3 class="anchored" data-anchor-id="no-inferential-stats-for-smdh-effect-sizes-calculated-from-means-sds-and-sample-size">No inferential stats for SMDH: effect sizes calculated from means, SDs, and sample size</h3>
<p>(i.e., excluding any effect sizes calculated from inferential statistics)</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [17]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="co">#----------------------------------------------------------------------------</span></span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a><span class="co"># for SMDH as Effect Size</span></span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a><span class="co">#----------------------------------------------------------------------------</span></span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-5"><a href="#cb100-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Preparing new variance-covariance matrix</span></span>
<span id="cb100-6"><a href="#cb100-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-7"><a href="#cb100-7" aria-hidden="true" tabindex="-1"></a>VCV_SMDH_no_inferential <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> SMDH_variance,</span>
<span id="cb100-8"><a href="#cb100-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb100-9"><a href="#cb100-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb100-10"><a href="#cb100-10" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> SMDH_no_inferential,</span>
<span id="cb100-11"><a href="#cb100-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb100-12"><a href="#cb100-12" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb100-13"><a href="#cb100-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-14"><a href="#cb100-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-15"><a href="#cb100-15" aria-hidden="true" tabindex="-1"></a><span class="do">## Running intercept only model</span></span>
<span id="cb100-16"><a href="#cb100-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-17"><a href="#cb100-17" aria-hidden="true" tabindex="-1"></a>SA_SMDH_no_inferential<span class="ot">&lt;-</span> <span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb100-18"><a href="#cb100-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_SMDH_no_inferential, <span class="co"># specify lnRR's sampling variance-covariance matrix;</span></span>
<span id="cb100-19"><a href="#cb100-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb100-20"><a href="#cb100-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb100-21"><a href="#cb100-21" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb100-22"><a href="#cb100-22" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb100-23"><a href="#cb100-23" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb100-24"><a href="#cb100-24" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb100-25"><a href="#cb100-25" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb100-26"><a href="#cb100-26" aria-hidden="true" tabindex="-1"></a>                    <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb100-27"><a href="#cb100-27" aria-hidden="true" tabindex="-1"></a>                    <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb100-28"><a href="#cb100-28" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> SMDH_no_inferential)</span>
<span id="cb100-29"><a href="#cb100-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-30"><a href="#cb100-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-31"><a href="#cb100-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-32"><a href="#cb100-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the model</span></span>
<span id="cb100-33"><a href="#cb100-33" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SA_SMDH_no_inferential, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/sensitivity_analysis/SA_SMDH_no_inferential.rds"</span>))</span>
<span id="cb100-34"><a href="#cb100-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-35"><a href="#cb100-35" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the dataset for this model</span></span>
<span id="cb100-36"><a href="#cb100-36" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(SMDH_no_inferential,here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/05_data_sensitivity_analysis/dataset_SMDH_no_inferential.csv"</span>))</span>
<span id="cb100-37"><a href="#cb100-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-38"><a href="#cb100-38" aria-hidden="true" tabindex="-1"></a><span class="co"># What is the effect now?</span></span>
<span id="cb100-39"><a href="#cb100-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-40"><a href="#cb100-40" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(SA_SMDH_no_inferential)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Multivariate Meta-Analysis Model (k = 245; method: REML)

   logLik   Deviance        AIC        BIC       AICc   
-319.1347   638.2693   650.2693   671.2524   650.6238   

Variance Components:

            estim    sqrt  nlvls  fixed          factor 
sigma^2.1  0.0000  0.0005     26     no        paper_ID 
sigma^2.2  0.2716  0.5212    245     no  Observation_ID 
sigma^2.3  0.0000  0.0018     46     no  group_ID_coded 
sigma^2.4  0.0000  0.0003     16     no   population_ID 
sigma^2.5  0.0000  0.0000      7     no    bird_species 

Test for Heterogeneity:
Q(df = 244) = 1105.9021, p-val &lt; .0001

Model Results:

estimate      se    tval   df    pval   ci.lb   ci.ub     
  0.1794  0.0685  2.6193  244  0.0094  0.0445  0.3143  ** 

---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="fu">predict</span>(SA_SMDH_no_inferential)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   pred     se  ci.lb  ci.ub   pi.lb  pi.ub 
 0.1794 0.0685 0.0445 0.3143 -0.8560 1.2148 </code></pre>
</div>
</div></div>
</section>
<section id="exluding-failed-gearys-test-for-lnrr" class="level3">
<h3 class="anchored" data-anchor-id="exluding-failed-gearys-test-for-lnrr">Exluding failed geary’s test for lnRR</h3>
<div class="cell-container"><div class="cell-decorator"><pre>In [18]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="co">#----------------------------------------------------------------------------</span></span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a><span class="co"># for  lnRR  as Effect Size</span></span>
<span id="cb104-3"><a href="#cb104-3" aria-hidden="true" tabindex="-1"></a><span class="co">#----------------------------------------------------------------------------</span></span>
<span id="cb104-4"><a href="#cb104-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-5"><a href="#cb104-5" aria-hidden="true" tabindex="-1"></a>VCV_lnRR_geary_passed <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> lnRR_variance,</span>
<span id="cb104-6"><a href="#cb104-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb104-7"><a href="#cb104-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb104-8"><a href="#cb104-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> lnRR_geary_passed,</span>
<span id="cb104-9"><a href="#cb104-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb104-10"><a href="#cb104-10" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb104-11"><a href="#cb104-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-12"><a href="#cb104-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-13"><a href="#cb104-13" aria-hidden="true" tabindex="-1"></a>SA_lnRR_geary_passed<span class="ot">&lt;-</span> <span class="fu">rma.mv</span>(<span class="at">yi =</span> lnRR_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb104-14"><a href="#cb104-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_lnRR_geary_passed, <span class="co"># specify lnRR's sampling variance-covariance matrix;</span></span>
<span id="cb104-15"><a href="#cb104-15" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb104-16"><a href="#cb104-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb104-17"><a href="#cb104-17" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb104-18"><a href="#cb104-18" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb104-19"><a href="#cb104-19" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb104-20"><a href="#cb104-20" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb104-21"><a href="#cb104-21" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb104-22"><a href="#cb104-22" aria-hidden="true" tabindex="-1"></a>                    <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb104-23"><a href="#cb104-23" aria-hidden="true" tabindex="-1"></a>                    <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb104-24"><a href="#cb104-24" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> lnRR_geary_passed)</span>
<span id="cb104-25"><a href="#cb104-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-26"><a href="#cb104-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the model</span></span>
<span id="cb104-27"><a href="#cb104-27" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SA_lnRR_geary_passed, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/sensitivity_analysis/SA_lnRR_geary_passed.rds"</span>))</span>
<span id="cb104-28"><a href="#cb104-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-29"><a href="#cb104-29" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the dataset for this model</span></span>
<span id="cb104-30"><a href="#cb104-30" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(lnRR_geary_passed,here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/05_data_sensitivity_analysis/dataset_lnRR_geary_passed.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
<section id="excluding-the-parasite-and-pathogen-load-fitness-proxies" class="level3">
<h3 class="anchored" data-anchor-id="excluding-the-parasite-and-pathogen-load-fitness-proxies">Excluding the parasite and pathogen load fitness proxies</h3>
<p>It excludes all the fitness proxies relating to parasite and pathogen loads/ abundance etc.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [19]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="co">#----------------------------------------------------------------------------</span></span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a><span class="co"># for  lnRR  as Effect Size</span></span>
<span id="cb105-3"><a href="#cb105-3" aria-hidden="true" tabindex="-1"></a><span class="co">#----------------------------------------------------------------------------</span></span>
<span id="cb105-4"><a href="#cb105-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-5"><a href="#cb105-5" aria-hidden="true" tabindex="-1"></a>dataset_lnRR_no_para<span class="ot">&lt;-</span>dataset_lnRR<span class="sc">%&gt;%</span><span class="fu">filter</span>(<span class="fu">is.na</span>(parasite_type))</span>
<span id="cb105-6"><a href="#cb105-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-7"><a href="#cb105-7" aria-hidden="true" tabindex="-1"></a><span class="do">## The dataset is reduced to almost half</span></span>
<span id="cb105-8"><a href="#cb105-8" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(dataset_lnRR)<span class="sc">-</span><span class="fu">nrow</span>(dataset_lnRR_no_para)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 82</code></pre>
</div>
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a>VCV_lnRR_no_para <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> lnRR_variance,</span>
<span id="cb107-2"><a href="#cb107-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb107-3"><a href="#cb107-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb107-4"><a href="#cb107-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_lnRR_no_para,</span>
<span id="cb107-5"><a href="#cb107-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb107-6"><a href="#cb107-6" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb107-7"><a href="#cb107-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-8"><a href="#cb107-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-9"><a href="#cb107-9" aria-hidden="true" tabindex="-1"></a>SA_lnRR_no_para<span class="ot">&lt;-</span> <span class="fu">rma.mv</span>(<span class="at">yi =</span> lnRR_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb107-10"><a href="#cb107-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_lnRR_no_para, <span class="co"># specify lnRR's sampling variance-covariance matrix;</span></span>
<span id="cb107-11"><a href="#cb107-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb107-12"><a href="#cb107-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb107-13"><a href="#cb107-13" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb107-14"><a href="#cb107-14" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb107-15"><a href="#cb107-15" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb107-16"><a href="#cb107-16" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb107-17"><a href="#cb107-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb107-18"><a href="#cb107-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb107-19"><a href="#cb107-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb107-20"><a href="#cb107-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> dataset_lnRR_no_para)</span>
<span id="cb107-21"><a href="#cb107-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-22"><a href="#cb107-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the model</span></span>
<span id="cb107-23"><a href="#cb107-23" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SA_lnRR_no_para, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/sensitivity_analysis/SA_lnRR_no_para.rds"</span>))</span>
<span id="cb107-24"><a href="#cb107-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-25"><a href="#cb107-25" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the dataset for this model</span></span>
<span id="cb107-26"><a href="#cb107-26" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(dataset_lnRR_no_para,here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/05_data_sensitivity_analysis/dataset_lnRR_no_para.csv"</span>))</span>
<span id="cb107-27"><a href="#cb107-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-28"><a href="#cb107-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-29"><a href="#cb107-29" aria-hidden="true" tabindex="-1"></a><span class="co">#----------------------------------------------------------------------------</span></span>
<span id="cb107-30"><a href="#cb107-30" aria-hidden="true" tabindex="-1"></a><span class="co"># for  SMDH  as Effect Size</span></span>
<span id="cb107-31"><a href="#cb107-31" aria-hidden="true" tabindex="-1"></a><span class="co">#----------------------------------------------------------------------------</span></span>
<span id="cb107-32"><a href="#cb107-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-33"><a href="#cb107-33" aria-hidden="true" tabindex="-1"></a>dataset_SMDH_no_para<span class="ot">&lt;-</span>dataset_SMDH<span class="sc">%&gt;%</span><span class="fu">filter</span>(<span class="fu">is.na</span>(parasite_type))</span>
<span id="cb107-34"><a href="#cb107-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-35"><a href="#cb107-35" aria-hidden="true" tabindex="-1"></a><span class="do">## The dataset is reduced to almost half</span></span>
<span id="cb107-36"><a href="#cb107-36" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(dataset_SMDH)<span class="sc">-</span><span class="fu">nrow</span>(dataset_SMDH_no_para)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 86</code></pre>
</div>
<div class="sourceCode cell-code" id="cb109"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a>VCV_SMDH_no_para <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> SMDH_variance,</span>
<span id="cb109-2"><a href="#cb109-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb109-3"><a href="#cb109-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb109-4"><a href="#cb109-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_SMDH_no_para,</span>
<span id="cb109-5"><a href="#cb109-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span>) <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb109-6"><a href="#cb109-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-7"><a href="#cb109-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-8"><a href="#cb109-8" aria-hidden="true" tabindex="-1"></a>SA_SMDH_no_para<span class="ot">&lt;-</span> <span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb109-9"><a href="#cb109-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_SMDH_no_para, <span class="co"># specify lnRR's sampling variance-covariance matrix;</span></span>
<span id="cb109-10"><a href="#cb109-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb109-11"><a href="#cb109-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb109-12"><a href="#cb109-12" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb109-13"><a href="#cb109-13" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb109-14"><a href="#cb109-14" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb109-15"><a href="#cb109-15" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb109-16"><a href="#cb109-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb109-17"><a href="#cb109-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb109-18"><a href="#cb109-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb109-19"><a href="#cb109-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> dataset_SMDH_no_para)</span>
<span id="cb109-20"><a href="#cb109-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-21"><a href="#cb109-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the model</span></span>
<span id="cb109-22"><a href="#cb109-22" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SA_SMDH_no_para, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/sensitivity_analysis/SA_SMDH_no_para.rds"</span>))</span>
<span id="cb109-23"><a href="#cb109-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-24"><a href="#cb109-24" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the dataset for this model</span></span>
<span id="cb109-25"><a href="#cb109-25" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(dataset_SMDH_no_para,here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/05_data_sensitivity_analysis/dataset_SMDH_no_para.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
<section id="different-values-of-sampling-variance-correlation" class="level2">
<h2 class="anchored" data-anchor-id="different-values-of-sampling-variance-correlation">Different values of sampling variance correlation</h2>
<p><u><strong>Robust variance estimation to guard against arbitary value of sampling correlation</strong></u> - ‘robust()’ function from R package ‘metafor’ (Viechtbauer 2010) using ‘CR2’ (bias reduced linearization correction) from the R package ‘clubSandwich’ (Pustejovsky 2024) for small-sample size adjustment.</p>
<p>At the moment, this is not working..</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [20]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="co">#MLMA_lnRR_RVE&lt;- robust(MLMA_lnRR_VCV,</span></span>
<span id="cb110-2"><a href="#cb110-2" aria-hidden="true" tabindex="-1"></a><span class="co">#                        cluster = paper_ID,</span></span>
<span id="cb110-3"><a href="#cb110-3" aria-hidden="true" tabindex="-1"></a><span class="co">#                        clubSandwich = TRUE,</span></span>
<span id="cb110-4"><a href="#cb110-4" aria-hidden="true" tabindex="-1"></a><span class="co">#                        verbose=T)</span></span>
<span id="cb110-5"><a href="#cb110-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-6"><a href="#cb110-6" aria-hidden="true" tabindex="-1"></a><span class="co"># # summary(MLMA_lnRR_RVE)</span></span>
<span id="cb110-7"><a href="#cb110-7" aria-hidden="true" tabindex="-1"></a><span class="co"># I2_CI &lt;- i2_ml(MLMA_lnRR_RVE) # Relative heterogeneity</span></span>
<span id="cb110-8"><a href="#cb110-8" aria-hidden="true" tabindex="-1"></a><span class="co"># round(I2_CI,2) # absolute heterogeneity is reported in the summary of the model</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Important
</div>
</div>
<div class="callout-body-container callout-body">
<p>Sensitivity Analysis</p>
<p>We will also perform a sensitivity analysis using 0.3 and 0.7 correlation between the sampling variances from the same paper_ID and report it in the supplementary materials. If this sensitivity analysis shows large differences in conclusions for the intercept-only model, we will proceed to perform such sensitivity analysis for the meta-regressions.</p>
</div>
</div>
<p>First, set a series of ρs (i.e., 0.3, 0.5, 0.7) (we assume these values arbitrarily):</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [21]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb111"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a><span class="co"># rho_range &lt;- c(0.3, 0.5, 0.7)</span></span>
<span id="cb111-2"><a href="#cb111-2" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb111-3"><a href="#cb111-3" aria-hidden="true" tabindex="-1"></a><span class="co"># MLMAlnRR_VCV_range &lt;- list() # repeatedly run the specified model with varying rho</span></span>
<span id="cb111-4"><a href="#cb111-4" aria-hidden="true" tabindex="-1"></a><span class="co"># for (i in 1:length(rho_range))</span></span>
<span id="cb111-5"><a href="#cb111-5" aria-hidden="true" tabindex="-1"></a><span class="co"># {VCV_range &lt;- vcalc(vi = lnRR_variance,</span></span>
<span id="cb111-6"><a href="#cb111-6" aria-hidden="true" tabindex="-1"></a><span class="co">#                   cluster = paper_ID,</span></span>
<span id="cb111-7"><a href="#cb111-7" aria-hidden="true" tabindex="-1"></a><span class="co">#                   subgroup = group_ID,</span></span>
<span id="cb111-8"><a href="#cb111-8" aria-hidden="true" tabindex="-1"></a><span class="co">#                   obs = Observation_ID,</span></span>
<span id="cb111-9"><a href="#cb111-9" aria-hidden="true" tabindex="-1"></a><span class="co">#                   data = dataset_lnRR,</span></span>
<span id="cb111-10"><a href="#cb111-10" aria-hidden="true" tabindex="-1"></a><span class="co">#                    rho = rho_range[i],</span></span>
<span id="cb111-11"><a href="#cb111-11" aria-hidden="true" tabindex="-1"></a><span class="co">#                     ) # impute sampling variance covariance matrix with varying rho</span></span>
<span id="cb111-12"><a href="#cb111-12" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb111-13"><a href="#cb111-13" aria-hidden="true" tabindex="-1"></a><span class="co"># MLMAlnRR_VCV_range[[i]] &lt;- rma.mv(yi = lnRR_sign,</span></span>
<span id="cb111-14"><a href="#cb111-14" aria-hidden="true" tabindex="-1"></a><span class="co">#                                   V = VCV_range, # sampling variance covariance matrix with varying values of rho.</span></span>
<span id="cb111-15"><a href="#cb111-15" aria-hidden="true" tabindex="-1"></a><span class="co">#                                  random = list(~ 1 | paper_ID,</span></span>
<span id="cb111-16"><a href="#cb111-16" aria-hidden="true" tabindex="-1"></a><span class="co">#                                   ~ 1 | Observation_ID,</span></span>
<span id="cb111-17"><a href="#cb111-17" aria-hidden="true" tabindex="-1"></a><span class="co">#                                   ~ 1 | experiment_ID,</span></span>
<span id="cb111-18"><a href="#cb111-18" aria-hidden="true" tabindex="-1"></a><span class="co">#                                   ~ 1 | group_ID,</span></span>
<span id="cb111-19"><a href="#cb111-19" aria-hidden="true" tabindex="-1"></a><span class="co">#                                   ~ 1 | repeated_trait_ID),</span></span>
<span id="cb111-20"><a href="#cb111-20" aria-hidden="true" tabindex="-1"></a><span class="co">#                                   method = "REML",</span></span>
<span id="cb111-21"><a href="#cb111-21" aria-hidden="true" tabindex="-1"></a><span class="co">#                                   data = dataset_lnRR # run the model with varying rho</span></span>
<span id="cb111-22"><a href="#cb111-22" aria-hidden="true" tabindex="-1"></a><span class="co"># )</span></span>
<span id="cb111-23"><a href="#cb111-23" aria-hidden="true" tabindex="-1"></a><span class="co"># }</span></span>
<span id="cb111-24"><a href="#cb111-24" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb111-25"><a href="#cb111-25" aria-hidden="true" tabindex="-1"></a><span class="co"># save(VCV_range, file = here::here("GNM/figures/VCV_range.Rdata"))</span></span>
<span id="cb111-26"><a href="#cb111-26" aria-hidden="true" tabindex="-1"></a><span class="co"># save(MLMAlnRR_VCV_range, file = here::here("GNM/figures/MLMAlnRR_VCV_range.Rdata"))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [22]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb112"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a><span class="co"># t4 &lt;- data.frame(rho  = rho_range,</span></span>
<span id="cb112-2"><a href="#cb112-2" aria-hidden="true" tabindex="-1"></a><span class="co">#                  "overall effect"  = sapply(MLMAlnRR_VCV_range, function(x) coef(x)),</span></span>
<span id="cb112-3"><a href="#cb112-3" aria-hidden="true" tabindex="-1"></a><span class="co">#                  "standard error" = sapply(MLMAlnRR_VCV_range, function(x) x$se),</span></span>
<span id="cb112-4"><a href="#cb112-4" aria-hidden="true" tabindex="-1"></a><span class="co">#                  "p-value" = sapply(MLMAlnRR_VCV_range, function(x) x$pval),</span></span>
<span id="cb112-5"><a href="#cb112-5" aria-hidden="true" tabindex="-1"></a><span class="co">#                  "Lower CI" = sapply(MLMAlnRR_VCV_range, function(x) x$ci.lb),</span></span>
<span id="cb112-6"><a href="#cb112-6" aria-hidden="true" tabindex="-1"></a><span class="co">#                  "Upper CI" = sapply(MLMAlnRR_VCV_range, function(x) x$ci.ub),</span></span>
<span id="cb112-7"><a href="#cb112-7" aria-hidden="true" tabindex="-1"></a><span class="co">#                  "Log-likehood" = sapply(MLMAlnRR_VCV_range, function(x) fitstats(x)[1,1]),</span></span>
<span id="cb112-8"><a href="#cb112-8" aria-hidden="true" tabindex="-1"></a><span class="co">#                  "AIC" = sapply(MLMAlnRR_VCV_range, function(x) fitstats(x)[3,1]),</span></span>
<span id="cb112-9"><a href="#cb112-9" aria-hidden="true" tabindex="-1"></a><span class="co">#                  "BIC" = sapply(MLMAlnRR_VCV_range, function(x) fitstats(x)[4,1]),</span></span>
<span id="cb112-10"><a href="#cb112-10" aria-hidden="true" tabindex="-1"></a><span class="co">#                  "AICc" = sapply(MLMAlnRR_VCV_range, function(x) fitstats(x)[5,1]))</span></span>
<span id="cb112-11"><a href="#cb112-11" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb112-12"><a href="#cb112-12" aria-hidden="true" tabindex="-1"></a><span class="co"># colnames(t4) &lt;- c("Sampling correlation", "Overall effect (lnRR)", "Standard error", "p-value", "Lower CI", "Upper CI", "Log-likehood", "AIC", "BIC", "AICc")</span></span>
<span id="cb112-13"><a href="#cb112-13" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb112-14"><a href="#cb112-14" aria-hidden="true" tabindex="-1"></a><span class="co"># t4 %&gt;% dfround(4) %&gt;% DT::datatable()</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
</section>
<section id="mechanistic-hypothesis" class="level1">
<h1>Mechanistic Hypothesis</h1>
<p>To address our secondary questions “Does the addition of green nest material by birds increase their reproductive success? (Courtship hypothesis)” and “Does the addition of green nest material have protective effects and health benefits for the nestlings? (Parental care hypothesis)”, we will make use of the unimoderator multilevel meta-regressions hypothesis type (levels: CH, PCH, Both) as the moderator.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [23]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a>lnRR_hypothesis <span class="ot">&lt;-</span></span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> lnRR_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb113-3"><a href="#cb113-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_lnRR,<span class="co"># specify lnRR's sampling variance;</span></span>
<span id="cb113-4"><a href="#cb113-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="sc">-</span> <span class="dv">1</span> <span class="sc">+</span> Hypothesis,</span>
<span id="cb113-5"><a href="#cb113-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb113-6"><a href="#cb113-6" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb113-7"><a href="#cb113-7" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb113-8"><a href="#cb113-8" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb113-9"><a href="#cb113-9" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb113-10"><a href="#cb113-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb113-11"><a href="#cb113-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb113-12"><a href="#cb113-12" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb113-13"><a href="#cb113-13" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_lnRR)</span>
<span id="cb113-14"><a href="#cb113-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Save model</span></span>
<span id="cb113-15"><a href="#cb113-15" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(lnRR_hypothesis, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/lnRR_hypothesis.rds"</span>))</span>
<span id="cb113-16"><a href="#cb113-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-17"><a href="#cb113-17" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the dataset for this model</span></span>
<span id="cb113-18"><a href="#cb113-18" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(dataset_lnRR,here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/04_data_analysis/hypothesis_lnRR.csv"</span>))</span>
<span id="cb113-19"><a href="#cb113-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-20"><a href="#cb113-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-21"><a href="#cb113-21" aria-hidden="true" tabindex="-1"></a>lnRR_hypothesis_R2 <span class="ot">&lt;-</span> <span class="fu">r2_ml</span>(lnRR_hypothesis)</span>
<span id="cb113-22"><a href="#cb113-22" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(lnRR_hypothesis_R2 <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   R2_marginal R2_conditional 
           1.2            1.2 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb115"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a>SMDH_hypothesis <span class="ot">&lt;-</span></span>
<span id="cb115-2"><a href="#cb115-2" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb115-3"><a href="#cb115-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_SMDH,<span class="co"># specify lnRR's sampling variance;</span></span>
<span id="cb115-4"><a href="#cb115-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="sc">-</span> <span class="dv">1</span> <span class="sc">+</span> Hypothesis,</span>
<span id="cb115-5"><a href="#cb115-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb115-6"><a href="#cb115-6" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb115-7"><a href="#cb115-7" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb115-8"><a href="#cb115-8" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb115-9"><a href="#cb115-9" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb115-10"><a href="#cb115-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb115-11"><a href="#cb115-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb115-12"><a href="#cb115-12" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb115-13"><a href="#cb115-13" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_SMDH)</span>
<span id="cb115-14"><a href="#cb115-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb115-15"><a href="#cb115-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Save model</span></span>
<span id="cb115-16"><a href="#cb115-16" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SMDH_hypothesis, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/SMDH_hypothesis.rds"</span>))</span>
<span id="cb115-17"><a href="#cb115-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb115-18"><a href="#cb115-18" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the dataset for this model</span></span>
<span id="cb115-19"><a href="#cb115-19" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(dataset_SMDH,here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/04_data_analysis/hypothesis_SMDH.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
<section id="meta-regressions" class="level1">
<h1>Meta-regressions</h1>
<p>For understanding the heterogeneity in our intercept-only model, we will run unimoderator multilevel meta-regressions (see section Exploratory Analyses in pre-registration or manuscript) using the following moderators:</p>
<section id="type-of-experimental-design" class="level2">
<h2 class="anchored" data-anchor-id="type-of-experimental-design">Type of experimental design</h2>
<p>(levels: 1 = non-aromatic vs.&nbsp;aromatic, 2 = no added material vs.&nbsp;aromatic, 3 = no added material vs.&nbsp;non-aromatic).</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [24]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a><span class="co"># This analysis only includes dataset that has a proxy relating to comaparision type</span></span>
<span id="cb116-2"><a href="#cb116-2" aria-hidden="true" tabindex="-1"></a>dataset_lnRR_design <span class="ot">&lt;-</span> dataset_lnRR <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(comparision_type))</span>
<span id="cb116-3"><a href="#cb116-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-4"><a href="#cb116-4" aria-hidden="true" tabindex="-1"></a>VCV_lnRR_design <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> lnRR_variance,</span>
<span id="cb116-5"><a href="#cb116-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb116-6"><a href="#cb116-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb116-7"><a href="#cb116-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_lnRR_design,</span>
<span id="cb116-8"><a href="#cb116-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb116-9"><a href="#cb116-9" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb116-10"><a href="#cb116-10" aria-hidden="true" tabindex="-1"></a>lnRR_design <span class="ot">&lt;-</span></span>
<span id="cb116-11"><a href="#cb116-11" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> lnRR_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb116-12"><a href="#cb116-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_lnRR_design ,<span class="co"># specify lnRR's sampling variance;</span></span>
<span id="cb116-13"><a href="#cb116-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="sc">-</span> <span class="dv">1</span> <span class="sc">+</span> comparision_type,</span>
<span id="cb116-14"><a href="#cb116-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb116-15"><a href="#cb116-15" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb116-16"><a href="#cb116-16" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb116-17"><a href="#cb116-17" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb116-18"><a href="#cb116-18" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb116-19"><a href="#cb116-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb116-20"><a href="#cb116-20" aria-hidden="true" tabindex="-1"></a>                  <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb116-21"><a href="#cb116-21" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb116-22"><a href="#cb116-22" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_lnRR_design )</span>
<span id="cb116-23"><a href="#cb116-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-24"><a href="#cb116-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the model</span></span>
<span id="cb116-25"><a href="#cb116-25" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(lnRR_design, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/lnRR_design.rds"</span>))</span>
<span id="cb116-26"><a href="#cb116-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-27"><a href="#cb116-27" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the dataset for this model</span></span>
<span id="cb116-28"><a href="#cb116-28" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(dataset_lnRR_design,here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/04_data_analysis/design_lnRR.csv"</span>))</span>
<span id="cb116-29"><a href="#cb116-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-30"><a href="#cb116-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-31"><a href="#cb116-31" aria-hidden="true" tabindex="-1"></a><span class="co"># This analysis only includes dataset that has a proxy relating to comaparision type</span></span>
<span id="cb116-32"><a href="#cb116-32" aria-hidden="true" tabindex="-1"></a>dataset_SMDH_design <span class="ot">&lt;-</span> dataset_SMDH <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(comparision_type))</span>
<span id="cb116-33"><a href="#cb116-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-34"><a href="#cb116-34" aria-hidden="true" tabindex="-1"></a>VCV_SMDH_design <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> SMDH_variance,</span>
<span id="cb116-35"><a href="#cb116-35" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb116-36"><a href="#cb116-36" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb116-37"><a href="#cb116-37" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_SMDH_design,</span>
<span id="cb116-38"><a href="#cb116-38" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb116-39"><a href="#cb116-39" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb116-40"><a href="#cb116-40" aria-hidden="true" tabindex="-1"></a>SMDH_design <span class="ot">&lt;-</span></span>
<span id="cb116-41"><a href="#cb116-41" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb116-42"><a href="#cb116-42" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_SMDH_design ,<span class="co"># specify lnRR's sampling variance;</span></span>
<span id="cb116-43"><a href="#cb116-43" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="sc">-</span> <span class="dv">1</span> <span class="sc">+</span> comparision_type,</span>
<span id="cb116-44"><a href="#cb116-44" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb116-45"><a href="#cb116-45" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb116-46"><a href="#cb116-46" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb116-47"><a href="#cb116-47" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb116-48"><a href="#cb116-48" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb116-49"><a href="#cb116-49" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb116-50"><a href="#cb116-50" aria-hidden="true" tabindex="-1"></a>                  <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb116-51"><a href="#cb116-51" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb116-52"><a href="#cb116-52" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_SMDH_design )</span>
<span id="cb116-53"><a href="#cb116-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-54"><a href="#cb116-54" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the model</span></span>
<span id="cb116-55"><a href="#cb116-55" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SMDH_design, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/SMDH_design.rds"</span>))</span>
<span id="cb116-56"><a href="#cb116-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-57"><a href="#cb116-57" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the dataset for this model</span></span>
<span id="cb116-58"><a href="#cb116-58" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(dataset_SMDH_design,here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/04_data_analysis/design_SMDH.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<p>Let me try to do a sensitivity analysis to see what happens when I remove the control condition of 3 in the overall model since that is expected to not have an effect according to the authors as well.. i.e.&nbsp;it is a double control..</p>
<section id="excluding-non-aromatic-and-blank-comparison-i.e.-the-two-control-group-comparison" class="level3">
<h3 class="anchored" data-anchor-id="excluding-non-aromatic-and-blank-comparison-i.e.-the-two-control-group-comparison">Excluding non-aromatic and blank comparison (i.e.&nbsp;the two control group comparison)</h3>
<div class="cell-container"><div class="cell-decorator"><pre>In [25]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb117"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a><span class="do">## For the lnRR effect size...</span></span>
<span id="cb117-2"><a href="#cb117-2" aria-hidden="true" tabindex="-1"></a>dataset_lnRR_no_comp3<span class="ot">&lt;-</span>dataset_lnRR<span class="sc">%&gt;%</span><span class="fu">filter</span>(comparision_type<span class="sc">!=</span><span class="dv">3</span>)</span>
<span id="cb117-3"><a href="#cb117-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Only some rows are removed... </span></span>
<span id="cb117-4"><a href="#cb117-4" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(dataset_lnRR)<span class="sc">-</span><span class="fu">nrow</span>(dataset_lnRR_no_comp3)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 30</code></pre>
</div>
<div class="sourceCode cell-code" id="cb119"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a>VCV_lnRR_no_comp3 <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> lnRR_variance,</span>
<span id="cb119-2"><a href="#cb119-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb119-3"><a href="#cb119-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb119-4"><a href="#cb119-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_lnRR_no_comp3,</span>
<span id="cb119-5"><a href="#cb119-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb119-6"><a href="#cb119-6" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb119-7"><a href="#cb119-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-8"><a href="#cb119-8" aria-hidden="true" tabindex="-1"></a>SA_lnRR_no_comp3<span class="ot">&lt;-</span> <span class="fu">rma.mv</span>(<span class="at">yi =</span> lnRR_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb119-9"><a href="#cb119-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_lnRR_no_comp3, <span class="co"># specify lnRR's sampling variance-covariance matrix;</span></span>
<span id="cb119-10"><a href="#cb119-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb119-11"><a href="#cb119-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb119-12"><a href="#cb119-12" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb119-13"><a href="#cb119-13" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb119-14"><a href="#cb119-14" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb119-15"><a href="#cb119-15" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb119-16"><a href="#cb119-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb119-17"><a href="#cb119-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb119-18"><a href="#cb119-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb119-19"><a href="#cb119-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> dataset_lnRR_no_comp3)</span>
<span id="cb119-20"><a href="#cb119-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-21"><a href="#cb119-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-22"><a href="#cb119-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-23"><a href="#cb119-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the model</span></span>
<span id="cb119-24"><a href="#cb119-24" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SA_lnRR_no_comp3, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/sensitivity_analysis/SA_lnRR_no_comp3.rds"</span>))</span>
<span id="cb119-25"><a href="#cb119-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-26"><a href="#cb119-26" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the dataset for this model</span></span>
<span id="cb119-27"><a href="#cb119-27" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(dataset_lnRR_no_comp3,here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/05_data_sensitivity_analysis/dataset_lnRR_no_comp3.csv"</span>))</span>
<span id="cb119-28"><a href="#cb119-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-29"><a href="#cb119-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-30"><a href="#cb119-30" aria-hidden="true" tabindex="-1"></a><span class="co">#----------------------------------------------------------------------------</span></span>
<span id="cb119-31"><a href="#cb119-31" aria-hidden="true" tabindex="-1"></a><span class="co"># for  SMDH  as Effect Size</span></span>
<span id="cb119-32"><a href="#cb119-32" aria-hidden="true" tabindex="-1"></a><span class="co">#----------------------------------------------------------------------------</span></span>
<span id="cb119-33"><a href="#cb119-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-34"><a href="#cb119-34" aria-hidden="true" tabindex="-1"></a>dataset_SMDH_no_comp3<span class="ot">&lt;-</span>dataset_SMDH<span class="sc">%&gt;%</span><span class="fu">filter</span>(comparision_type<span class="sc">!=</span><span class="dv">3</span>)</span>
<span id="cb119-35"><a href="#cb119-35" aria-hidden="true" tabindex="-1"></a><span class="do">## Only some rows are removed... </span></span>
<span id="cb119-36"><a href="#cb119-36" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(dataset_SMDH)<span class="sc">-</span><span class="fu">nrow</span>(dataset_SMDH_no_comp3)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 27</code></pre>
</div>
<div class="sourceCode cell-code" id="cb121"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a>VCV_SMDH_no_comp3 <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> SMDH_variance,</span>
<span id="cb121-2"><a href="#cb121-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb121-3"><a href="#cb121-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb121-4"><a href="#cb121-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_SMDH_no_comp3,</span>
<span id="cb121-5"><a href="#cb121-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb121-6"><a href="#cb121-6" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb121-7"><a href="#cb121-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-8"><a href="#cb121-8" aria-hidden="true" tabindex="-1"></a>SA_SMDH_no_comp3<span class="ot">&lt;-</span> <span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb121-9"><a href="#cb121-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_SMDH_no_comp3, <span class="co"># specify lnRR's sampling variance-covariance matrix;</span></span>
<span id="cb121-10"><a href="#cb121-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb121-11"><a href="#cb121-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb121-12"><a href="#cb121-12" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb121-13"><a href="#cb121-13" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb121-14"><a href="#cb121-14" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb121-15"><a href="#cb121-15" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb121-16"><a href="#cb121-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb121-17"><a href="#cb121-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb121-18"><a href="#cb121-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb121-19"><a href="#cb121-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> dataset_SMDH_no_comp3)</span>
<span id="cb121-20"><a href="#cb121-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-21"><a href="#cb121-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-22"><a href="#cb121-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-23"><a href="#cb121-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the model</span></span>
<span id="cb121-24"><a href="#cb121-24" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SA_SMDH_no_comp3, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/sensitivity_analysis/SA_SMDH_no_comp3.rds"</span>))</span>
<span id="cb121-25"><a href="#cb121-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-26"><a href="#cb121-26" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the dataset for this model</span></span>
<span id="cb121-27"><a href="#cb121-27" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(dataset_SMDH_no_comp3,here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/05_data_sensitivity_analysis/dataset_SMDH_no_comp3.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<p>Results remain practically the same irrespective..</p>
</section>
</section>
<section id="time-of-addition-of-green-nest-material" class="level2">
<h2 class="anchored" data-anchor-id="time-of-addition-of-green-nest-material">Time of addition of green nest material</h2>
<p>(levels: before egg hatching, after egg hatching, continuously throughout the nesting phase).</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [26]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb122"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a><span class="do">## We use the complete dataset in this case</span></span>
<span id="cb122-2"><a href="#cb122-2" aria-hidden="true" tabindex="-1"></a>lnRR_time_gnm_addition <span class="ot">&lt;-</span></span>
<span id="cb122-3"><a href="#cb122-3" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> lnRR_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb122-4"><a href="#cb122-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_lnRR,<span class="co"># specify lnRR's sampling variance using the variance-covariance matrix we created for the intercept-only model since the whole dataset is being used in this case as well</span></span>
<span id="cb122-5"><a href="#cb122-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="sc">-</span> <span class="dv">1</span> <span class="sc">+</span> time_of_gnm_addition,</span>
<span id="cb122-6"><a href="#cb122-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb122-7"><a href="#cb122-7" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb122-8"><a href="#cb122-8" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb122-9"><a href="#cb122-9" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb122-10"><a href="#cb122-10" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb122-11"><a href="#cb122-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb122-12"><a href="#cb122-12" aria-hidden="true" tabindex="-1"></a>                  <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb122-13"><a href="#cb122-13" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb122-14"><a href="#cb122-14" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_lnRR)</span>
<span id="cb122-15"><a href="#cb122-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-16"><a href="#cb122-16" aria-hidden="true" tabindex="-1"></a><span class="do">## Saving the model</span></span>
<span id="cb122-17"><a href="#cb122-17" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(lnRR_time_gnm_addition, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/lnRR_time_gnm_addition.rds"</span>))</span>
<span id="cb122-18"><a href="#cb122-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-19"><a href="#cb122-19" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the dataset for this model</span></span>
<span id="cb122-20"><a href="#cb122-20" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(dataset_lnRR,here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/04_data_analysis/time_gnm_lnRR.csv"</span>))</span>
<span id="cb122-21"><a href="#cb122-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-22"><a href="#cb122-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-23"><a href="#cb122-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-24"><a href="#cb122-24" aria-hidden="true" tabindex="-1"></a><span class="do">## Now similarly for the SMDH effect size using the complete dataset</span></span>
<span id="cb122-25"><a href="#cb122-25" aria-hidden="true" tabindex="-1"></a>SMDH_time_gnm_addition <span class="ot">&lt;-</span></span>
<span id="cb122-26"><a href="#cb122-26" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb122-27"><a href="#cb122-27" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_SMDH,<span class="co"># specify lnRR's sampling variance using the variance-covariance matrix we created for the intercept-only model since the whole dataset is being used in this case as well</span></span>
<span id="cb122-28"><a href="#cb122-28" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="sc">-</span> <span class="dv">1</span> <span class="sc">+</span> time_of_gnm_addition,</span>
<span id="cb122-29"><a href="#cb122-29" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb122-30"><a href="#cb122-30" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb122-31"><a href="#cb122-31" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb122-32"><a href="#cb122-32" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb122-33"><a href="#cb122-33" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb122-34"><a href="#cb122-34" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb122-35"><a href="#cb122-35" aria-hidden="true" tabindex="-1"></a>                  <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb122-36"><a href="#cb122-36" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb122-37"><a href="#cb122-37" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_SMDH)</span>
<span id="cb122-38"><a href="#cb122-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-39"><a href="#cb122-39" aria-hidden="true" tabindex="-1"></a><span class="do">## Saving the model</span></span>
<span id="cb122-40"><a href="#cb122-40" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SMDH_time_gnm_addition, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/SMDH_time_gnm_addition.rds"</span>))</span>
<span id="cb122-41"><a href="#cb122-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-42"><a href="#cb122-42" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the dataset for this model</span></span>
<span id="cb122-43"><a href="#cb122-43" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(dataset_SMDH,here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/04_data_analysis/time_gnm_SMDH.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
<section id="type-of-parasites" class="level2">
<h2 class="anchored" data-anchor-id="type-of-parasites">Type of parasites</h2>
<p>(levels: arthropods, micro-organisms)&nbsp;</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [27]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb123"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1" aria-hidden="true" tabindex="-1"></a><span class="co"># This analysis only includes dataset that has a proxy relating to parasite or pathogen</span></span>
<span id="cb123-2"><a href="#cb123-2" aria-hidden="true" tabindex="-1"></a>dataset_lnRR_parasite <span class="ot">&lt;-</span> dataset_lnRR<span class="sc">%&gt;%</span></span>
<span id="cb123-3"><a href="#cb123-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(parasite_type))<span class="sc">%&gt;%</span></span>
<span id="cb123-4"><a href="#cb123-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">parasite_type=</span><span class="fu">as.factor</span>(parasite_type),</span>
<span id="cb123-5"><a href="#cb123-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">paper_ID=</span><span class="fu">as.factor</span>(paper_ID),</span>
<span id="cb123-6"><a href="#cb123-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">Observation_ID=</span><span class="fu">as.factor</span>(Observation_ID),</span>
<span id="cb123-7"><a href="#cb123-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">group_ID_coded=</span><span class="fu">as.factor</span>(group_ID_coded),</span>
<span id="cb123-8"><a href="#cb123-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">bird_species=</span><span class="fu">as.factor</span>(bird_species))</span>
<span id="cb123-9"><a href="#cb123-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-10"><a href="#cb123-10" aria-hidden="true" tabindex="-1"></a>VCV_lnRR_parasite <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> lnRR_variance,</span>
<span id="cb123-11"><a href="#cb123-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb123-12"><a href="#cb123-12" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb123-13"><a href="#cb123-13" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_lnRR_parasite,</span>
<span id="cb123-14"><a href="#cb123-14" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb123-15"><a href="#cb123-15" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb123-16"><a href="#cb123-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-17"><a href="#cb123-17" aria-hidden="true" tabindex="-1"></a>lnRR_parasite <span class="ot">&lt;-</span></span>
<span id="cb123-18"><a href="#cb123-18" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> lnRR_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb123-19"><a href="#cb123-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_lnRR_parasite,<span class="co"># specify lnRR's sampling variance;</span></span>
<span id="cb123-20"><a href="#cb123-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="sc">-</span> <span class="dv">1</span> <span class="sc">+</span> parasite_type,</span>
<span id="cb123-21"><a href="#cb123-21" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb123-22"><a href="#cb123-22" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb123-23"><a href="#cb123-23" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb123-24"><a href="#cb123-24" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb123-25"><a href="#cb123-25" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb123-26"><a href="#cb123-26" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb123-27"><a href="#cb123-27" aria-hidden="true" tabindex="-1"></a>                   <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb123-28"><a href="#cb123-28" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb123-29"><a href="#cb123-29" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_lnRR_parasite)</span>
<span id="cb123-30"><a href="#cb123-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-31"><a href="#cb123-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-32"><a href="#cb123-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the model</span></span>
<span id="cb123-33"><a href="#cb123-33" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(lnRR_parasite, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/lnRR_parasite.rds"</span>))</span>
<span id="cb123-34"><a href="#cb123-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-35"><a href="#cb123-35" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the dataset for this model</span></span>
<span id="cb123-36"><a href="#cb123-36" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(dataset_lnRR_parasite,here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/04_data_analysis/parasite_lnRR.csv"</span>))</span>
<span id="cb123-37"><a href="#cb123-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-38"><a href="#cb123-38" aria-hidden="true" tabindex="-1"></a><span class="do">### Similiarly for SMDH as well selecting the dataset with parasite type</span></span>
<span id="cb123-39"><a href="#cb123-39" aria-hidden="true" tabindex="-1"></a><span class="co"># This analysis only includes dataset that has a proxy relating to parasite or pathogen</span></span>
<span id="cb123-40"><a href="#cb123-40" aria-hidden="true" tabindex="-1"></a>dataset_SMDH_parasite <span class="ot">&lt;-</span> dataset_SMDH<span class="sc">%&gt;%</span></span>
<span id="cb123-41"><a href="#cb123-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(parasite_type))</span>
<span id="cb123-42"><a href="#cb123-42" aria-hidden="true" tabindex="-1"></a>VCV_SMDH_parasite <span class="ot">&lt;-</span> <span class="fu">vcalc</span>(<span class="at">vi =</span> SMDH_variance,</span>
<span id="cb123-43"><a href="#cb123-43" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> paper_ID,</span>
<span id="cb123-44"><a href="#cb123-44" aria-hidden="true" tabindex="-1"></a>                  <span class="at">obs =</span> Observation_ID,</span>
<span id="cb123-45"><a href="#cb123-45" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_SMDH_parasite,</span>
<span id="cb123-46"><a href="#cb123-46" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rho =</span> <span class="fl">0.5</span> <span class="co"># assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb123-47"><a href="#cb123-47" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb123-48"><a href="#cb123-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-49"><a href="#cb123-49" aria-hidden="true" tabindex="-1"></a>SMDH_parasite <span class="ot">&lt;-</span></span>
<span id="cb123-50"><a href="#cb123-50" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb123-51"><a href="#cb123-51" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_SMDH_parasite,<span class="co"># specify lnRR's sampling variance;</span></span>
<span id="cb123-52"><a href="#cb123-52" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="sc">-</span> <span class="dv">1</span> <span class="sc">+</span> parasite_type,</span>
<span id="cb123-53"><a href="#cb123-53" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb123-54"><a href="#cb123-54" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb123-55"><a href="#cb123-55" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb123-56"><a href="#cb123-56" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb123-57"><a href="#cb123-57" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb123-58"><a href="#cb123-58" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb123-59"><a href="#cb123-59" aria-hidden="true" tabindex="-1"></a>                   <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb123-60"><a href="#cb123-60" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb123-61"><a href="#cb123-61" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_SMDH_parasite)</span>
<span id="cb123-62"><a href="#cb123-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-63"><a href="#cb123-63" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the model</span></span>
<span id="cb123-64"><a href="#cb123-64" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SMDH_parasite, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/SMDH_parasite.rds"</span>))</span>
<span id="cb123-65"><a href="#cb123-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-66"><a href="#cb123-66" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the dataset for this model</span></span>
<span id="cb123-67"><a href="#cb123-67" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(dataset_SMDH_parasite,here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/04_data_analysis/parasite_SMDH.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<section id="bird-species" class="level4">
<h4 class="anchored" data-anchor-id="bird-species">Bird species</h4>
<p>(levels: Cyanistes caeruleus, Sturnus unicolor, Tachycineta bicolor, Sturnus vulgaris; note that these levels reflect the list of species studied in our current database, which may increase after updating our search and/or receiving unpublished data from authors)</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [28]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb124"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1" aria-hidden="true" tabindex="-1"></a>lnRR_birds <span class="ot">&lt;-</span></span>
<span id="cb124-2"><a href="#cb124-2" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> lnRR_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb124-3"><a href="#cb124-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_lnRR,<span class="co"># specify lnRR's sampling variance;</span></span>
<span id="cb124-4"><a href="#cb124-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="sc">-</span> <span class="dv">1</span> <span class="sc">+</span> bird_species,</span>
<span id="cb124-5"><a href="#cb124-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb124-6"><a href="#cb124-6" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb124-7"><a href="#cb124-7" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb124-8"><a href="#cb124-8" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb124-9"><a href="#cb124-9" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb124-10"><a href="#cb124-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb124-11"><a href="#cb124-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb124-12"><a href="#cb124-12" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb124-13"><a href="#cb124-13" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_lnRR)</span>
<span id="cb124-14"><a href="#cb124-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-15"><a href="#cb124-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the model</span></span>
<span id="cb124-16"><a href="#cb124-16" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(lnRR_birds, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/lnRR_birds.rds"</span>))</span>
<span id="cb124-17"><a href="#cb124-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-18"><a href="#cb124-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the dataset for this model</span></span>
<span id="cb124-19"><a href="#cb124-19" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(dataset_lnRR,here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/04_data_analysis/birds_lnRR.csv"</span>))</span>
<span id="cb124-20"><a href="#cb124-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-21"><a href="#cb124-21" aria-hidden="true" tabindex="-1"></a><span class="do">## Now doing this for SMDH</span></span>
<span id="cb124-22"><a href="#cb124-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-23"><a href="#cb124-23" aria-hidden="true" tabindex="-1"></a>SMDH_birds <span class="ot">&lt;-</span></span>
<span id="cb124-24"><a href="#cb124-24" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb124-25"><a href="#cb124-25" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_SMDH,<span class="co"># specify lnRR's sampling variance;</span></span>
<span id="cb124-26"><a href="#cb124-26" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="sc">-</span> <span class="dv">1</span> <span class="sc">+</span> bird_species,</span>
<span id="cb124-27"><a href="#cb124-27" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb124-28"><a href="#cb124-28" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb124-29"><a href="#cb124-29" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb124-30"><a href="#cb124-30" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb124-31"><a href="#cb124-31" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb124-32"><a href="#cb124-32" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb124-33"><a href="#cb124-33" aria-hidden="true" tabindex="-1"></a>                  <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb124-34"><a href="#cb124-34" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb124-35"><a href="#cb124-35" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_SMDH)</span>
<span id="cb124-36"><a href="#cb124-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-37"><a href="#cb124-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the model</span></span>
<span id="cb124-38"><a href="#cb124-38" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SMDH_birds, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/SMDH_birds.rds"</span>))</span>
<span id="cb124-39"><a href="#cb124-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-40"><a href="#cb124-40" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the dataset for this model</span></span>
<span id="cb124-41"><a href="#cb124-41" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(dataset_SMDH,here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/04_data_analysis/birds_SMDH.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
<section id="type-of-trait-studied" class="level4">
<h4 class="anchored" data-anchor-id="type-of-trait-studied">Type of trait studied</h4>
<p>(levels:physiology, morphology, reproduction, behaviour, parasite and pathogenic load, phenology)</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [29]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb125"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1" aria-hidden="true" tabindex="-1"></a>lnRR_trait <span class="ot">&lt;-</span></span>
<span id="cb125-2"><a href="#cb125-2" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> lnRR_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb125-3"><a href="#cb125-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_lnRR,<span class="co"># specify lnRR's sampling variance;</span></span>
<span id="cb125-4"><a href="#cb125-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="sc">-</span> <span class="dv">1</span> <span class="sc">+</span> trait_type,</span>
<span id="cb125-5"><a href="#cb125-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb125-6"><a href="#cb125-6" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb125-7"><a href="#cb125-7" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb125-8"><a href="#cb125-8" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb125-9"><a href="#cb125-9" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb125-10"><a href="#cb125-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb125-11"><a href="#cb125-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb125-12"><a href="#cb125-12" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb125-13"><a href="#cb125-13" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_lnRR)</span>
<span id="cb125-14"><a href="#cb125-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-15"><a href="#cb125-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the model</span></span>
<span id="cb125-16"><a href="#cb125-16" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(lnRR_trait, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/lnRR_trait.rds"</span>))</span>
<span id="cb125-17"><a href="#cb125-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-18"><a href="#cb125-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the dataset for this model</span></span>
<span id="cb125-19"><a href="#cb125-19" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(dataset_lnRR,here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/04_data_analysis/trait_lnRR.csv"</span>))</span>
<span id="cb125-20"><a href="#cb125-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-21"><a href="#cb125-21" aria-hidden="true" tabindex="-1"></a><span class="do">## Now similarly for SMDH</span></span>
<span id="cb125-22"><a href="#cb125-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-23"><a href="#cb125-23" aria-hidden="true" tabindex="-1"></a>SMDH_trait <span class="ot">&lt;-</span></span>
<span id="cb125-24"><a href="#cb125-24" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb125-25"><a href="#cb125-25" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_SMDH,<span class="co"># specify lnRR's sampling variance;</span></span>
<span id="cb125-26"><a href="#cb125-26" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> <span class="sc">-</span> <span class="dv">1</span> <span class="sc">+</span> trait_type,</span>
<span id="cb125-27"><a href="#cb125-27" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb125-28"><a href="#cb125-28" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb125-29"><a href="#cb125-29" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb125-30"><a href="#cb125-30" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb125-31"><a href="#cb125-31" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb125-32"><a href="#cb125-32" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb125-33"><a href="#cb125-33" aria-hidden="true" tabindex="-1"></a>                  <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb125-34"><a href="#cb125-34" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb125-35"><a href="#cb125-35" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_SMDH)</span>
<span id="cb125-36"><a href="#cb125-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-37"><a href="#cb125-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the model</span></span>
<span id="cb125-38"><a href="#cb125-38" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SMDH_trait, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/SMDH_trait.rds"</span>))</span>
<span id="cb125-39"><a href="#cb125-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-40"><a href="#cb125-40" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the dataset for this model</span></span>
<span id="cb125-41"><a href="#cb125-41" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(dataset_SMDH,here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/04_data_analysis/SMDH_lnRR.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
</section>
</section>
<section id="risk-of-bias" class="level1">
<h1>Risk of Bias</h1>
<p>6. Blinding: (Levels: yes, no(</p>
<p>7. Random Assignment: (Levels: yes, no)</p>
<p>8. Missing Data: (Levels: yes, no)</p>
<p>Note that the specific subset of data for each meta-regression may differ based on the data availability (e.g., information about the type of parasites is only available for effect sizes based on parasite/pathogen related fitness proxy). We will conduct meta-regression on suitable subsets of data for each analysis, provided we have sufficient effect sizes per moderator level (i.e., at least 5 effect sizes per moderator level).</p>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Phylogenetic Effects
</div>
</div>
<div class="callout-body-container callout-body">
<p>Note also that if we obtain additional effect sizes for other bird species after we update our search and/or contact authors, our models may need to account for phylogenetic nonindependence via the inclusion of two additional random effects: “species” and “phylogeny” following Cinar et al.&nbsp;(2022).</p>
</div>
</div>
<section id="other-analysis" class="level2">
<h2 class="anchored" data-anchor-id="other-analysis">Other Analysis</h2>
<section id="risk-of-bias-assessment" class="level3">
<h3 class="anchored" data-anchor-id="risk-of-bias-assessment">Risk of Bias Assessment</h3>
<ul>
<li>RoB - The outcome will be presented in the form of a <u><strong>summary table</strong></u>. We will conduct uni-moderator meta-regressions using these variables.</li>
</ul>
</section>
</section>
</section>
<section id="publication-bias" class="level1">
<h1>Publication Bias</h1>
<p>Small study effects:</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [30]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb126"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb126-1"><a href="#cb126-1" aria-hidden="true" tabindex="-1"></a>dataset_lnRR_published<span class="ot">&lt;-</span>dataset_lnRR_published<span class="sc">%&gt;%</span></span>
<span id="cb126-2"><a href="#cb126-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">effective_n_inv=</span><span class="dv">1</span><span class="sc">/</span>((<span class="dv">4</span><span class="sc">*</span>effective_n_experiment<span class="sc">*</span>effective_n_control)<span class="sc">/</span>(effective_n_experiment<span class="sc">+</span>effective_n_control)))<span class="sc">%&gt;%</span></span>
<span id="cb126-3"><a href="#cb126-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sqrt_effective_n_inv=</span><span class="fu">sqrt</span>(effective_n_inv))</span>
<span id="cb126-4"><a href="#cb126-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-5"><a href="#cb126-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-6"><a href="#cb126-6" aria-hidden="true" tabindex="-1"></a>lnRR_small_study <span class="ot">&lt;-</span></span>
<span id="cb126-7"><a href="#cb126-7" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> lnRR_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb126-8"><a href="#cb126-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_lnRR_published,<span class="co"># specify lnRR's sampling variance;</span></span>
<span id="cb126-9"><a href="#cb126-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span>sqrt_effective_n_inv,</span>
<span id="cb126-10"><a href="#cb126-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb126-11"><a href="#cb126-11" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb126-12"><a href="#cb126-12" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb126-13"><a href="#cb126-13" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb126-14"><a href="#cb126-14" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb126-15"><a href="#cb126-15" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb126-16"><a href="#cb126-16" aria-hidden="true" tabindex="-1"></a>                  <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb126-17"><a href="#cb126-17" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb126-18"><a href="#cb126-18" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_lnRR_published)</span>
<span id="cb126-19"><a href="#cb126-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-20"><a href="#cb126-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-21"><a href="#cb126-21" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the model</span></span>
<span id="cb126-22"><a href="#cb126-22" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(lnRR_small_study, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/lnRR_small_study.rds"</span>))</span>
<span id="cb126-23"><a href="#cb126-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-24"><a href="#cb126-24" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(lnRR_small_study)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Multivariate Meta-Analysis Model (k = 230; method: REML)

  logLik  Deviance       AIC       BIC      AICc   
-22.7945   45.5890   59.5890   83.5944   60.0981   

Variance Components:

            estim    sqrt  nlvls  fixed          factor 
sigma^2.1  0.0000  0.0001     24     no        paper_ID 
sigma^2.2  0.0244  0.1561    230     no  Observation_ID 
sigma^2.3  0.0000  0.0002     44     no  group_ID_coded 
sigma^2.4  0.0000  0.0000     14     no   population_ID 
sigma^2.5  0.0000  0.0000      6     no    bird_species 

Test for Residual Heterogeneity:
QE(df = 228) = 2296.2770, p-val &lt; .0001

Test of Moderators (coefficient 2):
F(df1 = 1, df2 = 228) = 0.8065, p-val = 0.3701

Model Results:

                      estimate      se     tval   df    pval    ci.lb   ci.ub 
intrcpt                -0.0060  0.0301  -0.1991  228  0.8423  -0.0653  0.0533 
sqrt_effective_n_inv    0.1035  0.1152   0.8981  228  0.3701  -0.1236  0.3305 
                        
intrcpt                 
sqrt_effective_n_inv    

---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb128"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb128-1"><a href="#cb128-1" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(<span class="fu">r2_ml</span>(lnRR_small_study)<span class="sc">*</span><span class="dv">100</span>,<span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   R2_marginal R2_conditional 
          0.87           0.87 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb130"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb130-1"><a href="#cb130-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Now for SMDH model</span></span>
<span id="cb130-2"><a href="#cb130-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-3"><a href="#cb130-3" aria-hidden="true" tabindex="-1"></a>dataset_SMDH_published<span class="ot">&lt;-</span>dataset_SMDH_published<span class="sc">%&gt;%</span></span>
<span id="cb130-4"><a href="#cb130-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">effective_n_inv=</span><span class="dv">1</span><span class="sc">/</span>((<span class="dv">4</span><span class="sc">*</span>effective_n_experiment<span class="sc">*</span>effective_n_control)<span class="sc">/</span>(effective_n_experiment<span class="sc">+</span>effective_n_control)))<span class="sc">%&gt;%</span></span>
<span id="cb130-5"><a href="#cb130-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sqrt_effective_n_inv=</span><span class="fu">sqrt</span>(effective_n_inv))</span>
<span id="cb130-6"><a href="#cb130-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-7"><a href="#cb130-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-8"><a href="#cb130-8" aria-hidden="true" tabindex="-1"></a>SMDH_small_study <span class="ot">&lt;-</span></span>
<span id="cb130-9"><a href="#cb130-9" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb130-10"><a href="#cb130-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_SMDH_published,<span class="co"># specify lnRR's sampling variance;</span></span>
<span id="cb130-11"><a href="#cb130-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span>sqrt_effective_n_inv,</span>
<span id="cb130-12"><a href="#cb130-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb130-13"><a href="#cb130-13" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb130-14"><a href="#cb130-14" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb130-15"><a href="#cb130-15" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb130-16"><a href="#cb130-16" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb130-17"><a href="#cb130-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb130-18"><a href="#cb130-18" aria-hidden="true" tabindex="-1"></a>                  <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb130-19"><a href="#cb130-19" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb130-20"><a href="#cb130-20" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_SMDH_published)</span>
<span id="cb130-21"><a href="#cb130-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-22"><a href="#cb130-22" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the model</span></span>
<span id="cb130-23"><a href="#cb130-23" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SMDH_small_study, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/SMDH_small_study.rds"</span>))</span>
<span id="cb130-24"><a href="#cb130-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-25"><a href="#cb130-25" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(SMDH_small_study)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Multivariate Meta-Analysis Model (k = 235; method: REML)

   logLik   Deviance        AIC        BIC       AICc   
-305.6554   611.3107   625.3107   649.4680   625.8085   

Variance Components:

            estim    sqrt  nlvls  fixed          factor 
sigma^2.1  0.0000  0.0009     24     no        paper_ID 
sigma^2.2  0.2721  0.5216    235     no  Observation_ID 
sigma^2.3  0.0000  0.0005     45     no  group_ID_coded 
sigma^2.4  0.0000  0.0002     14     no   population_ID 
sigma^2.5  0.0000  0.0001      6     no    bird_species 

Test for Residual Heterogeneity:
QE(df = 233) = 1089.6262, p-val &lt; .0001

Test of Moderators (coefficient 2):
F(df1 = 1, df2 = 233) = 3.1936, p-val = 0.0752

Model Results:

                      estimate      se     tval   df    pval    ci.lb   ci.ub 
intrcpt                -0.0348  0.1441  -0.2417  233  0.8092  -0.3187  0.2491 
sqrt_effective_n_inv    1.3931  0.7795   1.7871  233  0.0752  -0.1428  2.9289 
                        
intrcpt                 
sqrt_effective_n_inv  . 

---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb132"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb132-1"><a href="#cb132-1" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(<span class="fu">r2_ml</span>(SMDH_small_study)<span class="sc">*</span><span class="dv">100</span>,<span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   R2_marginal R2_conditional 
          8.31           8.31 </code></pre>
</div>
</div></div>
<p>Decline effects (Time lag bias)</p>
<p>Next, run a uni-moderator meta-regression to test for evidence of decline effects (also known as time-lag bias) which includes year of publication (mean-centered) as the only moderator, and whose slope would provide information about whether the overall effect size has changed (decline) over time (Nakagawa et al.&nbsp;2022; Sánchez-Tójar et al.&nbsp;2018).</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [31]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb134"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb134-1"><a href="#cb134-1" aria-hidden="true" tabindex="-1"></a>dataset_lnRR_published<span class="ot">&lt;-</span>dataset_lnRR_published<span class="sc">%&gt;%</span></span>
<span id="cb134-2"><a href="#cb134-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">year_MeanCenter=</span><span class="fu">scale</span>(<span class="fu">as.numeric</span>(year_publication),<span class="at">scale=</span>F)[,<span class="dv">1</span>])</span>
<span id="cb134-3"><a href="#cb134-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-4"><a href="#cb134-4" aria-hidden="true" tabindex="-1"></a>lnRR_decline_eff <span class="ot">&lt;-</span></span>
<span id="cb134-5"><a href="#cb134-5" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> lnRR_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb134-6"><a href="#cb134-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_lnRR_published,<span class="co"># specify lnRR's sampling variance;</span></span>
<span id="cb134-7"><a href="#cb134-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> year_MeanCenter,</span>
<span id="cb134-8"><a href="#cb134-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb134-9"><a href="#cb134-9" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb134-10"><a href="#cb134-10" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb134-11"><a href="#cb134-11" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb134-12"><a href="#cb134-12" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb134-13"><a href="#cb134-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb134-14"><a href="#cb134-14" aria-hidden="true" tabindex="-1"></a>                  <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb134-15"><a href="#cb134-15" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb134-16"><a href="#cb134-16" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_lnRR_published)</span>
<span id="cb134-17"><a href="#cb134-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-18"><a href="#cb134-18" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the model</span></span>
<span id="cb134-19"><a href="#cb134-19" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(lnRR_decline_eff, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/lnRR_decline_eff.rds"</span>))</span>
<span id="cb134-20"><a href="#cb134-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb134-21"><a href="#cb134-21" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(lnRR_decline_eff)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Multivariate Meta-Analysis Model (k = 230; method: REML)

  logLik  Deviance       AIC       BIC      AICc   
-22.8760   45.7520   59.7520   83.7574   60.2611   

Variance Components:

            estim    sqrt  nlvls  fixed          factor 
sigma^2.1  0.0000  0.0001     24     no        paper_ID 
sigma^2.2  0.0241  0.1553    230     no  Observation_ID 
sigma^2.3  0.0000  0.0002     44     no  group_ID_coded 
sigma^2.4  0.0000  0.0000     14     no   population_ID 
sigma^2.5  0.0000  0.0000      6     no    bird_species 

Test for Residual Heterogeneity:
QE(df = 228) = 2067.6159, p-val &lt; .0001

Test of Moderators (coefficient 2):
F(df1 = 1, df2 = 228) = 0.5880, p-val = 0.4440

Model Results:

                 estimate      se     tval   df    pval    ci.lb   ci.ub    
intrcpt            0.0177  0.0160   1.1018  228  0.2717  -0.0139  0.0493    
year_MeanCenter   -0.0011  0.0014  -0.7668  228  0.4440  -0.0039  0.0017    

---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb136"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb136-1"><a href="#cb136-1" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(<span class="fu">r2_ml</span>(lnRR_decline_eff)<span class="sc">*</span><span class="dv">100</span>,<span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   R2_marginal R2_conditional 
          0.68           0.68 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb138"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb138-1"><a href="#cb138-1" aria-hidden="true" tabindex="-1"></a><span class="do">## --- Now for SMDH -----</span></span>
<span id="cb138-2"><a href="#cb138-2" aria-hidden="true" tabindex="-1"></a>dataset_SMDH_published<span class="ot">&lt;-</span>dataset_SMDH_published<span class="sc">%&gt;%</span></span>
<span id="cb138-3"><a href="#cb138-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">year_MeanCenter=</span><span class="fu">scale</span>(<span class="fu">as.numeric</span>(year_publication),<span class="at">scale=</span>F)[,<span class="dv">1</span>])</span>
<span id="cb138-4"><a href="#cb138-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb138-5"><a href="#cb138-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb138-6"><a href="#cb138-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb138-7"><a href="#cb138-7" aria-hidden="true" tabindex="-1"></a>SMDH_decline_eff <span class="ot">&lt;-</span></span>
<span id="cb138-8"><a href="#cb138-8" aria-hidden="true" tabindex="-1"></a>  metafor<span class="sc">::</span><span class="fu">rma.mv</span>(<span class="at">yi =</span> SMDH_sign, <span class="co"># specify lnRR as the effect size measure;</span></span>
<span id="cb138-9"><a href="#cb138-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">V =</span> VCV_SMDH_published,<span class="co"># specify lnRR's sampling variance;</span></span>
<span id="cb138-10"><a href="#cb138-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mods =</span> <span class="sc">~</span> year_MeanCenter,</span>
<span id="cb138-11"><a href="#cb138-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">random =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> paper_ID,</span>
<span id="cb138-12"><a href="#cb138-12" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> Observation_ID,</span>
<span id="cb138-13"><a href="#cb138-13" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> group_ID_coded,</span>
<span id="cb138-14"><a href="#cb138-14" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> population_ID,</span>
<span id="cb138-15"><a href="#cb138-15" aria-hidden="true" tabindex="-1"></a>                                  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> bird_species),</span>
<span id="cb138-16"><a href="#cb138-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">"REML"</span>,</span>
<span id="cb138-17"><a href="#cb138-17" aria-hidden="true" tabindex="-1"></a>                  <span class="at">test =</span> <span class="st">"t"</span>,</span>
<span id="cb138-18"><a href="#cb138-18" aria-hidden="true" tabindex="-1"></a>                  <span class="at">control =</span> <span class="fu">list</span>(<span class="at">optimizer=</span><span class="st">"optim"</span>),</span>
<span id="cb138-19"><a href="#cb138-19" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> dataset_SMDH_published)</span>
<span id="cb138-20"><a href="#cb138-20" aria-hidden="true" tabindex="-1"></a><span class="co"># saving the model</span></span>
<span id="cb138-21"><a href="#cb138-21" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(SMDH_decline_eff, <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"model/SMDH_decline_eff.rds"</span>))</span>
<span id="cb138-22"><a href="#cb138-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb138-23"><a href="#cb138-23" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(SMDH_decline_eff)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Multivariate Meta-Analysis Model (k = 235; method: REML)

   logLik   Deviance        AIC        BIC       AICc   
-306.7002   613.4005   627.4005   651.5577   627.8982   

Variance Components:

            estim    sqrt  nlvls  fixed          factor 
sigma^2.1  0.0000  0.0014     24     no        paper_ID 
sigma^2.2  0.2768  0.5261    235     no  Observation_ID 
sigma^2.3  0.0000  0.0030     45     no  group_ID_coded 
sigma^2.4  0.0000  0.0011     14     no   population_ID 
sigma^2.5  0.0000  0.0000      6     no    bird_species 

Test for Residual Heterogeneity:
QE(df = 233) = 1091.5803, p-val &lt; .0001

Test of Moderators (coefficient 2):
F(df1 = 1, df2 = 233) = 1.2515, p-val = 0.2644

Model Results:

                 estimate      se     tval   df    pval    ci.lb   ci.ub     
intrcpt            0.1935  0.0690   2.8051  233  0.0055   0.0576  0.3293  ** 
year_MeanCenter   -0.0083  0.0074  -1.1187  233  0.2644  -0.0228  0.0063     

---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb140"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb140-1"><a href="#cb140-1" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(<span class="fu">r2_ml</span>(SMDH_decline_eff)<span class="sc">*</span><span class="dv">100</span>,<span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   R2_marginal R2_conditional 
          2.93           2.93 </code></pre>
</div>
</div></div>
<p>Last, run an all-in meta-regression including both the (square root of the) inverse of the ‘effective sample size’ and the mean-centered year of publication as well as all the other moderators tested in our study to estimate how much heterogeneity is explained by all moderators combined as well as to explore whether evidence remains similar after accounting for all the other moderators (see equation 29 in Nakagawa et al.&nbsp;2022 for more information about this approach).&nbsp;</p>
<section id="all-in-model-for-heterogeneity" class="level2">
<h2 class="anchored" data-anchor-id="all-in-model-for-heterogeneity">All-in Model for heterogeneity</h2>
<div class="cell-container"><div class="cell-decorator"><pre>In [32]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb142"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb142-1"><a href="#cb142-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ## For lnRR complete dataset</span></span>
<span id="cb142-2"><a href="#cb142-2" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb142-3"><a href="#cb142-3" aria-hidden="true" tabindex="-1"></a><span class="co"># ## I am changing the year of unpublished studies to 2025 which is to now, assuming that if they were to be published now, what would be the effect. I will keep in mind that in this model if I do find an evidence of publication bias over years, I should keep in mind it is non-independent of the unpublished studies</span></span>
<span id="cb142-4"><a href="#cb142-4" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb142-5"><a href="#cb142-5" aria-hidden="true" tabindex="-1"></a><span class="co"># dataset_lnRR_all_in&lt;-dataset_lnRR%&gt;%mutate(year_publication = case_when(is.na(year_publication) ~ 2025, TRUE ~ year_publication))%&gt;%</span></span>
<span id="cb142-6"><a href="#cb142-6" aria-hidden="true" tabindex="-1"></a><span class="co">#   mutate(year_MeanCenter=scale(as.numeric(year_publication),scale=F)[,1])%&gt;%</span></span>
<span id="cb142-7"><a href="#cb142-7" aria-hidden="true" tabindex="-1"></a><span class="co">#   mutate(effective_n_inv=1/((4*effective_n_experiment*effective_n_control)/(effective_n_experiment+effective_n_control)))%&gt;%</span></span>
<span id="cb142-8"><a href="#cb142-8" aria-hidden="true" tabindex="-1"></a><span class="co">#   mutate(sqrt_effective_n_inv=sqrt(effective_n_inv))%&gt;%</span></span>
<span id="cb142-9"><a href="#cb142-9" aria-hidden="true" tabindex="-1"></a><span class="co">#   mutate(comparision_type= case_when(is.na(comparision_type) ~ 4, TRUE ~ comparision_type))</span></span>
<span id="cb142-10"><a href="#cb142-10" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb142-11"><a href="#cb142-11" aria-hidden="true" tabindex="-1"></a><span class="co"># VCV_lnRR_all_in&lt;- vcalc(vi = lnRR_variance,</span></span>
<span id="cb142-12"><a href="#cb142-12" aria-hidden="true" tabindex="-1"></a><span class="co">#                   cluster = paper_ID,</span></span>
<span id="cb142-13"><a href="#cb142-13" aria-hidden="true" tabindex="-1"></a><span class="co">#                   obs = Observation_ID,</span></span>
<span id="cb142-14"><a href="#cb142-14" aria-hidden="true" tabindex="-1"></a><span class="co">#                   data = dataset_lnRR_all_in,</span></span>
<span id="cb142-15"><a href="#cb142-15" aria-hidden="true" tabindex="-1"></a><span class="co">#                   rho = 0.5 # assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb142-16"><a href="#cb142-16" aria-hidden="true" tabindex="-1"></a><span class="co">#                   )</span></span>
<span id="cb142-17"><a href="#cb142-17" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb142-18"><a href="#cb142-18" aria-hidden="true" tabindex="-1"></a><span class="co"># lnRR_all_in &lt;- rma.mv(yi = lnRR_sign, # specify lnRR as the effect size measure;</span></span>
<span id="cb142-19"><a href="#cb142-19" aria-hidden="true" tabindex="-1"></a><span class="co">#                     V = VCV_lnRR_all_in, # specify lnRR's sampling variance-covariance matrix;</span></span>
<span id="cb142-20"><a href="#cb142-20" aria-hidden="true" tabindex="-1"></a><span class="co">#                     mods = ~ time_of_gnm_addition + </span></span>
<span id="cb142-21"><a href="#cb142-21" aria-hidden="true" tabindex="-1"></a><span class="co">#                       year_MeanCenter + </span></span>
<span id="cb142-22"><a href="#cb142-22" aria-hidden="true" tabindex="-1"></a><span class="co">#                       sqrt_effective_n_inv + </span></span>
<span id="cb142-23"><a href="#cb142-23" aria-hidden="true" tabindex="-1"></a><span class="co">#                       as.factor(comparision_type) + </span></span>
<span id="cb142-24"><a href="#cb142-24" aria-hidden="true" tabindex="-1"></a><span class="co">#                       bird_species + </span></span>
<span id="cb142-25"><a href="#cb142-25" aria-hidden="true" tabindex="-1"></a><span class="co">#                       trait_type,</span></span>
<span id="cb142-26"><a href="#cb142-26" aria-hidden="true" tabindex="-1"></a><span class="co">#                     random = list(~ 1 | paper_ID,</span></span>
<span id="cb142-27"><a href="#cb142-27" aria-hidden="true" tabindex="-1"></a><span class="co">#                                   ~ 1 | Observation_ID,</span></span>
<span id="cb142-28"><a href="#cb142-28" aria-hidden="true" tabindex="-1"></a><span class="co">#                                   ~ 1 | group_ID_coded,</span></span>
<span id="cb142-29"><a href="#cb142-29" aria-hidden="true" tabindex="-1"></a><span class="co">#                                   ~ 1 | population_ID,</span></span>
<span id="cb142-30"><a href="#cb142-30" aria-hidden="true" tabindex="-1"></a><span class="co">#                                   ~ 1 | bird_species),</span></span>
<span id="cb142-31"><a href="#cb142-31" aria-hidden="true" tabindex="-1"></a><span class="co">#                     method = "REML",</span></span>
<span id="cb142-32"><a href="#cb142-32" aria-hidden="true" tabindex="-1"></a><span class="co">#                     control = list(optimizer="optim"),</span></span>
<span id="cb142-33"><a href="#cb142-33" aria-hidden="true" tabindex="-1"></a><span class="co">#                     test = "t",</span></span>
<span id="cb142-34"><a href="#cb142-34" aria-hidden="true" tabindex="-1"></a><span class="co">#                     data = dataset_lnRR_all_in)</span></span>
<span id="cb142-35"><a href="#cb142-35" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb142-36"><a href="#cb142-36" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb142-37"><a href="#cb142-37" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb142-38"><a href="#cb142-38" aria-hidden="true" tabindex="-1"></a><span class="co"># ## For SMDH complete published dataset</span></span>
<span id="cb142-39"><a href="#cb142-39" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb142-40"><a href="#cb142-40" aria-hidden="true" tabindex="-1"></a><span class="co"># dataset_SMDH_all_in&lt;-dataset_SMDH%&gt;%mutate(year_publication = case_when(is.na(year_publication) ~ 2025, TRUE ~ year_publication))%&gt;%</span></span>
<span id="cb142-41"><a href="#cb142-41" aria-hidden="true" tabindex="-1"></a><span class="co">#   mutate(year_MeanCenter=scale(as.numeric(year_publication),scale=F)[,1])%&gt;%</span></span>
<span id="cb142-42"><a href="#cb142-42" aria-hidden="true" tabindex="-1"></a><span class="co">#   mutate(effective_n_inv=1/((4*effective_n_experiment*effective_n_control)/(effective_n_experiment+effective_n_control)))%&gt;%</span></span>
<span id="cb142-43"><a href="#cb142-43" aria-hidden="true" tabindex="-1"></a><span class="co">#   mutate(sqrt_effective_n_inv=sqrt(effective_n_inv))%&gt;%</span></span>
<span id="cb142-44"><a href="#cb142-44" aria-hidden="true" tabindex="-1"></a><span class="co">#   mutate(comparision_type= case_when(is.na(comparision_type) ~ 4, TRUE ~ comparision_type))</span></span>
<span id="cb142-45"><a href="#cb142-45" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb142-46"><a href="#cb142-46" aria-hidden="true" tabindex="-1"></a><span class="co"># VCV_SMDH_all_in&lt;- vcalc(vi = SMDH_variance,</span></span>
<span id="cb142-47"><a href="#cb142-47" aria-hidden="true" tabindex="-1"></a><span class="co">#                   cluster = paper_ID,</span></span>
<span id="cb142-48"><a href="#cb142-48" aria-hidden="true" tabindex="-1"></a><span class="co">#                   obs = Observation_ID,</span></span>
<span id="cb142-49"><a href="#cb142-49" aria-hidden="true" tabindex="-1"></a><span class="co">#                   data = dataset_SMDH_all_in,</span></span>
<span id="cb142-50"><a href="#cb142-50" aria-hidden="true" tabindex="-1"></a><span class="co">#                   rho = 0.5 # assuming that the effect sizes within the same study are correlated with rho = 0.5</span></span>
<span id="cb142-51"><a href="#cb142-51" aria-hidden="true" tabindex="-1"></a><span class="co">#                   )</span></span>
<span id="cb142-52"><a href="#cb142-52" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb142-53"><a href="#cb142-53" aria-hidden="true" tabindex="-1"></a><span class="co"># SMDH_all_in&lt;- rma.mv(yi = SMDH_sign, # specify lnRR as the effect size measure;</span></span>
<span id="cb142-54"><a href="#cb142-54" aria-hidden="true" tabindex="-1"></a><span class="co">#                     V = VCV_SMDH_all_in, # specify lnRR's sampling variance-covariance matrix;</span></span>
<span id="cb142-55"><a href="#cb142-55" aria-hidden="true" tabindex="-1"></a><span class="co">#                     mods = ~ time_of_gnm_addition + </span></span>
<span id="cb142-56"><a href="#cb142-56" aria-hidden="true" tabindex="-1"></a><span class="co">#                       year_MeanCenter + </span></span>
<span id="cb142-57"><a href="#cb142-57" aria-hidden="true" tabindex="-1"></a><span class="co">#                       sqrt_effective_n_inv + </span></span>
<span id="cb142-58"><a href="#cb142-58" aria-hidden="true" tabindex="-1"></a><span class="co">#                       as.factor(comparision_type) + </span></span>
<span id="cb142-59"><a href="#cb142-59" aria-hidden="true" tabindex="-1"></a><span class="co">#                       bird_species + </span></span>
<span id="cb142-60"><a href="#cb142-60" aria-hidden="true" tabindex="-1"></a><span class="co">#                       trait_type,</span></span>
<span id="cb142-61"><a href="#cb142-61" aria-hidden="true" tabindex="-1"></a><span class="co">#                     random = list(~ 1 | paper_ID,</span></span>
<span id="cb142-62"><a href="#cb142-62" aria-hidden="true" tabindex="-1"></a><span class="co">#                                   ~ 1 | Observation_ID,</span></span>
<span id="cb142-63"><a href="#cb142-63" aria-hidden="true" tabindex="-1"></a><span class="co">#                                   ~ 1 | group_ID_coded,</span></span>
<span id="cb142-64"><a href="#cb142-64" aria-hidden="true" tabindex="-1"></a><span class="co">#                                   ~ 1 | population_ID,</span></span>
<span id="cb142-65"><a href="#cb142-65" aria-hidden="true" tabindex="-1"></a><span class="co">#                                   ~ 1 | bird_species),</span></span>
<span id="cb142-66"><a href="#cb142-66" aria-hidden="true" tabindex="-1"></a><span class="co">#                     method = "REML",</span></span>
<span id="cb142-67"><a href="#cb142-67" aria-hidden="true" tabindex="-1"></a><span class="co">#                     control = list(optimizer="optim"),</span></span>
<span id="cb142-68"><a href="#cb142-68" aria-hidden="true" tabindex="-1"></a><span class="co">#                     test = "t",</span></span>
<span id="cb142-69"><a href="#cb142-69" aria-hidden="true" tabindex="-1"></a><span class="co">#                     data = dataset_SMDH_all_in)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
</section>
<section id="pair-wise-comparison-for-meta-regression" class="level1">
<h1>Pair-wise comparison for meta-regression</h1>
<p>including including post-hoc Wald tests</p>
<p>For this I have to run all the meta-regressions again without removing the intercepts, good luck to me!</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [33]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb143"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb143-1"><a href="#cb143-1" aria-hidden="true" tabindex="-1"></a><span class="co">#----------------------------------------------------------------------------------</span></span>
<span id="cb143-2"><a href="#cb143-2" aria-hidden="true" tabindex="-1"></a><span class="co"># For Different hypothesis</span></span>
<span id="cb143-3"><a href="#cb143-3" aria-hidden="true" tabindex="-1"></a><span class="co">#----------------------------------------------------------------------------------</span></span>
<span id="cb143-4"><a href="#cb143-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb143-5"><a href="#cb143-5" aria-hidden="true" tabindex="-1"></a><span class="co"># lnRR_hypothesis_pc &lt;-</span></span>
<span id="cb143-6"><a href="#cb143-6" aria-hidden="true" tabindex="-1"></a><span class="co">#   metafor::rma.mv(yi = lnRR_sign, # specify lnRR as the effect size measure;</span></span>
<span id="cb143-7"><a href="#cb143-7" aria-hidden="true" tabindex="-1"></a><span class="co">#                     V = VCV_lnRR,# specify lnRR's sampling variance;</span></span>
<span id="cb143-8"><a href="#cb143-8" aria-hidden="true" tabindex="-1"></a><span class="co">#                     mods = ~ Hypothesis,</span></span>
<span id="cb143-9"><a href="#cb143-9" aria-hidden="true" tabindex="-1"></a><span class="co">#                     random = list(~ 1 | paper_ID,</span></span>
<span id="cb143-10"><a href="#cb143-10" aria-hidden="true" tabindex="-1"></a><span class="co">#                                   ~ 1 | Observation_ID,</span></span>
<span id="cb143-11"><a href="#cb143-11" aria-hidden="true" tabindex="-1"></a><span class="co">#                                   ~ 1 | group_ID_coded,</span></span>
<span id="cb143-12"><a href="#cb143-12" aria-hidden="true" tabindex="-1"></a><span class="co">#                                   ~ 1 | population_ID,</span></span>
<span id="cb143-13"><a href="#cb143-13" aria-hidden="true" tabindex="-1"></a><span class="co">#                                   ~ 1 | bird_species),</span></span>
<span id="cb143-14"><a href="#cb143-14" aria-hidden="true" tabindex="-1"></a><span class="co">#                     method = "REML",</span></span>
<span id="cb143-15"><a href="#cb143-15" aria-hidden="true" tabindex="-1"></a><span class="co">#                   test = "t",</span></span>
<span id="cb143-16"><a href="#cb143-16" aria-hidden="true" tabindex="-1"></a><span class="co">#                   control = list(optimizer="optim"),</span></span>
<span id="cb143-17"><a href="#cb143-17" aria-hidden="true" tabindex="-1"></a><span class="co">#                   data = dataset_lnRR)</span></span>
<span id="cb143-18"><a href="#cb143-18" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb143-19"><a href="#cb143-19" aria-hidden="true" tabindex="-1"></a><span class="co"># # I now want to test for statistical differences between groups. Male and fem</span></span>
<span id="cb143-20"><a href="#cb143-20" aria-hidden="true" tabindex="-1"></a><span class="co"># # information for androgens can be obtained from the outcome of the table. For thyroid</span></span>
<span id="cb143-21"><a href="#cb143-21" aria-hidden="true" tabindex="-1"></a><span class="co"># # hormones I need to do Post-hoc Wald tests: </span></span>
<span id="cb143-22"><a href="#cb143-22" aria-hidden="true" tabindex="-1"></a><span class="co"># car::linearHypothesis(lnRR_hypothesis_pc, rbind(c(0,1,-1)))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
     </main>
<!-- /main column -->  <script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>  </div> <!-- /content -->  <script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script> 
  
</body></html>